--- php-5.2.13/ext/standard/tests/file/fileinode_variation3.phpt~	2008-11-25 13:33:41.000000000 +0200
+++ php-5.2.13/ext/standard/tests/file/fileinode_variation3.phpt	2010-06-14 00:23:51.340272864 +0300
@@ -57,15 +57,15 @@
 --EXPECTF--
 *** Testing fileinode() with different notations of file names ***
 - Iteration 1 -
-int(%d)
+int(%i)
 - Iteration 2 -
 
 Warning: fileinode(): stat failed for %s//fileinode_variation3/fileinode_variation3.tmp/ in %s on line %d
 bool(false)
 - Iteration 3 -
-int(%d)
+int(%i)
 - Iteration 4 -
-int(%d)
+int(%i)
 - Iteration 5 -
 
 Warning: fileinode(): stat failed for %s//fileinode_variation3/*.tmp in %s on line %d
@@ -75,8 +75,8 @@
 Warning: fileinode(): stat failed for %s/fileinode_variation3/fileinode*.tmp in %s on line %d
 bool(false)
 - Iteration 7 -
-int(%d)
+int(%i)
 - Iteration 8 -
-int(%d)
+int(%i)
 
 *** Done ***
--- php-5.3.3/ext/standard/tests/file/fileinode_basic.phpt~	2007-11-05 19:43:21.000000000 +0200
+++ php-5.3.3/ext/standard/tests/file/fileinode_basic.phpt	2010-07-26 19:31:43.613553760 +0300
@@ -28,9 +28,9 @@
 ?>
 --EXPECTF--
 *** Testing fileinode() with file, directory ***
-%d
-%d
-%d
-%d
+%i
+%i
+%i
+%i
 
 *** Done ***
--- php-5.3.3/ext/standard/tests/file/fileinode_variation.phpt~	2007-11-05 19:43:21.000000000 +0200
+++ php-5.3.3/ext/standard/tests/file/fileinode_variation.phpt	2010-07-26 19:34:01.610414378 +0300
@@ -89,24 +89,24 @@
 --EXPECTF--
 *** Testing fileinode() with files, links and directories ***
 -- Testing with files --
-%d
-%d
+%i
+%i
 -- Testing with links: hard link --
-%d
-%d
+%i
+%i
 -- Testing with links: soft link --
-%d
-%d
+%i
+%i
 -- Testing after copying a file --
-%d
-%d
+%i
+%i
 -- Testing after renaming the file --
-%d
-%d
+%i
+%i
 -- Testing with directories --
-%d
-%d
+%i
+%i
 -- Testing with binary input --
-%d
-%d
+%i
+%i
 *** Done ***
--- php-5.3.3/ext/standard/tests/file/fileinode_variation1.phpt~	2008-11-26 12:10:19.000000000 +0200
+++ php-5.3.3/ext/standard/tests/file/fileinode_variation1.phpt	2010-07-26 19:35:31.216291870 +0300
@@ -42,7 +42,7 @@
 
 --EXPECTF--
 *** Testing fileinode() with links ***
-int(%d)
-int(%d)
+int(%i)
+int(%i)
 
 *** Done ***
--- php-5.3.3/ext/standard/tests/file/filestat.phpt~	2006-11-15 19:27:03.000000000 +0200
+++ php-5.3.3/ext/standard/tests/file/filestat.phpt	2010-07-26 19:37:10.367060798 +0300
@@ -30,17 +30,17 @@
 echo "Done\n";
 ?>
 --EXPECTF--	
+int(%i)
 int(%d)
 int(%d)
 int(%d)
 int(%d)
+int(%i)
 int(%d)
 int(%d)
 int(%d)
 int(%d)
-int(%d)
-int(%d)
-int(%d)
+int(%i)
 int(%d)
 int(%d)
 int(%d)
--- php-5.3.3/ext/standard/tests/file/lstat_stat_variation18.phpt~	2010-07-26 19:43:12.000000000 +0300
+++ php-5.3.3/ext/standard/tests/file/lstat_stat_variation18.phpt	2010-07-26 19:43:14.470984630 +0300
@@ -66,7 +66,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -92,7 +92,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -122,7 +122,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -148,7 +148,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
--- php-5.3.3/ext/standard/tests/file/lstat_stat_variation19.phpt~	2007-11-05 19:43:21.000000000 +0200
+++ php-5.3.3/ext/standard/tests/file/lstat_stat_variation19.phpt	2010-07-26 19:44:58.320079730 +0300
@@ -67,7 +67,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -93,7 +93,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -121,7 +121,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -147,7 +147,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -177,7 +177,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -203,7 +203,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -231,7 +231,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -257,7 +257,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
--- php-5.3.3/ext/standard/tests/file/lstat_stat_variation20.phpt~	2007-07-26 16:38:24.000000000 +0300
+++ php-5.3.3/ext/standard/tests/file/lstat_stat_variation20.phpt	2010-07-26 19:46:33.907004377 +0300
@@ -69,7 +69,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -95,7 +95,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -125,7 +125,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -151,7 +151,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -179,7 +179,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -205,7 +205,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -233,7 +233,7 @@
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -259,7 +259,7 @@
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
--- php-5.3.3/ext/standard/tests/file/tempnam_variation1.phpt~	2008-11-26 12:10:19.000000000 +0200
+++ php-5.3.3/ext/standard/tests/file/tempnam_variation1.phpt	2010-07-26 20:24:46.505943227 +0300
@@ -66,51 +66,51 @@
 -- Iteration 1 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 2 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 3 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 4 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 5 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 6 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 7 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 8 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 9 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 10 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 *** Done ***
--- php-5.3.3/tests/security/open_basedir_fileinode.phpt~	2008-05-09 11:39:44.000000000 +0300
+++ php-5.3.3/tests/security/open_basedir_fileinode.phpt	2010-07-27 22:51:29.603089549 +0300
@@ -46,10 +46,10 @@
 
 Warning: fileinode(): open_basedir restriction in effect. File(./../.) is not within the allowed path(s): (.) in %s on line %d
 bool(false)
-int(%d)
-int(%d)
-int(%d)
-int(%d)
-int(%d)
+int(%i)
+int(%i)
+int(%i)
+int(%i)
+int(%i)
 *** Finished testing open_basedir configuration [fileinode] ***
 
