diff -urNp -x '*.orig' php-7.1.33.org/ext/spl/tests/SplFileInfo_getInode_basic.phpt php-7.1.33/ext/spl/tests/SplFileInfo_getInode_basic.phpt
--- php-7.1.33.org/ext/spl/tests/SplFileInfo_getInode_basic.phpt	2021-08-23 23:46:46.137994907 +0200
+++ php-7.1.33/ext/spl/tests/SplFileInfo_getInode_basic.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -11,6 +11,7 @@ Jacopo Romei <jacopo@sviluppoagile.it>
 --SKIPIF--
 <?php
 if (substr(PHP_OS, 0, 3) == 'WIN') die("skip this test not for Windows platforms");
+if (PHP_INT_SIZE == 4) die("skip this test is for >32bit platform only (inodes overflow there)");
 ?>
 --FILE--
 <?php
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/fileinode_basic.phpt php-7.1.33/ext/standard/tests/file/fileinode_basic.phpt
--- php-7.1.33.org/ext/standard/tests/file/fileinode_basic.phpt	2019-10-22 19:00:00.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/fileinode_basic.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -28,9 +28,9 @@ unlink (dirname(__FILE__)."/inode.tmp");
 ?>
 --EXPECTF--
 *** Testing fileinode() with file, directory ***
-%d
-%d
-%d
-%d
+%i
+%i
+%i
+%i
 
 *** Done ***
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/fileinode_variation.phpt php-7.1.33/ext/standard/tests/file/fileinode_variation.phpt
--- php-7.1.33.org/ext/standard/tests/file/fileinode_variation.phpt	2019-10-22 19:00:00.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/fileinode_variation.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -88,24 +88,24 @@ echo "\n*** Done ***";
 --EXPECTF--
 *** Testing fileinode() with files, links and directories ***
 -- Testing with files --
-%d
-%d
+%i
+%i
 -- Testing with links: hard link --
-%d
-%d
+%i
+%i
 -- Testing with links: soft link --
-%d
-%d
+%i
+%i
 -- Testing after copying a file --
-%d
-%d
+%i
+%i
 -- Testing after renaming the file --
-%d
-%d
+%i
+%i
 -- Testing with directories --
-%d
-%d
+%i
+%i
 -- Testing with binary input --
-%d
-%d
+%i
+%i
 *** Done ***
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/fileinode_variation1.phpt php-7.1.33/ext/standard/tests/file/fileinode_variation1.phpt
--- php-7.1.33.org/ext/standard/tests/file/fileinode_variation1.phpt	2019-10-22 18:59:59.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/fileinode_variation1.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -41,7 +41,7 @@ unlink($file_path."/fileinode_variation1
 ?>
 --EXPECTF--
 *** Testing fileinode() with links ***
-int(%d)
-int(%d)
+int(%i)
+int(%i)
 
 *** Done ***
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/filestat.phpt php-7.1.33/ext/standard/tests/file/filestat.phpt
--- php-7.1.33.org/ext/standard/tests/file/filestat.phpt	2019-10-22 18:59:59.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/filestat.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -30,17 +30,17 @@ var_dump(filectime("/no/such/file/or/dir
 echo "Done\n";
 ?>
 --EXPECTF--
+int(%i)
 int(%d)
 int(%d)
 int(%d)
 int(%d)
+int(%i)
 int(%d)
 int(%d)
 int(%d)
 int(%d)
-int(%d)
-int(%d)
-int(%d)
+int(%i)
 int(%d)
 int(%d)
 int(%d)
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/lstat_stat_variation18.phpt php-7.1.33/ext/standard/tests/file/lstat_stat_variation18.phpt
--- php-7.1.33.org/ext/standard/tests/file/lstat_stat_variation18.phpt	2019-10-22 18:59:59.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/lstat_stat_variation18.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -65,7 +65,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -91,7 +91,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -121,7 +121,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -147,7 +147,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/lstat_stat_variation19.phpt php-7.1.33/ext/standard/tests/file/lstat_stat_variation19.phpt
--- php-7.1.33.org/ext/standard/tests/file/lstat_stat_variation19.phpt	2019-10-22 19:00:00.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/lstat_stat_variation19.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -66,7 +66,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -92,7 +92,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -120,7 +120,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -146,7 +146,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -176,7 +176,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -202,7 +202,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -230,7 +230,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -256,7 +256,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/lstat_stat_variation20.phpt php-7.1.33/ext/standard/tests/file/lstat_stat_variation20.phpt
--- php-7.1.33.org/ext/standard/tests/file/lstat_stat_variation20.phpt	2019-10-22 19:00:00.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/lstat_stat_variation20.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -68,7 +68,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -94,7 +94,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -124,7 +124,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -150,7 +150,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -178,7 +178,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -204,7 +204,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
@@ -232,7 +232,7 @@ array(26) {
   [0]=>
   int(%d)
   [1]=>
-  int(%d)
+  int(%i)
   [2]=>
   int(%d)
   [3]=>
@@ -258,7 +258,7 @@ array(26) {
   ["dev"]=>
   int(%d)
   ["ino"]=>
-  int(%d)
+  int(%i)
   ["mode"]=>
   int(%d)
   ["nlink"]=>
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/tests/file/tempnam_variation1.phpt php-7.1.33/ext/standard/tests/file/tempnam_variation1.phpt
--- php-7.1.33.org/ext/standard/tests/file/tempnam_variation1.phpt	2019-10-22 18:59:59.000000000 +0200
+++ php-7.1.33/ext/standard/tests/file/tempnam_variation1.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -66,51 +66,51 @@ echo "*** Done ***\n";
 -- Iteration 1 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 2 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 3 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 4 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 5 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 6 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 7 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 8 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 9 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 -- Iteration 10 --
 File name is => %s%etempnam_variation1.tmp%s
 File permissions are => 100600
-File inode is => %d
+File inode is => %i
 File created in => directory specified
 *** Done ***
diff -urNp -x '*.orig' php-7.1.33.org/tests/security/open_basedir_fileinode.phpt php-7.1.33/tests/security/open_basedir_fileinode.phpt
--- php-7.1.33.org/tests/security/open_basedir_fileinode.phpt	2019-10-22 18:59:36.000000000 +0200
+++ php-7.1.33/tests/security/open_basedir_fileinode.phpt	2021-08-23 23:46:47.097991082 +0200
@@ -46,9 +46,9 @@ bool(false)
 
 Warning: fileinode(): open_basedir restriction in effect. File(./../.) is not within the allowed path(s): (.) in %s on line %d
 bool(false)
-int(%d)
-int(%d)
-int(%d)
-int(%d)
-int(%d)
+int(%i)
+int(%i)
+int(%i)
+int(%i)
+int(%i)
 *** Finished testing open_basedir configuration [fileinode] ***
