diff -urNp -x '*.orig' php-7.2.34.org/Makefile.global php-7.2.34/Makefile.global
--- php-7.2.34.org/Makefile.global	2021-08-27 22:34:16.728238938 +0200
+++ php-7.2.34/Makefile.global	2021-08-27 22:34:17.704884843 +0200
@@ -81,8 +81,15 @@ install-headers:
 		done; \
 	fi
 
-PHP_TEST_SETTINGS = -d 'open_basedir=' -d 'output_buffering=0' -d 'memory_limit=-1'
+PHP_TEST_SETTINGS = -d 'open_basedir=' -d 'output_buffering=0' -d 'memory_limit=-1' -d 'safe_mode=0'
+PHP_TEST_SHARED_SYSTEM_EXTENSIONS =
 PHP_TEST_SHARED_EXTENSIONS =  ` \
+	if test "x$(PHP_TEST_SHARED_SYSTEM_EXTENSIONS)" != "x"; then \
+		for i in $(PHP_TEST_SHARED_SYSTEM_EXTENSIONS)""; do \
+			dlname=$$i.$(SHLIB_DL_SUFFIX_NAME); \
+			$(top_srcdir)/build/shtool echo -n -- " -d extension=$(EXTENSION_DIR)/$$dlname"; \
+		done; \
+	fi; \
 	if test "x$(PHP_MODULES)" != "x"; then \
 		for i in $(PHP_MODULES)""; do \
 			. $$i; $(top_srcdir)/build/shtool echo -n -- " -d extension=$$dlname"; \
@@ -111,7 +118,10 @@ test: all
 		TEST_PHP_EXECUTABLE=$(PHP_EXECUTABLE) \
 		TEST_PHP_SRCDIR=$(top_srcdir) \
 		CC="$(CC)" \
-			$(PHP_EXECUTABLE) -n -c $(top_builddir)/tmp-php.ini $(PHP_TEST_SETTINGS) $(top_srcdir)/run-tests.php -n -c $(top_builddir)/tmp-php.ini -d extension_dir=$(top_builddir)/modules/ $(PHP_TEST_SHARED_EXTENSIONS) $(TESTS); \
+			$(PHP_EXECUTABLE) -n -c $(top_builddir)/tmp-php.ini \
+			-d extension_dir=$(top_builddir)/modules/ -d 'extension=$(EXTENSION_DIR)/pcre.$(SHLIB_DL_SUFFIX_NAME)' \
+			$(PHP_TEST_SETTINGS) $(top_srcdir)/run-tests.php -n -c $(top_builddir)/tmp-php.ini \
+			-d extension_dir=$(top_builddir)/modules/ $(PHP_TEST_SHARED_EXTENSIONS) $(RUN_TESTS_SETTINGS) $(TESTS); \
 		TEST_RESULT_EXIT_CODE=$$?; \
 		rm $(top_builddir)/tmp-php.ini; \
 		exit $$TEST_RESULT_EXIT_CODE; \
diff -urNp -x '*.orig' php-7.2.34.org/configure.ac php-7.2.34/configure.ac
--- php-7.2.34.org/configure.ac	2021-08-27 22:34:16.738238725 +0200
+++ php-7.2.34/configure.ac	2021-08-27 22:34:17.704884843 +0200
@@ -1305,6 +1305,11 @@ AC_DEFINE_UNQUOTED(PHP_UNAME,"$PHP_UNAME
 PHP_OS=`uname | xargs`
 AC_DEFINE_UNQUOTED(PHP_OS,"$PHP_OS",[uname output])
 
+# shift so that pcre, spl, ... are loaded first
+PHP_MODULES=$(echo "$PHP_MODULES" | sed -e 's,\(.*\)\(\$(phplibdir)/xml.la \),\2\1,')
+PHP_MODULES=$(echo "$PHP_MODULES" | sed -e 's,\(.*\)\(\$(phplibdir)/spl.la \),\2\1,')
+PHP_MODULES=$(echo "$PHP_MODULES" | sed -e 's,\(.*\)\(\$(phplibdir)/pcre.la \),\2\1,')
+
 PHP_SUBST_OLD(PHP_INSTALLED_SAPIS)
 
 PHP_SUBST(PHP_EXECUTABLE)
