--- php-8.4.1/configure.ac~	2024-11-22 11:30:13.000000000 +0100
+++ php-8.4.1/configure.ac	2024-11-22 11:32:52.836654521 +0100
@@ -1428,6 +1428,10 @@ PHP_SUBST([PHP_SAPI_OBJS])
 PHP_SUBST([PHP_BINARY_OBJS])
 PHP_SUBST([PHP_GLOBAL_OBJS])
 PHP_SUBST([PHP_BINARIES])
+
+# shift so that extensions like xml are loaded first
+PHP_MODULES=$(echo "$PHP_MODULES" | sed -e 's,\(.*\)\(\$(phplibdir)/xml.la \),\2\1,')
+
 PHP_SUBST([PHP_MODULES])
 PHP_SUBST([PHP_ZEND_EX])
 PHP_SUBST([bindir])
--- php-8.1.10/build/Makefile.global.orig	2022-09-25 19:52:15.520810107 +0200
+++ php-8.1.10/build/Makefile.global	2022-09-25 19:54:52.676625388 +0200
@@ -81,8 +81,15 @@ install-headers:
 		done; \
 	fi
 
-PHP_TEST_SETTINGS = -d 'open_basedir=' -d 'output_buffering=0' -d 'memory_limit=-1'
+PHP_TEST_SETTINGS = -d 'open_basedir=' -d 'output_buffering=0' -d 'memory_limit=-1' -d 'safe_mode=0'
+PHP_TEST_SHARED_SYSTEM_EXTENSIONS =
 PHP_TEST_SHARED_EXTENSIONS =  ` \
+	if test "x$(PHP_TEST_SHARED_SYSTEM_EXTENSIONS)" != "x"; then \
+		for i in $(PHP_TEST_SHARED_SYSTEM_EXTENSIONS)""; do \
+			dlname=$$i.$(SHLIB_DL_SUFFIX_NAME); \
+			$(top_srcdir)/build/shtool echo -n -- " -d extension=$(EXTENSION_DIR)/$$dlname"; \
+		done; \
+	fi; \
 	if test "x$(PHP_MODULES)" != "x"; then \
 		for i in $(PHP_MODULES)""; do \
 			. $$i; \
@@ -114,7 +121,9 @@ test: all
 		TEST_PHP_EXECUTABLE=$(PHP_EXECUTABLE) \
 		TEST_PHP_SRCDIR=$(top_srcdir) \
 		CC="$(CC)" \
-			$(PHP_EXECUTABLE) -n -c $(top_builddir)/tmp-php.ini $(PHP_TEST_SETTINGS) $(top_srcdir)/run-tests.php -n -c $(top_builddir)/tmp-php.ini -d extension_dir=$(top_builddir)/modules/ $(PHP_TEST_SHARED_EXTENSIONS) $(TESTS); \
+			$(PHP_EXECUTABLE) -n -c $(top_builddir)/tmp-php.ini \
+			$(PHP_TEST_SETTINGS) $(top_srcdir)/run-tests.php -n -c $(top_builddir)/tmp-php.ini \
+			-d extension_dir=$(top_builddir)/modules/ $(PHP_TEST_SHARED_EXTENSIONS) $(RUN_TESTS_SETTINGS) $(TESTS); \
 		TEST_RESULT_EXIT_CODE=$$?; \
 		rm $(top_builddir)/tmp-php.ini; \
 		exit $$TEST_RESULT_EXIT_CODE; \
