--- php-7.4.33/sapi/litespeed/lsapi_main.c.orig	2022-10-31 11:36:05.000000000 +0100
+++ php-7.4.33/sapi/litespeed/lsapi_main.c	2025-01-23 01:03:28.685293073 +0100
@@ -24,6 +24,7 @@
 #include "zend_highlight.h"
 #include "zend.h"
 #include "ext/standard/basic_functions.h"
+#include "ext/standard/head.h"
 #include "ext/standard/info.h"
 #include "lsapilib.h"
 
--- php-7.1.33/ext/zip/lib/mkstemp.c.orig	2019-10-22 18:59:49.000000000 +0200
+++ php-7.1.33/ext/zip/lib/mkstemp.c	2025-01-25 00:53:13.113061891 +0100
@@ -43,6 +43,7 @@
 #endif
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #ifndef O_BINARY
 #define O_BINARY 0
