--- php-5.6.40/sapi/litespeed/lsapi_main.c.orig	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/sapi/litespeed/lsapi_main.c	2025-01-27 22:38:03.269825030 +0100
@@ -23,6 +23,8 @@
 #include "php_variables.h"
 #include "zend_highlight.h"
 #include "zend.h"
+#include "ext/standard/basic_functions.h"
+#include "ext/standard/info.h"
 
 #include "lsapilib.h"
 
--- php-5.4.45/ext/gd/gd.c.orig	2015-09-01 22:09:37.000000000 +0200
+++ php-5.4.45/ext/gd/gd.c	2025-01-30 11:13:10.169237812 +0100
@@ -74,6 +74,8 @@
 #include <gdfontl.h>  /* 4 Large font */
 #include <gdfontg.h>  /* 5 Giant font */
 
+#include "gdhelpers.h"
+
 #ifdef HAVE_GD_WBMP
 #include "libgd/wbmp.h"
 #endif
--- php-5.3.29/sapi/litespeed/lsapi_main.c.orig	2025-01-30 12:21:08.288824792 +0100
+++ php-5.3.29/sapi/litespeed/lsapi_main.c	2025-01-30 12:40:58.378839232 +0100
@@ -27,6 +27,7 @@
 #include "zend.h"
 #include "ext/standard/basic_functions.h"
 #include "ext/standard/info.h"
+#include "ext/standard/php_string.h"
 
 #include "lsapilib.h"
 
