--- php-5.6.0RC4/sapi/milter/config.m4~	2014-08-14 03:54:13.000000000 +0300
+++ php-5.6.0RC4/sapi/milter/config.m4	2014-08-21 19:57:21.584244904 +0300
@@ -6,18 +6,13 @@
 [  --with-milter[=DIR]       Build PHP as Milter application], no, no)
 
 if test "$PHP_MILTER" != "no"; then
-  if test "$PHP_MILTER" = "yes"; then
-    if test -f /usr/lib/libmilter.a ; then
-      MILTERPATH=/usr/lib
-    else
-      if test -f /usr/lib/libmilter/libmilter.a ; then
-        MILTERPATH=/usr/lib/libmilter
-      else
-        AC_MSG_ERROR([Unable to find libmilter.a])
-      fi
-    fi
-  else
-    MILTERPATH=$PHP_MILTER
+  for i in $PHP_MILTER /usr/lib/libmilter /usr/local /usr; do
+    test -f $i/$PHP_LIBDIR/libmilter.a && MILTERPATH=$i/$PHP_LIBDIR && break
+    test -f $i/libmilter.a && MILTERPATH=$i && break
+  done
+
+  if test -z "$MILTERPATH"; then
+    AC_MSG_ERROR([Unable to find libmilter.a])
   fi
   
   SAPI_MILTER_PATH=sapi/milter/php-milter
