diff -urNp -x '*.orig' php-7.1.33.org/ext/mysqli/mysqli_nonapi.c php-7.1.33/ext/mysqli/mysqli_nonapi.c
--- php-7.1.33.org/ext/mysqli/mysqli_nonapi.c	2019-10-22 18:59:53.000000000 +0200
+++ php-7.1.33/ext/mysqli/mysqli_nonapi.c	2021-08-23 23:46:59.891277884 +0200
@@ -72,7 +72,7 @@ void mysqli_common_connect(INTERNAL_FUNC
 	zend_bool			self_alloced = 0;
 
 
-#if !defined(MYSQL_USE_MYSQLND)
+#if 0 && !defined(MYSQL_USE_MYSQLND)
 	if ((MYSQL_VERSION_ID / 100) != (mysql_get_client_version() / 100)) {
 		php_error_docref(NULL, E_WARNING,
 						"Headers and client library minor version mismatch. Headers:%d Library:%ld",
