--- php-5.4.0alpha2/ext/mysqlnd/config9.m4~	2011-03-23 19:14:28.000000000 +0200
+++ php-5.4.0alpha2/ext/mysqlnd/config9.m4	2011-08-07 12:45:42.205921715 +0300
@@ -28,7 +28,7 @@
   AC_DEFINE([MYSQLND_SSL_SUPPORTED], 1, [Enable SSL support])
 
   mysqlnd_sources="$mysqlnd_base_sources $mysqlnd_ps_sources"
-  PHP_NEW_EXTENSION(mysqlnd, $mysqlnd_sources, no)
+  PHP_NEW_EXTENSION(mysqlnd, $mysqlnd_sources, yes)
   PHP_ADD_BUILD_DIR([ext/mysqlnd], 1)
   PHP_INSTALL_HEADERS([ext/mysqlnd/])
 fi
--- php-5.3.2/ext/mysqlnd/php_mysqlnd.c~	2010-01-20 19:09:28.000000000 +0200
+++ php-5.3.2/ext/mysqlnd/php_mysqlnd.c	2010-04-14 09:18:30.725599897 +0300
@@ -273,6 +273,9 @@
 };
 /* }}} */
 
+#ifdef COMPILE_DL_MYSQLND
+ZEND_GET_MODULE(mysqlnd)
+#endif
 
 /*
  * Local variables:
