diff -urNp -x '*.orig' php-5.5.38.org/ext/mysql/php_mysql.c php-5.5.38/ext/mysql/php_mysql.c
--- php-5.5.38.org/ext/mysql/php_mysql.c	2021-08-23 23:03:46.113256764 +0200
+++ php-5.5.38/ext/mysql/php_mysql.c	2021-08-23 23:03:47.269922856 +0200
@@ -738,7 +738,7 @@ static void php_mysql_do_connect(INTERNA
                      E_DEPRECATED,
                      "The mysql extension is deprecated and will be removed in the future: use mysqli or PDO instead");
 
-#if !defined(MYSQL_USE_MYSQLND)
+#if 0 && !defined(MYSQL_USE_MYSQLND)
 	if ((MYSQL_VERSION_ID / 100) != (mysql_get_client_version() / 100)) {
 		php_error_docref(NULL TSRMLS_CC, E_NOTICE,
 						"Headers and client library minor version mismatch. Headers:%d Library:%ld",
diff -urNp -x '*.orig' php-5.5.38.org/ext/mysqli/mysqli_nonapi.c php-5.5.38/ext/mysqli/mysqli_nonapi.c
--- php-5.5.38.org/ext/mysqli/mysqli_nonapi.c	2021-08-23 23:03:46.113256764 +0200
+++ php-5.5.38/ext/mysqli/mysqli_nonapi.c	2021-08-23 23:03:47.269922856 +0200
@@ -74,7 +74,7 @@ void mysqli_common_connect(INTERNAL_FUNC
 	zend_bool			self_alloced = 0;
 
 
-#if !defined(MYSQL_USE_MYSQLND)
+#if 0 && !defined(MYSQL_USE_MYSQLND)
 	if ((MYSQL_VERSION_ID / 100) != (mysql_get_client_version() / 100)) {
 		php_error_docref(NULL TSRMLS_CC, E_NOTICE,
 						"Headers and client library minor version mismatch. Headers:%d Library:%ld",
