diff -urNp -x '*.orig' php-5.2.17.org/ext/mysqli/mysqli.c php-5.2.17/ext/mysqli/mysqli.c
--- php-5.2.17.org/ext/mysqli/mysqli.c	2021-10-23 19:15:35.113125046 +0200
+++ php-5.2.17/ext/mysqli/mysqli.c	2021-10-23 19:15:37.713125046 +0200
@@ -1063,7 +1063,7 @@ int php_local_infile_init(void **ptr, co
 	}
 
 	if (!(mysql = (MY_MYSQL *)userdata)) {
-		LOCAL_INFILE_ERROR_MSG(data->error_msg, ER(CR_UNKNOWN_ERROR));
+		LOCAL_INFILE_ERROR_MSG(data->error_msg, "Unknown MySQL error");
 		return 1;
 	}
 
@@ -1112,7 +1112,7 @@ int php_local_infile_read(void *ptr, cha
 		count = (int)php_stream_read(mysql->li_stream, buf, buf_len);
 
 		if (count < 0) {
-			LOCAL_INFILE_ERROR_MSG(data->error_msg, ER(2));
+			LOCAL_INFILE_ERROR_MSG(data->error_msg, "Can't connect to local MySQL server");
 		}
 
 		return count;
@@ -1174,7 +1174,7 @@ int php_local_infile_error(void *ptr, ch
 		strlcpy(error_msg, data->error_msg, error_msg_len);
 		return 2000;
 	} 
-	strlcpy(error_msg, ER(CR_OUT_OF_MEMORY), error_msg_len);
+	strlcpy(error_msg, "MySQL client ran out of memory", error_msg_len);
 	return CR_OUT_OF_MEMORY;
 }
 /* }}} */
