--- php-5.3.27/ext/mysqli/mysqli_nonapi.c~	2013-11-20 00:19:11.000000000 +0200
+++ php-5.3.27/ext/mysqli/mysqli_nonapi.c	2013-11-20 00:19:44.725872552 +0200
@@ -73,7 +73,7 @@
	bool			self_alloced = 0;


-#if !defined(MYSQL_USE_MYSQLND)
+#if 0 && !defined(MYSQL_USE_MYSQLND)
 	if ((MYSQL_VERSION_ID / 100) != (mysql_get_client_version() / 100)) {
 		php_error_docref(NULL, E_WARNING,
 						"Headers and client library minor version mismatch. Headers:%d Library:%ld",
