diff -urNp -x '*.orig' php-7.0.33.org/ext/mysqli/mysqli_nonapi.c php-7.0.33/ext/mysqli/mysqli_nonapi.c
--- php-7.0.33.org/ext/mysqli/mysqli_nonapi.c	2018-12-04 18:22:49.000000000 +0100
+++ php-7.0.33/ext/mysqli/mysqli_nonapi.c	2021-08-23 23:36:57.788993126 +0200
@@ -72,7 +72,7 @@ void mysqli_common_connect(INTERNAL_FUNC
 	zend_bool			self_alloced = 0;
 
 
-#if !defined(MYSQL_USE_MYSQLND)
+#if 0 && !defined(MYSQL_USE_MYSQLND)
 	if ((MYSQL_VERSION_ID / 100) != (mysql_get_client_version() / 100)) {
 		php_error_docref(NULL, E_WARNING,
 						"Headers and client library minor version mismatch. Headers:%d Library:%ld",
