diff -urNp -x '*.orig' php-7.1.33.org/ext/pcre/config0.m4 php-7.1.33/ext/pcre/config0.m4
--- php-7.1.33.org/ext/pcre/config0.m4	2019-10-22 18:59:47.000000000 +0200
+++ php-7.1.33/ext/pcre/config0.m4	2021-08-23 23:46:42.121344768 +0200
@@ -11,7 +11,7 @@ PHP_ARG_WITH(pcre-regex,,
 
   if test "$PHP_PCRE_REGEX" != "yes" && test "$PHP_PCRE_REGEX" != "no"; then
     AC_MSG_CHECKING([for PCRE headers location])
-    for i in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/include $PHP_PCRE_REGEX/include/pcre $PHP_PCRE_REGEX/local/include; do
+    for i in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/include $PHP_PCRE_REGEX/include/pcre $PHP_PCRE_REGEX/local/include /usr/include; do
       test -f $i/pcre.h && PCRE_INCDIR=$i
     done
 
@@ -21,7 +21,7 @@ PHP_ARG_WITH(pcre-regex,,
     AC_MSG_RESULT([$PCRE_INCDIR])
 
     AC_MSG_CHECKING([for PCRE library location])
-    for j in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/$PHP_LIBDIR; do
+    for j in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/$PHP_LIBDIR /usr/$PHP_LIBDIR; do
       test -f $j/libpcre.a || test -f $j/libpcre.$SHLIB_SUFFIX_NAME && PCRE_LIBDIR=$j
     done
 
@@ -43,12 +43,13 @@ PHP_ARG_WITH(pcre-regex,,
       AC_MSG_ERROR([The PCRE extension requires PCRE library version >= 6.6])
     fi
 
-    PHP_ADD_LIBRARY_WITH_PATH(pcre, $PCRE_LIBDIR)
+    PHP_ADD_LIBRARY_WITH_PATH(pcre, $PCRE_LIBDIR, PCRE_SHARED_LIBADD)
 
     AC_DEFINE(HAVE_PCRE, 1, [ ])
     PHP_ADD_INCLUDE($PCRE_INCDIR)
-    PHP_NEW_EXTENSION(pcre, php_pcre.c, no,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+    PHP_NEW_EXTENSION(pcre, php_pcre.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
     PHP_INSTALL_HEADERS([ext/pcre], [php_pcre.h])
+    EXTRA_LIBS="$EXTRA_LIBS -lpcre"
   else
     AC_MSG_CHECKING([for PCRE library to use])
     AC_MSG_RESULT([bundled])
@@ -61,7 +62,7 @@ PHP_ARG_WITH(pcre-regex,,
     				 pcrelib/pcre_version.c pcrelib/pcre_xclass.c \
     				 pcrelib/pcre_jit_compile.c"
     PHP_PCRE_CFLAGS="-DHAVE_CONFIG_H -I@ext_srcdir@/pcrelib -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1"
-    PHP_NEW_EXTENSION(pcre, $pcrelib_sources php_pcre.c, no,,$PHP_PCRE_CFLAGS)
+    PHP_NEW_EXTENSION(pcre, $pcrelib_sources php_pcre.c, $ext_shared,,$PHP_PCRE_CFLAGS)
     PHP_ADD_BUILD_DIR($ext_builddir/pcrelib)
     PHP_INSTALL_HEADERS([ext/pcre], [php_pcre.h pcrelib/])
     AC_DEFINE(HAVE_BUNDLED_PCRE, 1, [ ])
diff -urNp -x '*.orig' php-7.1.33.org/ext/pcre/php_pcre.c php-7.1.33/ext/pcre/php_pcre.c
--- php-7.1.33.org/ext/pcre/php_pcre.c	2019-10-22 18:59:47.000000000 +0200
+++ php-7.1.33/ext/pcre/php_pcre.c	2021-08-23 23:46:42.121344768 +0200
@@ -28,6 +28,11 @@
 
 #if HAVE_PCRE || HAVE_BUNDLED_PCRE
 
+#if COMPILE_DL_PCRE
+#undef pcre_get_compiled_regex
+PHPAPI pcre* pcre_get_compiled_regex(zend_string *regex, pcre_extra **extra, int *options);
+#endif
+
 #include "ext/standard/php_string.h"
 
 #define PREG_PATTERN_ORDER			1
@@ -216,6 +221,10 @@ static PHP_MINIT_FUNCTION(pcre)
 	REGISTER_LONG_CONSTANT("PREG_JIT_STACKLIMIT_ERROR", PHP_PCRE_JIT_STACKLIMIT_ERROR, CONST_CS | CONST_PERSISTENT);
 	REGISTER_STRING_CONSTANT("PCRE_VERSION", (char *)pcre_version(), CONST_CS | CONST_PERSISTENT);
 
+#if COMPILE_DL_PCRE
+	pcre_get_compiled_regex_p = pcre_get_compiled_regex;
+#endif
+
 	return SUCCESS;
 }
 /* }}} */
@@ -630,6 +639,7 @@ PHPAPI pcre_cache_entry* pcre_get_compil
 }
 /* }}} */
 
+
 /* {{{ pcre_get_compiled_regex
  */
 PHPAPI pcre* pcre_get_compiled_regex(zend_string *regex, pcre_extra **extra, int *preg_options)
diff -urNp -x '*.orig' php-7.1.33.org/ext/pcre/php_pcre.h php-7.1.33/ext/pcre/php_pcre.h
--- php-7.1.33.org/ext/pcre/php_pcre.h	2019-10-22 18:59:47.000000000 +0200
+++ php-7.1.33/ext/pcre/php_pcre.h	2021-08-23 23:46:42.121344768 +0200
@@ -34,7 +34,12 @@
 #endif
 
 PHPAPI zend_string *php_pcre_replace(zend_string *regex, zend_string *subject_str, char *subject, int subject_len, zval *replace_val, int is_callable_replace, int limit, int *replace_count);
+#if COMPILE_DL_PCRE
+#define pcre_get_compiled_regex pcre_get_compiled_regex_p
+extern PHPAPI pcre* (*pcre_get_compiled_regex)(zend_string *regex, pcre_extra **extra, int *options);
+#else
 PHPAPI pcre* pcre_get_compiled_regex(zend_string *regex, pcre_extra **extra, int *options);
+#endif
 PHPAPI pcre* pcre_get_compiled_regex_ex(zend_string *regex, pcre_extra **extra, int *preg_options, int *coptions);
 
 extern zend_module_entry pcre_module_entry;
diff -urNp -x '*.orig' php-7.1.33.org/ext/standard/browscap.c php-7.1.33/ext/standard/browscap.c
--- php-7.1.33.org/ext/standard/browscap.c	2019-10-22 19:00:02.000000000 +0200
+++ php-7.1.33/ext/standard/browscap.c	2021-08-23 23:46:42.121344768 +0200
@@ -65,6 +65,11 @@ ZEND_END_MODULE_GLOBALS(browscap)
 ZEND_DECLARE_MODULE_GLOBALS(browscap)
 #define BROWSCAP_G(v) ZEND_MODULE_GLOBALS_ACCESSOR(browscap, v)
 
+#if COMPILE_DL_PCRE
+// will be visible in here
+PHPAPI pcre* (*pcre_get_compiled_regex_p)(zend_string *regex, pcre_extra **extra, int *options);
+#endif
+
 #define DEFAULT_SECTION_NAME "Default Browser Capability Settings"
 
 /* OBJECTS_FIXME: This whole extension needs going through. The use of objects looks pretty broken here */
diff -urNp -x '*.orig' php-7.1.33.org/main/php_ini.c php-7.1.33/main/php_ini.c
--- php-7.1.33.org/main/php_ini.c	2021-08-23 23:46:41.154682209 +0200
+++ php-7.1.33/main/php_ini.c	2021-08-23 23:46:42.121344768 +0200
@@ -771,8 +771,8 @@ int php_shutdown_config(void)
  */
 void php_ini_register_extensions(void)
 {
-	zend_llist_apply(&extension_lists.engine, php_load_zend_extension_cb);
 	zend_llist_apply(&extension_lists.functions, php_load_php_extension_cb);
+	zend_llist_apply(&extension_lists.engine, php_load_zend_extension_cb);
 
 	zend_llist_destroy(&extension_lists.engine);
 	zend_llist_destroy(&extension_lists.functions);
diff -urNp -x '*.orig' php-7.1.33.org/sapi/cli/tests/018.phpt php-7.1.33/sapi/cli/tests/018.phpt
--- php-7.1.33.org/sapi/cli/tests/018.phpt	2019-10-22 19:00:03.000000000 +0200
+++ php-7.1.33/sapi/cli/tests/018.phpt	2021-08-23 23:46:42.121344768 +0200
@@ -20,8 +20,6 @@ echo "Done\n";
 --EXPECTF--
 [PHP Modules]
 %a
-pcre
-%a
 
 [Zend Modules]
 %aDone
