--- php-7.0/ext/pcre/config0.m4~	2015-05-24 14:39:04.000000000 +0300
+++ php-7.0/ext/pcre/config0.m4	2015-05-24 14:53:49.980250297 +0300
@@ -11,7 +11,7 @@
 
   if test "$PHP_PCRE_REGEX" != "yes" && test "$PHP_PCRE_REGEX" != "no"; then
     AC_MSG_CHECKING([for PCRE headers location])
-    for i in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/include $PHP_PCRE_REGEX/include/pcre $PHP_PCRE_REGEX/local/include; do
+    for i in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/include $PHP_PCRE_REGEX/include/pcre $PHP_PCRE_REGEX/local/include /usr/include; do
       test -f $i/pcre.h && PCRE_INCDIR=$i
     done
 
@@ -21,7 +21,7 @@
     AC_MSG_RESULT([$PCRE_INCDIR])
 
     AC_MSG_CHECKING([for PCRE library location])
-    for j in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/$PHP_LIBDIR; do
+    for j in $PHP_PCRE_REGEX $PHP_PCRE_REGEX/$PHP_LIBDIR /usr/$PHP_LIBDIR; do
       test -f $j/libpcre.a || test -f $j/libpcre.$SHLIB_SUFFIX_NAME && PCRE_LIBDIR=$j
     done
     
@@ -43,12 +43,13 @@
       AC_MSG_ERROR([The PCRE extension requires PCRE library version >= 6.6])
     fi
 
-    PHP_ADD_LIBRARY_WITH_PATH(pcre, $PCRE_LIBDIR)
+    PHP_ADD_LIBRARY_WITH_PATH(pcre, $PCRE_LIBDIR, PCRE_SHARED_LIBADD)
     
     AC_DEFINE(HAVE_PCRE, 1, [ ])
     PHP_ADD_INCLUDE($PCRE_INCDIR)
-    PHP_NEW_EXTENSION(pcre, php_pcre.c, no,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+    PHP_NEW_EXTENSION(pcre, php_pcre.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
     PHP_INSTALL_HEADERS([ext/pcre], [php_pcre.h])
+    EXTRA_LIBS="$EXTRA_LIBS -lpcre"
   else
     AC_MSG_CHECKING([for PCRE library to use])
     AC_MSG_RESULT([bundled])
@@ -61,7 +62,7 @@
     				 pcrelib/pcre_version.c pcrelib/pcre_xclass.c \
     				 pcrelib/pcre_jit_compile.c"
     PHP_PCRE_CFLAGS="-DHAVE_CONFIG_H -I@ext_srcdir@/pcrelib -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1"
-    PHP_NEW_EXTENSION(pcre, $pcrelib_sources php_pcre.c, no,,$PHP_PCRE_CFLAGS)
+    PHP_NEW_EXTENSION(pcre, $pcrelib_sources php_pcre.c, $ext_shared,,$PHP_PCRE_CFLAGS)
     PHP_ADD_BUILD_DIR($ext_builddir/pcrelib)
     PHP_INSTALL_HEADERS([ext/pcre], [php_pcre.h pcrelib/])
     AC_DEFINE(HAVE_BUNDLED_PCRE, 1, [ ])
--- php-7.2.0alpha3/ext/pcre/php_pcre.h~	2017-07-04 12:26:40.000000000 +0300
+++ php-7.2.0alpha3/ext/pcre/php_pcre.h	2017-07-07 01:45:58.017098044 +0300
@@ -34,7 +34,12 @@
 #endif
 
 PHPAPI zend_string *php_pcre_replace(zend_string *regex, zend_string *subject_str, char *subject, int subject_len, zend_string *replace_str, int limit, int *replace_count);
+#if COMPILE_DL_PCRE
+#define pcre_get_compiled_regex pcre_get_compiled_regex_p
+extern PHPAPI pcre* (*pcre_get_compiled_regex)(zend_string *regex, pcre_extra **extra, int *options);
+#else
 PHPAPI pcre* pcre_get_compiled_regex(zend_string *regex, pcre_extra **extra, int *options);
+#endif
 PHPAPI pcre* pcre_get_compiled_regex_ex(zend_string *regex, pcre_extra **extra, int *preg_options, int *coptions);
 
 extern zend_module_entry pcre_module_entry;
--- php-5.3.1/ext/pcre/php_pcre.c	2009-11-30 21:10:01.370473754 +0200
+++ php-5.3.1.pcre/ext/pcre/php_pcre.c	2009-11-30 21:38:01.759684456 +0000
@@ -27,6 +27,11 @@
 
 #if HAVE_PCRE || HAVE_BUNDLED_PCRE
 
+#if COMPILE_DL_PCRE
+#undef pcre_get_compiled_regex
+PHPAPI pcre* pcre_get_compiled_regex(zend_string *regex, pcre_extra **extra, int *options);
+#endif
+
 #include "ext/standard/php_string.h"
 
 #define PREG_PATTERN_ORDER			1
@@ -153,6 +158,10 @@
 	REGISTER_LONG_CONSTANT("PREG_BAD_UTF8_OFFSET_ERROR", PHP_PCRE_BAD_UTF8_OFFSET_ERROR, CONST_CS | CONST_PERSISTENT);
 	REGISTER_STRING_CONSTANT("PCRE_VERSION", (char *)pcre_version(), CONST_CS | CONST_PERSISTENT);
 
+#if COMPILE_DL_PCRE
+	pcre_get_compiled_regex_p = pcre_get_compiled_regex;
+#endif
+
 	return SUCCESS;
 }
 /* }}} */
@@ -429,6 +438,7 @@
 }
 /* }}} */
 
+
 /* {{{ pcre_get_compiled_regex
  */
 PHPAPI pcre* pcre_get_compiled_regex(char *regex, pcre_extra **extra, int *preg_options TSRMLS_DC)
--- php-5.3.1/ext/standard/browscap.c	2009-06-06 02:40:49.000000000 +0000
+++ php-5.3.1.pcre/ext/standard/browscap.c	2009-11-30 21:33:00.775241138 +0000
@@ -31,6 +31,11 @@
 static zval *current_section;
 static char *current_section_name;
 
+#if COMPILE_DL_PCRE
+// will be visible in here
+PHPAPI pcre* (*pcre_get_compiled_regex_p)(zend_string *regex, pcre_extra **extra, int *options);
+#endif
+
 #define DEFAULT_SECTION_NAME "Default Browser Capability Settings"
 
 /* OBJECTS_FIXME: This whole extension needs going through. The use of objects looks pretty broken here */
--- php-5.3.2/sapi/cli/tests/018.phpt~	2008-03-17 16:05:39.000000000 +0200
+++ php-5.3.2/sapi/cli/tests/018.phpt	2010-03-16 20:41:11.341251246 +0200
@@ -20,8 +20,6 @@
 --EXPECTF--     
 [PHP Modules]
 %a
-pcre
-%a
 
 [Zend Modules]
 %aDone
--- php-7.0.0beta3/main/php_ini.c~	2015-08-26 18:50:15.000000000 +0300
+++ php-7.0.0beta3/main/php_ini.c	2015-08-26 22:33:59.126957396 +0300
@@ -765,8 +765,8 @@
  */
 void php_ini_register_extensions(void)
 {
-	zend_llist_apply(&extension_lists.engine, php_load_zend_extension_cb);
 	zend_llist_apply(&extension_lists.functions, php_load_php_extension_cb);
+	zend_llist_apply(&extension_lists.engine, php_load_zend_extension_cb);
 
 	zend_llist_destroy(&extension_lists.engine);
 	zend_llist_destroy(&extension_lists.functions);
--- php-5.3.1/ext/phar/Makefile.frag~	2009-07-23 15:48:04.000000000 +0000
+++ php-5.3.1/ext/phar/Makefile.frag	2009-11-30 16:10:29.687175948 +0000
@@ -10,7 +10,7 @@
 		$(top_srcdir)/build/shtool echo -n -- "$(top_builddir)/$(SAPI_CLI_PATH) -n"; \
 		if test "x$(PHP_MODULES)" != "x"; then \
 		$(top_srcdir)/build/shtool echo -n -- " -d extension_dir=$(top_builddir)/modules"; \
-		for i in spl bz2 zlib phar; do \
+		for i in pcre spl bz2 zlib phar; do \
 			if test -f "$(top_builddir)/modules/$$i.la"; then \
 				. $(top_builddir)/modules/$$i.la; $(top_srcdir)/build/shtool echo -n -- " -d extension=$$dlname"; \
 			fi; \
