diff -urNp -x '*.orig' php-5.6.40.org/ext/phar/Makefile.frag php-5.6.40/ext/phar/Makefile.frag
--- php-5.6.40.org/ext/phar/Makefile.frag	2021-08-23 23:20:33.969055218 +0200
+++ php-5.6.40/ext/phar/Makefile.frag	2021-08-23 23:20:34.982387483 +0200
@@ -9,7 +9,7 @@ PHP_PHARCMD_EXECUTABLE = ` \
 		$(top_srcdir)/build/shtool echo -n -- "$(top_builddir)/$(SAPI_CLI_PATH) -n"; \
 		if test "x$(PHP_MODULES)" != "x"; then \
 		$(top_srcdir)/build/shtool echo -n -- " -d extension_dir=$(top_builddir)/modules"; \
-		for i in pcre spl bz2 zlib phar; do \
+		for i in pcre spl bz2 zlib hash phar; do \
 			if test -f "$(top_builddir)/modules/$$i.la"; then \
 				. $(top_builddir)/modules/$$i.la; $(top_srcdir)/build/shtool echo -n -- " -d extension=$$dlname"; \
 			fi; \
diff -urNp -x '*.orig' php-5.6.40.org/ext/phar/config.m4 php-5.6.40/ext/phar/config.m4
--- php-5.6.40.org/ext/phar/config.m4	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/phar/config.m4	2021-08-23 23:20:34.982387483 +0200
@@ -7,13 +7,13 @@ PHP_ARG_ENABLE(phar, for phar archive su
 if test "$PHP_PHAR" != "no"; then
   PHP_NEW_EXTENSION(phar, util.c tar.c zip.c stream.c func_interceptors.c dirstream.c phar.c phar_object.c phar_path_check.c, $ext_shared)
   AC_MSG_CHECKING([for phar openssl support])
-  if test "$PHP_HASH_SHARED" != "yes"; then
-    if test "$PHP_HASH" != "no"; then
+dnl  if test "$PHP_HASH_SHARED" != "yes"; then
+dnl    if test "$PHP_HASH" != "no"; then
       AC_DEFINE(PHAR_HASH_OK,1,[ ])
-    fi
-  else
-    AC_MSG_WARN([Phar: sha256/sha512 signature support disabled if ext/hash is built shared])
-  fi
+dnl    fi
+dnl  else
+dnl    AC_MSG_WARN([Phar: sha256/sha512 signature support disabled if ext/hash is built shared])
+dnl  fi
   if test "$PHP_OPENSSL_SHARED" = "yes"; then
     AC_MSG_RESULT([no (shared openssl)])
   else
