diff -urNp -x '*.orig' php-7.2.34.org/scripts/Makefile.frag php-7.2.34/scripts/Makefile.frag
--- php-7.2.34.org/scripts/Makefile.frag	2020-09-30 07:15:50.000000000 +0200
+++ php-7.2.34/scripts/Makefile.frag	2021-08-24 01:07:20.716600751 +0200
@@ -3,7 +3,7 @@
 #
 
 phpincludedir = $(includedir)/php
-phpbuilddir = $(libdir)/build
+phpbuilddir = $(libdir)/php/build
 
 BUILD_FILES = \
 	scripts/phpize.m4 \
diff -urNp -x '*.orig' php-7.2.34.org/scripts/phpize.in php-7.2.34/scripts/phpize.in
--- php-7.2.34.org/scripts/phpize.in	2020-09-30 07:15:50.000000000 +0200
+++ php-7.2.34/scripts/phpize.in	2021-08-24 01:07:20.716600751 +0200
@@ -4,7 +4,7 @@
 prefix='@prefix@'
 datarootdir='@datarootdir@'
 exec_prefix="`eval echo @exec_prefix@`"
-phpdir="`eval echo @libdir@`/build"
+phpdir="`eval echo @libdir@`/php/build"
 includedir="`eval echo @includedir@`/php"
 builddir="`pwd`"
 SED="@SED@"
