--- PHP_5_3/ext/filter/sanitizing_filters.c	2011/04/03 12:25:43	309919
+++ PHP_5_3/ext/filter/sanitizing_filters.c	2011/04/03 16:30:31	309920
@@ -205,7 +205,11 @@
 
 	if (new_len == 0) {
 		zval_dtor(value);
-		ZVAL_EMPTY_STRING(value);
+		if (flags & FILTER_FLAG_EMPTY_STRING_NULL) {
+			ZVAL_NULL(value);
+		} else {
+			ZVAL_EMPTY_STRING(value);			
+		}
 		return;
 	}
 }
@@ -280,6 +284,9 @@
 		}
 
 		php_filter_encode_html(value, enc);	
+	} else if (flags & FILTER_FLAG_EMPTY_STRING_NULL && Z_STRLEN_P(value) == 0) {
+		zval_dtor(value);
+		ZVAL_NULL(value);
 	}
 }
 /* }}} */
