diff -u -r1.87 zend_API.h
--- php-4.0.6.orig/Zend/zend_API.h	2001/04/19 17:51:23
+++ php-4.0.6/Zend/zend_API.h	2001/08/24 13:18:51
@@ -268,8 +268,8 @@
 		(z)->type = IS_STRING;		    \
 	}
 
-#define ZVAL_FALSE  { (z)->value.lval = 0;  (z)->type = IS_BOOL; }
-#define ZVAL_TRUE   { (z)->value.lval = 1;  (z)->type = IS_BOOL; }
+#define ZVAL_FALSE(z)  { (z)->value.lval = 0;  (z)->type = IS_BOOL; }
+#define ZVAL_TRUE(z)   { (z)->value.lval = 1;  (z)->type = IS_BOOL; }
 
 #define RETVAL_RESOURCE(l) {			\
 		return_value->type = IS_RESOURCE;\
