--- php-4.3.2/ext/cpdf/config.m4.orig	Wed Apr 30 10:23:06 2003
+++ php-4.3.2/ext/cpdf/config.m4	Fri Jun 13 12:23:26 2003
@@ -24,9 +24,9 @@
     ],) 
     AC_CHECK_LIB(jpeg,jpeg_read_header, ,[AC_MSG_ERROR(Problem with libjpeg.(a|so). Please check config.log for more information.)],)
     if test -z "$CPDF_JPEG_DIR"; then
-      PHP_ADD_LIBRARY(jpeg)
+      PHP_ADD_LIBRARY(jpeg, , CPDF_SHARED_LIBADD)
     else
-      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $CPDF_JPEG_DIR/lib)
+      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $CPDF_JPEG_DIR/lib, CPDF_SHARED_LIBADD)
     fi
   ])
 ])
@@ -49,22 +49,23 @@
     ],) 
     AC_CHECK_LIB(tiff,TIFFOpen, ,[AC_MSG_ERROR(Problem with libtiff.(a|so). Please check config.log for more information.)],)
     if test -z "$CPDF_TIFF_DIR"; then
-      PHP_ADD_LIBRARY(tiff)
+      PHP_ADD_LIBRARY(tiff, , CPDF_SHARED_LIBADD)
     else
-      PHP_ADD_LIBRARY_WITH_PATH(tiff, $CPDF_TIFF_DIR/lib)
+      PHP_ADD_LIBRARY_WITH_PATH(tiff, $CPDF_TIFF_DIR/lib, CPDF_SHARED_LIBADD)
     fi
   ])
 ])
 
-AC_MSG_CHECKING(for cpdflib support)
 AC_ARG_WITH(cpdflib,
 [  --with-cpdflib[=DIR]    Include cpdflib support (requires cpdflib >= 2).],
 [
+  AC_MSG_CHECKING(for cpdflib support)
   if test "$withval" = "no"; then
     AC_MSG_RESULT(no)
   else
     AC_MSG_RESULT(yes)
-    PHP_NEW_EXTENSION(cpdf, cpdf.c)
+    PHP_NEW_EXTENSION(cpdf, cpdf.c, $ext_shared)
+    PHP_SUBST(CPDF_SHARED_LIBADD)
     CPDF_JPEG_TEST
     CPDF_TIFF_TEST
     for i in $withval /usr /usr/local; do
@@ -78,10 +79,10 @@
             [AC_MSG_ERROR([Cpdflib module requires cpdflib >= 2.])])
         ])
         if test "$i" != "/usr" -a "$i" != "/usr/local"; then
-          PHP_ADD_LIBRARY_WITH_PATH(cpdf, $i/lib)
+          PHP_ADD_LIBRARY_WITH_PATH(cpdf, $i/lib, CPDF_SHARED_LIBADD)
           PHP_ADD_INCLUDE($CPDFLIB_INCLUDE)
         else
-          PHP_ADD_LIBRARY(cpdf)
+          PHP_ADD_LIBRARY(cpdf, , CPDF_SHARED_LIBADD)
         fi
         break
       fi
