diff -urNp -x '*.orig' php-7.0.33.org/scripts/Makefile.frag php-7.0.33/scripts/Makefile.frag
--- php-7.0.33.org/scripts/Makefile.frag	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/scripts/Makefile.frag	2021-08-23 23:36:04.841475774 +0200
@@ -4,7 +4,7 @@
 #
 
 phpincludedir = $(includedir)/php
-phpbuilddir = $(libdir)/build
+phpbuilddir = $(libdir)/php/build
 
 BUILD_FILES = \
 	scripts/phpize.m4 \
diff -urNp -x '*.orig' php-7.0.33.org/scripts/phpize.in php-7.0.33/scripts/phpize.in
--- php-7.0.33.org/scripts/phpize.in	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/scripts/phpize.in	2021-08-23 23:36:04.841475774 +0200
@@ -4,7 +4,7 @@
 prefix='@prefix@'
 datarootdir='@datarootdir@'
 exec_prefix="`eval echo @exec_prefix@`"
-phpdir="`eval echo @libdir@`/build"
+phpdir="`eval echo @libdir@`/php/build"
 includedir="`eval echo @includedir@`/php"
 builddir="`pwd`"
 SED="@SED@"
