diff -burN php-5.0.3RC2.orig/acinclude.m4 php-5.0.3RC2/acinclude.m4
--- php-5.0.3RC2.orig/acinclude.m4	2004-12-13 11:25:23.854667632 +0100
+++ php-5.0.3RC2/acinclude.m4	2004-12-13 11:26:08.477883872 +0100
@@ -1625,6 +1625,9 @@
       if test -r $i/lib/libssl.a -o -r $i/lib/libssl.$SHLIB_SUFFIX_NAME; then
         OPENSSL_LIBDIR=$i/lib
       fi
+      if test -r $i/lib64/libssl.a -o -r $i/lib64/libssl.$SHLIB_SUFFIX_NAME; then
+        OPENSSL_LIBDIR=$i/lib64
+      fi
       test -n "$OPENSSL_INCDIR" && test -n "$OPENSSL_LIBDIR" && break
     done
 
diff -burN php-5.0.3RC2.orig/ext/cpdf/config.m4 php-5.0.3RC2/ext/cpdf/config.m4
--- php-5.0.3RC2.orig/ext/cpdf/config.m4	2004-12-13 11:25:21.724991392 +0100
+++ php-5.0.3RC2/ext/cpdf/config.m4	2004-12-13 11:26:08.485882656 +0100
@@ -4,7 +4,7 @@
   AC_ARG_WITH(jpeg-dir,
   [  --with-jpeg-dir[=DIR]     CPDF: Set the path to libjpeg install prefix.],[
     for i in $withval /usr/local /usr; do
-      if test -f "$i/lib/libjpeg.$SHLIB_SUFFIX_NAME" -o -f "$i/lib/libjpeg.a"; then
+      if test -f "$i/lib64/libjpeg.$SHLIB_SUFFIX_NAME" -o -f "$i/lib64/libjpeg.a"; then
         CPDF_JPEG_DIR=$i
         break;
       fi
@@ -13,11 +13,11 @@
       AC_MSG_ERROR([libjpeg.(a|so) not found.])
     fi
     PHP_CHECK_LIBRARY(jpeg, jpeg_read_header, [
-      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $CPDF_JPEG_DIR/lib, CPDF_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $CPDF_JPEG_DIR/lib64, CPDF_SHARED_LIBADD)
     ] ,[
       AC_MSG_ERROR([CPDF: Problem with libjpeg.(a|so). Please check config.log for more information.])
     ], [
-      -L$CPDF_JPEG_DIR/lib
+      -L$CPDF_JPEG_DIR/lib64
     ])
   ],)
 ])
@@ -26,7 +26,7 @@
   AC_ARG_WITH(tiff-dir,
   [  --with-tiff-dir[=DIR]     CPDF: Set the path to libtiff install prefix.],[
     for i in $withval /usr/local /usr; do
-      if test -f "$i/lib/libtiff.$SHLIB_SUFFIX_NAME" -o -f "$i/lib/libtiff.a"; then
+      if test -f "$i/lib64/libtiff.$SHLIB_SUFFIX_NAME" -o -f "$i/lib64/libtiff.a"; then
         CPDF_TIFF_DIR=$i
         break;
       fi
@@ -35,11 +35,11 @@
       AC_MSG_ERROR([libtiff.(a|so) not found.])
     fi
     PHP_CHECK_LIBRARY(tiff, TIFFOpen, [
-      PHP_ADD_LIBRARY_WITH_PATH(tiff, $CPDF_TIFF_DIR/lib, CPDF_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(tiff, $CPDF_TIFF_DIR/lib64, CPDF_SHARED_LIBADD)
     ] ,[
       AC_MSG_ERROR([CPDF: Problem with libtiff.(a|so). Please check config.log for more information.])
     ], [
-      -L$CPDF_TIFF_DIR/lib
+      -L$CPDF_TIFF_DIR/lib64
     ])
   ],)
 ])
@@ -59,12 +59,12 @@
 
       PHP_CHECK_LIBRARY(cpdf, cpdf_open, [
         PHP_ADD_INCLUDE($CPDFLIB_INCLUDE)
-        PHP_ADD_LIBRARY_WITH_PATH(cpdf, $i/lib, CPDF_SHARED_LIBADD)
+        PHP_ADD_LIBRARY_WITH_PATH(cpdf, $i/lib64, CPDF_SHARED_LIBADD)
         AC_DEFINE(HAVE_CPDFLIB,1,[Whether you have cpdflib])
       ], [
         AC_MSG_ERROR([Cpdflib module requires cpdflib >= 2.])
       ], [
-        -L$i/lib $CPDF_SHARED_LIBADD
+        -L$i/lib64 $CPDF_SHARED_LIBADD
       ])
       break
     fi
diff -burN php-5.0.3RC2.orig/ext/dba/config.m4 php-5.0.3RC2/ext/dba/config.m4
--- php-5.0.3RC2.orig/ext/dba/config.m4	2004-12-13 11:25:21.641004160 +0100
+++ php-5.0.3RC2/ext/dba/config.m4	2004-12-13 11:26:08.493881440 +0100
@@ -21,7 +21,7 @@
 dnl Assign INCLUDE/LFLAGS from PREFIX
 AC_DEFUN(PHP_DBA_STD_ASSIGN,[
   if test -n "$THIS_PREFIX" && test "$THIS_PREFIX" != "/usr"; then
-    THIS_LFLAGS=$THIS_PREFIX/lib
+    THIS_LFLAGS=$THIS_PREFIX/lib64
   fi
 ])
 
@@ -121,7 +121,7 @@
         AC_DEFINE_UNQUOTED(GDBM_INCLUDE_FILE, "$THIS_INCLUDE", [ ])
         AC_DEFINE(DBA_GDBM, 1, [ ]) 
         THIS_LIBS=gdbm
-      ], [], [-L$THIS_PREFIX/lib])
+      ], [], [-L$THIS_PREFIX/lib64])
     fi
     
     PHP_DBA_STD_ASSIGN
@@ -153,7 +153,7 @@
           AC_DEFINE_UNQUOTED(NDBM_INCLUDE_FILE, "$THIS_INCLUDE", [ ])
           AC_DEFINE(DBA_NDBM, 1, [ ]) 
           THIS_LIBS=$LIB
-        ], [], [-L$THIS_PREFIX/lib])
+        ], [], [-L$THIS_PREFIX/lib64])
         if test -n "$THIS_LIBS"; then
           break
         fi
@@ -171,8 +171,8 @@
 dnl parameters(version, library list, function)
 AC_DEFUN(PHP_DBA_DB_CHECK,[
   for LIB in $2; do
-    if test -f $THIS_PREFIX/lib/lib$LIB.a -o -f $THIS_PREFIX/lib/lib$LIB.$SHLIB_SUFFIX_NAME; then
-      PHP_TEMP_LDFLAGS(-L$THIS_PREFIX/lib, -l$LIB,[
+    if test -f $THIS_PREFIX/lib64/lib$LIB.a -o -f $THIS_PREFIX/lib64/lib$LIB.$SHLIB_SUFFIX_NAME; then
+      PHP_TEMP_LDFLAGS(-L$THIS_PREFIX/lib64, -l$LIB,[
         AC_TRY_LINK([
 #include "$THIS_INCLUDE"
         ],[
@@ -367,7 +367,7 @@
           fi
           AC_DEFINE(DBA_DBM, 1, [ ]) 
           THIS_LIBS=$LIB
-        ], [], [-L$THIS_PREFIX/lib])
+        ], [], [-L$THIS_PREFIX/lib64])
         if test -n "$THIS_LIBS"; then
           break
         fi
@@ -409,7 +409,7 @@
           AC_DEFINE_UNQUOTED(CDB_INCLUDE_FILE, "$THIS_INCLUDE", [ ])
           AC_DEFINE(DBA_CDB, 1, [ ]) 
           THIS_LIBS=$LIB
-        ], [], [-L$THIS_PREFIX/lib])
+        ], [], [-L$THIS_PREFIX/lib64])
         if test -n "$THIS_LIBS"; then
           break
         fi
diff -burN php-5.0.3RC2.orig/ext/gd/config.m4 php-5.0.3RC2/ext/gd/config.m4
--- php-5.0.3RC2.orig/ext/gd/config.m4	2004-12-13 11:25:21.629005984 +0100
+++ php-5.0.3RC2/ext/gd/config.m4	2004-12-13 11:26:08.497880832 +0100
@@ -51,7 +51,7 @@
   if test "$PHP_JPEG_DIR" != "no"; then
 
     for i in $PHP_JPEG_DIR /usr/local /usr; do
-      test -f $i/lib/libjpeg.$SHLIB_SUFFIX_NAME -o -f $i/lib/libjpeg.a && GD_JPEG_DIR=$i && break
+      test -f $i/lib64/libjpeg.$SHLIB_SUFFIX_NAME -o -f $i/lib64/libjpeg.a && GD_JPEG_DIR=$i && break
     done
 
     if test -z "$GD_JPEG_DIR"; then
@@ -61,11 +61,11 @@
     PHP_CHECK_LIBRARY(jpeg,jpeg_read_header,
     [
       PHP_ADD_INCLUDE($GD_JPEG_DIR/include)
-      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $GD_JPEG_DIR/lib, GD_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $GD_JPEG_DIR/lib64, GD_SHARED_LIBADD)
     ],[
       AC_MSG_ERROR([Problem with libjpeg.(a|so). Please check config.log for more information.]) 
     ],[
-      -L$GD_JPEG_DIR/lib
+      -L$GD_JPEG_DIR/lib64
     ])
   else 
     AC_MSG_RESULT([If configure fails try --with-jpeg-dir=<DIR>])
@@ -76,7 +76,7 @@
   if test "$PHP_PNG_DIR" != "no"; then
 
     for i in $PHP_PNG_DIR /usr/local /usr; do
-      test -f $i/lib/libpng.$SHLIB_SUFFIX_NAME -o -f $i/lib/libpng.a && GD_PNG_DIR=$i && break
+      test -f $i/lib64/libpng.$SHLIB_SUFFIX_NAME -o -f $i/lib64/libpng.a && GD_PNG_DIR=$i && break
     done
 
     if test -z "$GD_PNG_DIR"; then
@@ -94,12 +94,12 @@
     PHP_CHECK_LIBRARY(png,png_write_image,
     [
       PHP_ADD_INCLUDE($GD_PNG_DIR/include)
-      PHP_ADD_LIBRARY_WITH_PATH(z, $PHP_ZLIB_DIR/lib, GD_SHARED_LIBADD)
-      PHP_ADD_LIBRARY_WITH_PATH(png, $GD_PNG_DIR/lib, GD_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(z, $PHP_ZLIB_DIR/lib64, GD_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(png, $GD_PNG_DIR/lib64, GD_SHARED_LIBADD)
     ],[
       AC_MSG_ERROR([Problem with libpng.(a|so) or libz.(a|so). Please check config.log for more information.]) 
     ],[
-      -L$PHP_ZLIB_DIR/lib -lz -L$GD_PNG_DIR/lib
+      -L$PHP_ZLIB_DIR/lib64 -lz -L$GD_PNG_DIR/lib64
     ])
 
   else 
@@ -111,7 +111,7 @@
   if test "$PHP_XPM_DIR" != "no"; then
 
     for i in $PHP_XPM_DIR /usr/local /usr/X11R6 /usr; do
-      test -f $i/lib/libXpm.$SHLIB_SUFFIX_NAME -o -f $i/lib/libXpm.a && GD_XPM_DIR=$i && break
+      test -f $i/lib64/libXpm.$SHLIB_SUFFIX_NAME -o -f $i/lib64/libXpm.a && GD_XPM_DIR=$i && break
     done
 
     if test -z "$GD_XPM_DIR"; then
@@ -129,12 +129,12 @@
     PHP_CHECK_LIBRARY(Xpm,XpmFreeXpmImage, 
     [
       PHP_ADD_INCLUDE($GD_XPM_INC)
-      PHP_ADD_LIBRARY_WITH_PATH(Xpm, $GD_XPM_DIR/lib, GD_SHARED_LIBADD)
-      PHP_ADD_LIBRARY_WITH_PATH(X11, $GD_XPM_DIR/lib, GD_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(Xpm, $GD_XPM_DIR/lib64, GD_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(X11, $GD_XPM_DIR/lib64, GD_SHARED_LIBADD)
     ],[
       AC_MSG_ERROR([Problem with libXpm.(a|so) or libX11.(a|so). Please check config.log for more information.]) 
     ],[
-      -L$GD_XPM_DIR/lib -lX11
+      -L$GD_XPM_DIR/lib64 -lX11
     ])
   else 
     AC_MSG_RESULT(If configure fails try --with-xpm-dir=<DIR>)
@@ -163,7 +163,7 @@
       fi
       if test -n "$TTF_DIR" ; then
         AC_DEFINE(HAVE_LIBTTF,1,[ ])
-        PHP_ADD_LIBRARY_WITH_PATH(ttf, $TTF_DIR/lib, GD_SHARED_LIBADD)
+        PHP_ADD_LIBRARY_WITH_PATH(ttf, $TTF_DIR/lib64, GD_SHARED_LIBADD)
       fi
       if test -z "$TTF_INC_DIR"; then
         TTF_INC_DIR=$TTF_DIR/include
@@ -188,7 +188,7 @@
     done
     
     if test -n "$FREETYPE2_DIR" ; then
-      PHP_ADD_LIBRARY_WITH_PATH(freetype, $FREETYPE2_DIR/lib, GD_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(freetype, $FREETYPE2_DIR/lib64, GD_SHARED_LIBADD)
       PHP_ADD_INCLUDE($FREETYPE2_DIR/include)
       PHP_ADD_INCLUDE($FREETYPE2_INC_DIR)
       AC_DEFINE(USE_GD_IMGSTRTTF, 1, [ ])
@@ -216,11 +216,11 @@
     [
       AC_DEFINE(HAVE_LIBT1,1,[ ])
       PHP_ADD_INCLUDE($GD_T1_DIR/include)
-      PHP_ADD_LIBRARY_WITH_PATH(t1, $GD_T1_DIR/lib, GD_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(t1, $GD_T1_DIR/lib64, GD_SHARED_LIBADD)
     ],[
       AC_MSG_ERROR([Problem with libt1.(a|so). Please check config.log for more information.]) 
     ],[
-      -L$GD_T1_DIR/lib
+      -L$GD_T1_DIR/lib64
     ])
   fi
 ])
@@ -363,7 +363,7 @@
   done
 
 dnl Library path
-  for i in lib/gd1.3 lib/gd lib gd1.3 gd ""; do
+  for i in lib64/gd1.3 lib64/gd lib64 gd1.3 gd ""; do
     test -f "$PHP_GD/$i/libgd.$SHLIB_SUFFIX_NAME" -o -f "$PHP_GD/$i/libgd.a" && GD_LIB="$PHP_GD/$i"
   done
 
diff -burN php-5.0.3RC2.orig/ext/imap/config.m4 php-5.0.3RC2/ext/imap/config.m4
--- php-5.0.3RC2.orig/ext/imap/config.m4	2004-12-13 11:25:21.735989720 +0100
+++ php-5.0.3RC2/ext/imap/config.m4	2004-12-13 11:26:08.503879920 +0100
@@ -59,7 +59,7 @@
     fi
 
     for i in $SEARCH_PATHS; do
-      if test -f $i/lib/libkrb5.a || test -f $i/lib/libkrb5.$SHLIB_SUFFIX_NAME; then
+      if test -f $i/lib64/libkrb5.a || test -f $i/lib64/libkrb5.$SHLIB_SUFFIX_NAME; then
         PHP_KERBEROS_DIR=$i
         break
       fi
@@ -72,7 +72,7 @@
       ])
     fi
     AC_DEFINE(HAVE_IMAP_KRB,1,[ ])
-    PHP_ADD_LIBPATH($PHP_KERBEROS_DIR/lib, IMAP_SHARED_LIBADD)
+    PHP_ADD_LIBPATH($PHP_KERBEROS_DIR/lib64, IMAP_SHARED_LIBADD)
     PHP_ADD_LIBRARY(gssapi_krb5, 1, IMAP_SHARED_LIBADD)
     PHP_ADD_LIBRARY(krb5, 1, IMAP_SHARED_LIBADD)
     PHP_ADD_LIBRARY(k5crypto, 1, IMAP_SHARED_LIBADD)
@@ -104,11 +104,11 @@
   AC_MSG_CHECKING([whether SSL libraries are needed for c-client])
 
   if test "$PHP_IMAP_SSL" != "no"; then
-    AC_MSG_RESULT([$PHP_IMAP_SSL/lib])
+    AC_MSG_RESULT([$PHP_IMAP_SSL/lib64])
     AC_DEFINE(HAVE_IMAP_SSL,1,[ ])
     PHP_ADD_LIBRARY_DEFER(ssl,    1, IMAP_SHARED_LIBADD)
     PHP_ADD_LIBRARY_DEFER(crypto, 1, IMAP_SHARED_LIBADD)
-    PHP_ADD_LIBPATH($PHP_IMAP_SSL/lib, IMAP_SHARED_LIBADD)
+    PHP_ADD_LIBPATH($PHP_IMAP_SSL/lib64, IMAP_SHARED_LIBADD)
   else
     AC_MSG_RESULT(no)
   fi
@@ -166,13 +166,13 @@
 
     if test -r "$IMAP_DIR/c-client/c-client.a"; then
       ln -s "$IMAP_DIR/c-client/c-client.a" "$IMAP_DIR/c-client/libc-client.a" >/dev/null 2>&1
-    elif test -r "$IMAP_DIR/lib/c-client.a"; then
-      ln -s "$IMAP_DIR/lib/c-client.a" "$IMAP_DIR/lib/libc-client.a" >/dev/null 2>&1
+    elif test -r "$IMAP_DIR/lib64/c-client.a"; then
+      ln -s "$IMAP_DIR/lib64/c-client.a" "$IMAP_DIR/lib64/libc-client.a" >/dev/null 2>&1
     fi
 
     for lib in c-client4 c-client imap; do
       IMAP_LIB=$lib
-      IMAP_LIB_CHK(lib)
+      IMAP_LIB_CHK(lib64)
       IMAP_LIB_CHK(c-client)
     done
 
diff -burN php-5.0.3RC2.orig/ext/ldap/config.m4 php-5.0.3RC2/ext/ldap/config.m4
--- php-5.0.3RC2.orig/ext/ldap/config.m4	2004-12-13 11:25:21.741988808 +0100
+++ php-5.0.3RC2/ext/ldap/config.m4	2004-12-13 11:26:08.506879464 +0100
@@ -6,15 +6,15 @@
   if test -f $1/include/ldap.h; then
     LDAP_DIR=$1
     LDAP_INCDIR=$1/include
-    LDAP_LIBDIR=$1/lib
+    LDAP_LIBDIR=$1/lib64
   elif test -f $1/include/umich-ldap/ldap.h; then
     LDAP_DIR=$1
     LDAP_INCDIR=$1/include/umich-ldap
-    LDAP_LIBDIR=$1/lib
+    LDAP_LIBDIR=$1/lib64
   elif test -f $1/ldap/public/ldap.h; then
     LDAP_DIR=$1
     LDAP_INCDIR=$1/ldap/public
-    LDAP_LIBDIR=$1/lib
+    LDAP_LIBDIR=$1/lib64
   fi
 ])
 
diff -burN php-5.0.3RC2.orig/ext/ming/config.m4 php-5.0.3RC2/ext/ming/config.m4
--- php-5.0.3RC2.orig/ext/ming/config.m4	2004-12-13 11:25:21.747987896 +0100
+++ php-5.0.3RC2/ext/ming/config.m4	2004-12-13 11:26:08.519877488 +0100
@@ -9,7 +9,7 @@
   AC_CHECK_LIB(m, sin)
 
   for i in $PHP_MING /usr/local /usr; do
-    if test -f $i/lib/libming.$SHLIB_SUFFIX_NAME -o -f $i/lib/libming.a; then
+    if test -f $i/lib64/libming.$SHLIB_SUFFIX_NAME -o -f $i/lib64/libming.a; then
       MING_DIR=$i
     fi
   done
@@ -33,11 +33,11 @@
   ],[
     AC_MSG_ERROR([Ming library 0.2a or greater required.])
   ],[
-    -L$MING_DIR/lib
+    -L$MING_DIR/lib64
   ])
   
   PHP_ADD_INCLUDE($MING_INC_DIR)
-  PHP_ADD_LIBRARY_WITH_PATH(ming, $MING_DIR/lib, MING_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(ming, $MING_DIR/lib64, MING_SHARED_LIBADD)
 
   AC_MSG_CHECKING([for destroySWFBlock])
   AC_TRY_RUN([
diff -burN php-5.0.3RC2.orig/ext/mssql/config.m4 php-5.0.3RC2/ext/mssql/config.m4
--- php-5.0.3RC2.orig/ext/mssql/config.m4	2004-12-13 11:25:21.898964944 +0100
+++ php-5.0.3RC2/ext/mssql/config.m4	2004-12-13 11:26:58.696249520 +0100
@@ -32,12 +32,12 @@
     fi
   fi  
 
-  if test ! -r "$FREETDS_INSTALLATION_DIR/lib/libtds.a" && test ! -r "$FREETDS_INSTALLATION_DIR/lib/libtds.so"; then
-     AC_MSG_ERROR(Could not find $FREETDS_INSTALLATION_DIR/lib/libtds.[a|so])
+  if test ! -r "$FREETDS_INSTALLATION_DIR/lib64/libtds.a" && test ! -r "$FREETDS_INSTALLATION_DIR/lib64/libtds.so"; then
+     AC_MSG_ERROR(Could not find $FREETDS_INSTALLATION_DIR/lib64/libtds.[a|so])
   fi
 
   MSSQL_INCDIR=$FREETDS_INSTALLATION_DIR/include
-  MSSQL_LIBDIR=$FREETDS_INSTALLATION_DIR/lib
+  MSSQL_LIBDIR=$FREETDS_INSTALLATION_DIR/lib64
 
   PHP_ADD_INCLUDE($MSSQL_INCDIR)
   PHP_ADD_LIBRARY_WITH_PATH(sybdb, $MSSQL_LIBDIR, MSSQL_SHARED_LIBADD)
diff -burN php-5.0.3RC2.orig/ext/mysql/config.m4 php-5.0.3RC2/ext/mysql/config.m4
--- php-5.0.3RC2.orig/ext/mysql/config.m4	2004-12-13 11:25:21.901964488 +0100
+++ php-5.0.3RC2/ext/mysql/config.m4	2004-12-13 11:26:08.524876728 +0100
@@ -17,7 +17,7 @@
     /var/run/mysqld/mysqld.sock \
     /var/tmp/mysql.sock \
     /var/run/mysql/mysql.sock \
-    /var/lib/mysql/mysql.sock \
+    /var/lib64/mysql/mysql.sock \
     /var/mysql/mysql.sock \
     /usr/local/mysql/var/mysql.sock \
     /Private/tmp/mysql.sock \
@@ -84,7 +84,7 @@
 Note that the MySQL client library is not bundled anymore.])
   fi
 
-  for i in lib lib/mysql; do
+  for i in lib64 lib64/mysql; do
     MYSQL_LIB_CHK($i)
   done
 
@@ -100,9 +100,9 @@
       PHP_CHECK_LIBRARY(mysqlclient, mysql_error, [], [
         AC_MSG_ERROR([mysql configure failed. Please check config.log for more information.])
       ], [
-        -L$PHP_ZLIB_DIR/lib -L$MYSQL_LIB_DIR 
+        -L$PHP_ZLIB_DIR/lib64 -L$MYSQL_LIB_DIR 
       ])  
-      MYSQL_LIBS="-L$PHP_ZLIB_DIR/lib -lz"
+      MYSQL_LIBS="-L$PHP_ZLIB_DIR/lib64 -lz"
     else
       PHP_ADD_LIBRARY(z,, MYSQL_SHARED_LIBADD)
       PHP_CHECK_LIBRARY(mysqlclient, mysql_errno, [], [
diff -burN php-5.0.3RC2.orig/ext/pgsql/config.m4 php-5.0.3RC2/ext/pgsql/config.m4
--- php-5.0.3RC2.orig/ext/pgsql/config.m4	2004-12-13 11:25:21.913962664 +0100
+++ php-5.0.3RC2/ext/pgsql/config.m4	2004-12-13 11:26:08.526876424 +0100
@@ -44,7 +44,7 @@
         fi
       done
 
-      for j in lib lib/pgsql lib/postgres lib/postgresql ""; do
+      for j in lib64 lib64/pgsql lib64/postgres lib64/postgresql ""; do
         if test -f "$i/$j/libpq.so" || test -f "$i/$j/libpq.a"; then 
           PGSQL_LIBDIR=$i/$j
         fi
diff -burN php-5.0.3RC2.orig/ext/xml/config.m4 php-5.0.3RC2/ext/xml/config.m4
--- php-5.0.3RC2.orig/ext/xml/config.m4	2004-12-13 11:25:21.716992608 +0100
+++ php-5.0.3RC2/ext/xml/config.m4	2004-12-13 11:26:08.528876120 +0100
@@ -30,7 +30,7 @@
   dnl
   if test "$PHP_LIBEXPAT_DIR" != "no"; then
     for i in $PHP_XML $PHP_LIBEXPAT_DIR; do
-      if test -f "$i/lib/libexpat.a" -o -f "$i/lib/libexpat.$SHLIB_SUFFIX_NAME"; then
+      if test -f "$i/lib64/libexpat.a" -o -f "$i/lib64/libexpat.$SHLIB_SUFFIX_NAME"; then
         EXPAT_DIR=$i
         break
       fi
@@ -41,7 +41,7 @@
     fi
 
     PHP_ADD_INCLUDE($EXPAT_DIR/include)
-    PHP_ADD_LIBRARY_WITH_PATH(expat, $EXPAT_DIR/lib, XML_SHARED_LIBADD)
+    PHP_ADD_LIBRARY_WITH_PATH(expat, $EXPAT_DIR/lib64, XML_SHARED_LIBADD)
     AC_DEFINE(HAVE_LIBEXPAT, 1, [ ])
   fi
 
diff -burN php-5.0.3RC2.orig/ext/xmlrpc/config.m4 php-5.0.3RC2/ext/xmlrpc/config.m4
--- php-5.0.3RC2.orig/ext/xmlrpc/config.m4	2004-12-13 11:25:22.261909768 +0100
+++ php-5.0.3RC2/ext/xmlrpc/config.m4	2004-12-13 11:26:08.532875512 +0100
@@ -23,9 +23,9 @@
 
   testval=no
   for i in $PHP_EXPAT_DIR $XMLRPC_DIR /usr/local /usr; do
-    if test -f $i/lib/libexpat.a -o -f $i/lib/libexpat.$SHLIB_SUFFIX_NAME; then
+    if test -f $i/lib64/libexpat.a -o -f $i/lib64/libexpat.$SHLIB_SUFFIX_NAME; then
       AC_DEFINE(HAVE_LIBEXPAT2,1,[ ])
-      PHP_ADD_LIBRARY_WITH_PATH(expat, $i/lib, XMLRPC_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(expat, $i/lib64, XMLRPC_SHARED_LIBADD)
       PHP_ADD_INCLUDE($i/include)
       testval=yes
       break
@@ -90,6 +90,6 @@
   PHP_NEW_EXTENSION(xmlrpc,xmlrpc-epi-php.c,$ext_shared)
   XMLRPC_MODULE_TYPE=external
   PHP_ADD_INCLUDE($XMLRPC_DIR)
-  PHP_ADD_LIBRARY_WITH_PATH(xmlrpc, $XMLRPC_DIR/lib, XMLRPC_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(xmlrpc, $XMLRPC_DIR/lib64, XMLRPC_SHARED_LIBADD)
 fi
 
diff -burN php-5.0.3RC2.orig/php.ini php-5.0.3RC2/php.ini
--- php-5.0.3RC2.orig/php.ini	2004-12-13 11:25:23.870665200 +0100
+++ php-5.0.3RC2/php.ini	2004-12-13 11:26:08.538874600 +0100
@@ -514,7 +514,7 @@
 user_dir =
 
 ; Directory in which the loadable extensions (modules) reside.
-extension_dir = "/usr/lib/php"
+extension_dir = "/usr/lib64/php"
 
 ; Whether or not to enable the dl() function.  The dl() function does NOT work
 ; properly in multithreaded servers, such as IIS or Zeus, and is automatically
diff -burN php-5.0.3RC2.orig/scripts/Makefile.frag php-5.0.3RC2/scripts/Makefile.frag
--- php-5.0.3RC2.orig/scripts/Makefile.frag	2004-12-13 11:25:23.782678576 +0100
+++ php-5.0.3RC2/scripts/Makefile.frag	2004-12-13 11:26:08.544873688 +0100
@@ -4,7 +4,7 @@
 #
 
 phpincludedir = $(includedir)/php
-phpbuilddir = $(prefix)/lib/php/build
+phpbuilddir = $(libdir)/php/build
 
 BUILD_FILES = \
 	scripts/phpize.m4 \
