--- php-5.2.6/sapi/cgi/fpm/conf/php-fpm.conf.in	2008-11-16 17:30:12.321010353 +0200
+++ php-5.2.9/sapi/cgi/fpm/conf/php-fpm.conf.in	2009-06-12 10:24:54.843655130 +0300
@@ -38,7 +38,7 @@
 
 			Address to accept fastcgi requests on.
 			Valid syntax is 'ip.ad.re.ss:port' or just 'port' or '/path/to/unix/socket'
-			<value name="listen_address">127.0.0.1:9000</value>
+			<value name="listen_address">/var/run/php/fpm.sock</value>
 
 			<value name="listen_options">
 
@@ -60,10 +60,10 @@
 			</value>
 
 			Unix user of processes
-		<!--	<value name="user">nobody</value>				-->
+			<value name="user">http</value>
 
 			Unix group of processes
-		<!--	<value name="group">@php_fpm_group@</value>		-->
+			<value name="group">http</value>
 
 			Process manager settings
 			<value name="pm">
@@ -129,7 +129,7 @@
 			Useful to work around memory leaks in 3rd party libraries.
 			For endless request processing please specify 0
 			Equivalent to PHP_FCGI_MAX_REQUESTS
-			<value name="max_requests">500</value>
+			<value name="max_requests">100000</value>
 
 			Comma separated list of ipv4 addresses of FastCGI clients that allowed to connect.
 			Equivalent to FCGI_WEB_SERVER_ADDRS environment in original php.fcgi (5.2.2+)
@@ -139,14 +139,7 @@
 			Pass environment variables like LD_LIBRARY_PATH
 			All $VARIABLEs are taken from current environment
 			<value name="environment">
-				<value name="HOSTNAME">$HOSTNAME</value>
-				<value name="PATH">/usr/local/bin:/usr/bin:/bin</value>
-				<value name="TMP">/tmp</value>
-				<value name="TMPDIR">/tmp</value>
-				<value name="TEMP">/tmp</value>
-				<value name="OSTYPE">$OSTYPE</value>
-				<value name="MACHTYPE">$MACHTYPE</value>
-				<value name="MALLOC_CHECK_">2</value>
+				<value name="PATH">$PATH</value>
 			</value>
 
 		</section>
