diff -uNr php-5.0.4.orig/main/hardening_patch.c php-5.0.4/main/hardening_patch.c
--- php-5.0.4.orig/main/hardening_patch.c	2005-07-03 02:05:18.000000000 +0200
+++ php-5.0.4/main/hardening_patch.c	2005-07-03 02:06:58.000000000 +0200
@@ -126,9 +126,9 @@
 		i++;
 	}
 	
-	if (zend_is_executing(TSRMLS_CC)) {
-		lineno = zend_get_executed_lineno(TSRMLS_CC);
-		fname = zend_get_executed_filename(TSRMLS_CC);
+	if (zend_is_executing(TSRMLS_C)) {
+		lineno = zend_get_executed_lineno(TSRMLS_C);
+		fname = zend_get_executed_filename(TSRMLS_C);
 		ap_php_snprintf(buf, sizeof(buf), "ALERT - %s (attacker '%s', file '%s', line %u)", error, ip_address, fname, lineno);
 	} else {
 		fname = sapi_getenv("SCRIPT_FILENAME", 15 TSRMLS_CC);
diff -uNr php-5.0.4.orig/Zend/zend_execute_API.c php-5.0.4/Zend/zend_execute_API.c
--- php-5.0.4.orig/Zend/zend_execute_API.c	2005-07-03 02:05:18.000000000 +0200
+++ php-5.0.4/Zend/zend_execute_API.c	2005-07-03 02:06:32.000000000 +0200
@@ -1037,7 +1037,7 @@
 
 ZEND_API int zend_eval_string(char *str, zval *retval_ptr, char *string_name TSRMLS_DC)
 {
-	return (zend_eval_string_ex_ex(str, retval_ptr, string_name, ZEND_EVAL_CODE TSRMLS_C));
+	return (zend_eval_string_ex_ex(str, retval_ptr, string_name, ZEND_EVAL_CODE TSRMLS_CC));
 }
 
 
