--- php-4.3.0/ext/hyperwave/config.m4.orig	Tue Mar 12 17:47:14 2002
+++ php-4.3.0/ext/hyperwave/config.m4	Thu Jan  2 13:18:32 2003
@@ -9,7 +9,7 @@
   if test "$withval" != "no"; then
     AC_DEFINE(HYPERWAVE,1,[ ])
     AC_MSG_RESULT(yes)
-    PHP_NEW_EXTENSION(hyperwave, hw.c hg_comm.c)
+    PHP_NEW_EXTENSION(hyperwave, hw.c hg_comm.c, $ext_shared)
   else
     AC_DEFINE(HYPERWAVE,0,[ ])
     AC_MSG_RESULT(no)
