diff -urNp -x '*.orig' php-5.2.17.org/scripts/php-config.in php-5.2.17/scripts/php-config.in
--- php-5.2.17.org/scripts/php-config.in	2007-08-24 13:44:10.000000000 +0200
+++ php-5.2.17/scripts/php-config.in	2021-10-23 19:04:25.856458437 +0200
@@ -3,6 +3,7 @@
 SED="@SED@"
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
+sysconfdir='@sysconfdir@'
 version="@PHP_VERSION@"
 vernum="@PHP_VERSION_ID@"
 include_dir="@includedir@/php"
@@ -53,6 +54,8 @@ case "$1" in
   echo $extension_dir;;
 --include-dir)
   echo $include_dir;;
+--sysconfdir)
+  echo $sysconfdir;;
 --php-binary)
   echo $php_binary;;
 --php-sapis)
@@ -72,6 +75,7 @@ Options:
   --ldflags           [$ldflags]
   --libs              [$libs]
   --extension-dir     [$extension_dir]
+  --sysconfdir        [$sysconfdir]
   --include-dir       [$include_dir]
   --php-binary        [$php_binary]
   --php-sapis         [$php_sapis]
