--- php-5.2.7/ext/xml.org/config.m4	2005-12-22 09:57:38.000000000 +0100
+++ php-5.2.7/ext/xml/config.m4	2008-12-06 12:13:57.781991558 +0100
@@ -5,11 +5,6 @@
 PHP_ARG_ENABLE(xml,whether to enable XML support,
 [  --disable-xml           Disable XML support], yes)
 
-if test -z "$PHP_LIBXML_DIR"; then
-  PHP_ARG_WITH(libxml-dir, libxml2 install dir,
-  [  --with-libxml-dir=DIR     XML: libxml2 install prefix], no, no)
-fi
-
 PHP_ARG_WITH(libexpat-dir, libexpat install dir,
 [  --with-libexpat-dir=DIR   XML: libexpat install prefix (deprecated)], no, no)
 
@@ -19,17 +14,7 @@
   dnl Default to libxml2 if --with-libexpat-dir is not used.
   dnl
   if test "$PHP_LIBEXPAT_DIR" = "no"; then
-
-    if test "$PHP_LIBXML" = "no"; then
-      AC_MSG_ERROR([XML extension requires LIBXML extension, add --enable-libxml])
-    fi
-
-    PHP_SETUP_LIBXML(XML_SHARED_LIBADD, [
-      xml_extra_sources="compat.c"
-      PHP_ADD_EXTENSION_DEP(xml, libxml)
-    ], [
-      AC_MSG_ERROR([xml2-config not found. Use --with-libxml-dir=<DIR>])
-    ])
+      AC_MSG_ERROR([XML extension requires EXPAT extension, add --enable-expat])
   fi
   
   dnl
