--- php-4.3.0/main/php_ini.c.orig	Tue Nov 12 21:56:47 2002
+++ php-4.3.0/main/php_ini.c	Sat Jan  4 12:22:39 2003
@@ -351,6 +351,22 @@
 				}
 			}
 		}
+		/* Search (global) php.ini file in search path */
+		if (!fh.handle.fp) {
+			fh.handle.fp = php_fopen_with_path("php.ini", "r", php_ini_search_path, &php_ini_opened_path TSRMLS_CC);
+			if (fh.handle.fp) {
+				fh.filename = php_ini_opened_path;
+				PG(safe_mode) = safe_mode_state;
+				PG(open_basedir) = open_basedir;
+				fh.type = ZEND_HANDLE_FP;
+				zend_parse_ini_file(&fh, 1, php_config_ini_parser_cb, &extension_lists);
+				safe_mode_state = PG(safe_mode);
+				open_basedir = PG(open_basedir);
+				PG(safe_mode) = 0;
+				PG(open_basedir) = NULL;
+				fh.handle.fp = NULL;
+			}
+		}
 		/* Search php-%sapi-module-name%.ini file in search path */
 		if (!fh.handle.fp) {
 			const char *fmt = "php-%s.ini";
@@ -361,13 +377,6 @@
 			if (fh.handle.fp) {
 				fh.filename = php_ini_opened_path;
 			}
-		}
-		/* Search php.ini file in search path */
-		if (!fh.handle.fp) {
-			fh.handle.fp = php_fopen_with_path("php.ini", "r", php_ini_search_path, &php_ini_opened_path TSRMLS_CC);
-			if (fh.handle.fp) {
-				fh.filename = php_ini_opened_path;
-			}
 		}
 	}
 	if (free_ini_search_path) {
