--- PHP_5_3/ext/pdo_pgsql/config.m4	2011/03/22 09:08:00	309544
+++ PHP_5_3/ext/pdo_pgsql/config.m4	2011/03/22 09:12:01	309545
@@ -69,7 +69,8 @@
   AC_DEFINE(HAVE_PDO_PGSQL,1,[Whether to build PostgreSQL for PDO support or not])
 
   AC_MSG_CHECKING([for openssl dependencies])
-  if grep -q openssl $PGSQL_INCLUDE/libpq-fe.h ; then
+  grep openssl $PGSQL_INCLUDE/libpq-fe.h >/dev/null 2>&1
+  if test $? -eq 0 ; then
     AC_MSG_RESULT([yes])
     dnl First try to find pkg-config
     AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
