diff -urNp -x '*.orig' php-5.5.38.org/scripts/php-config.in php-5.5.38/scripts/php-config.in
--- php-5.5.38.org/scripts/php-config.in	2016-07-20 10:41:48.000000000 +0200
+++ php-5.5.38/scripts/php-config.in	2021-08-23 23:02:47.433283632 +0200
@@ -4,6 +4,7 @@ SED="@SED@"
 prefix="@prefix@"
 datarootdir="@datarootdir@"
 exec_prefix="@exec_prefix@"
+sysconfdir='@sysconfdir@'
 version="@PHP_VERSION@"
 vernum="@PHP_VERSION_ID@"
 include_dir="@includedir@/php"
@@ -55,6 +56,8 @@ case "$1" in
   echo $extension_dir;;
 --include-dir)
   echo $include_dir;;
+--sysconfdir)
+  echo $sysconfdir;;
 --php-binary)
   echo $php_binary;;
 --php-sapis)
@@ -76,6 +79,7 @@ Options:
   --ldflags           [$ldflags]
   --libs              [$libs]
   --extension-dir     [$extension_dir]
+  --sysconfdir        [$sysconfdir]
   --include-dir       [$include_dir]
   --man-dir           [$man_dir]
   --php-binary        [$php_binary]
