diff -urN php5.3-200803022130/ext/pgsql/php_pgsql.h mod/ext/pgsql/php_pgsql.h
--- php5.3-200803022130/ext/pgsql/php_pgsql.h	2008-01-30 11:31:48.000000000 +0000
+++ mod/ext/pgsql/php_pgsql.h	2008-03-02 22:27:42.292062316 +0000
@@ -41,7 +41,7 @@
 #define PHP_PGSQL_API __declspec(dllimport)
 #endif
 #else
-#include <libpq/libpq-fs.h>
+#include <postgresql/server/libpq/libpq-fs.h>
 # if defined(__GNUC__) && __GNUC__ >= 4
 #  define PHP_PGSQL_API __attribute__ ((visibility("default")))
 # else
