--- php-5.2.6/acinclude.m4~	2008-10-15 13:44:25.000000000 +0300
+++ php-5.2.6/acinclude.m4	2008-10-15 13:55:05.359850285 +0300
@@ -2037,8 +2037,10 @@
 dnl Search for the sendmail binary
 dnl
 AC_DEFUN([PHP_PROG_SENDMAIL], [
-  PHP_ALT_PATH=/usr/bin:/usr/sbin:/usr/etc:/etc:/usr/ucblib:/usr/lib
-  AC_PATH_PROG(PROG_SENDMAIL, sendmail,[], $PATH:$PHP_ALT_PATH)
+  if test -z "$PROG_SENDMAIL"; then
+    PHP_ALT_PATH=/usr/bin:/usr/sbin:/usr/etc:/etc:/usr/ucblib:/usr/lib
+    AC_PATH_PROG(PROG_SENDMAIL, sendmail,[], $PATH:$PHP_ALT_PATH)
+  fi
   PHP_SUBST(PROG_SENDMAIL)
 ])
 
