--- php-4.2.0/ext/wddx/Makefile.in.orig	Wed May  1 17:42:43 2002
+++ php-4.2.0/ext/wddx/Makefile.in	Wed May  1 17:53:04 2002
@@ -1,5 +1,9 @@
 
 LTLIBRARY_NAME    = libwddx.la
 LTLIBRARY_SOURCES = wddx.c
+LTLIBRARY_DEPENDENCIES  = $(EXPAT_INTERNAL_LIBADD)
+LTLIBRARY_LIBADD        = $(LTLIBRARY_DEPENDENCIES)
+LTLIBRARY_SHARED_NAME = wddx.la
+LTLIBRARY_SHARED_LIBADD = $(EXPAT_SHARED_LIBADD)
 
 include $(top_srcdir)/build/dynlib.mk
