--- ./main/SAPI.c.org	Sat Jul 27 15:15:42 2002
+++ ./main/SAPI.c	Sat Sep  7 11:13:49 2002
@@ -27,9 +27,6 @@
 #include "SAPI.h"
 #include "ext/standard/php_string.h"
 #include "ext/standard/pageinfo.h"
-#if (HAVE_PCRE || HAVE_BUNDLED_PCRE) && !defined(COMPILE_DL_PCRE)
-#include "ext/pcre/php_pcre.h"
-#endif
 #ifdef ZTS
 #include "TSRM.h"
 #endif
@@ -480,62 +477,6 @@
 				SG(sapi_headers).http_response_code = 401; /* authentication-required */
 
 				if(PG(safe_mode)) 
-#if (HAVE_PCRE || HAVE_BUNDLED_PCRE) && !defined(COMPILE_DL_PCRE)
-				{
-					zval *repl_temp;
-					char *ptr = colon_offset+1;
-					int ptr_len=0, result_len = 0;
-
-					myuid = php_getuid();
-
-					ptr_len = strlen(ptr);
-					MAKE_STD_ZVAL(repl_temp);
-					Z_TYPE_P(repl_temp) = IS_STRING;
-					Z_STRVAL_P(repl_temp) = emalloc(32);
-					Z_STRLEN_P(repl_temp) = sprintf(Z_STRVAL_P(repl_temp), "realm=\"\\1-%ld\"", myuid);
-					/* Modify quoted realm value */
-					result = php_pcre_replace("/realm=\"(.*?)\"/i", 16,
-											 ptr, ptr_len,
-											 repl_temp,
-											 0, &result_len, -1 TSRMLS_CC);
-					if(result_len==ptr_len) {
-						efree(result);
-						sprintf(Z_STRVAL_P(repl_temp), "realm=\\1-%ld\\2", myuid);
-						/* modify unquoted realm value */
-						result = php_pcre_replace("/realm=([^\\s]+)(.*)/i", 21, 
-											 	ptr, ptr_len,
-											 	repl_temp,
-											 	0, &result_len, -1 TSRMLS_CC);
-						if(result_len==ptr_len) {
-							char *lower_temp = estrdup(ptr);	
-							char conv_temp[32];
-							int conv_len;
-
-							php_strtolower(lower_temp,strlen(lower_temp));
-							/* If there is no realm string at all, append one */
-							if(!strstr(lower_temp,"realm")) {
-								efree(result);
-								conv_len = sprintf(conv_temp," realm=\"%ld\"",myuid);		
-								result = emalloc(ptr_len+conv_len+1);
-								result_len = ptr_len+conv_len;
-								memcpy(result, ptr, ptr_len);	
-								memcpy(result+ptr_len, conv_temp, conv_len);
-								*(result+ptr_len+conv_len) = '\0';
-							}
-							efree(lower_temp);
-						}
-					}
-					newlen = sizeof("WWW-Authenticate: ") + result_len;
-					newheader = emalloc(newlen+1);
-					sprintf(newheader,"WWW-Authenticate: %s", result);
-					efree(header_line);
-					sapi_header.header = newheader;
-					sapi_header.header_len = newlen;
-					efree(result);
-					efree(Z_STRVAL_P(repl_temp));
-					efree(repl_temp);
-				} 
-#else
 				{
 					myuid = php_getuid();
 					result = emalloc(32);
@@ -546,7 +487,6 @@
 					sapi_header.header_len = newlen;
 					efree(result);
 				} 
-#endif
 			}
 			if (sapi_header.header==header_line) {
 				*colon_offset = ':';
