diff -up php-5.2.17/ext/standard/info.c.bug-52461 php-5.2.17/ext/standard/info.c
--- php-5.2.17/ext/standard/info.c.bug-52461	2012-01-12 10:23:00.000000000 +0700
+++ php-5.2.17/ext/standard/info.c	2012-01-12 10:23:27.000000000 +0700
@@ -415,7 +415,7 @@ PHPAPI void php_print_info_htmlhead(TSRM
 
 
 	PUTS("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"DTD/xhtml1-transitional.dtd\">\n");
-	PUTS("<html>");
+	PUTS("<html xmlns=\"http://www.w3.org/1999/xhtml\">>");
 	PUTS("<head>\n");
 	php_info_print_style(TSRMLS_C);
 	PUTS("<title>phpinfo()</title>");
