--- PHP_5_3/ext/pdo/pdo_stmt.c	2011/06/01 12:53:07	311710
+++ PHP_5_3/ext/pdo/pdo_stmt.c	2011/06/01 13:23:25	311711
@@ -349,7 +349,10 @@
 		/* if you prepare and then execute passing an array of params keyed by names,
 		 * then this will trigger, and we don't want that */
 		if (param->paramno == -1) {
-			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Did not found column name '%s' in the defined columns; it will not be bound", param->name);
+			char *tmp;
+			spprintf(&tmp, 0, "Did not find column name '%s' in the defined columns; it will not be bound", param->name);
+			pdo_raise_impl_error(stmt->dbh, stmt, "HY000", tmp TSRMLS_CC);
+			efree(tmp);
 		}
 	}
 
