diff -urNp -x '*.orig' php-5.6.40.org/ext/sybase_ct/config.m4 php-5.6.40/ext/sybase_ct/config.m4
--- php-5.6.40.org/ext/sybase_ct/config.m4	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/sybase_ct/config.m4	2021-08-23 23:19:35.112450270 +0200
@@ -50,7 +50,7 @@ if test "$PHP_SYBASE_CT" != "no"; then
   AC_MSG_CHECKING([Checking for Sybase platform libraries])
 
   PHP_ADD_LIBPATH($SYBASE_CT_LIBDIR, SYBASE_CT_SHARED_LIBADD)
-  if test -f $SYBASE_CT_INCDIR/tds.h || test -f $SYBASE_CT_INCDIR/tds_sysdep_public.h; then
+  if test -f $SYBASE_CT_INCDIR/tds.h || test -f $SYBASE_CT_INCDIR/tds_sysdep_public.h || test -f $SYBASE_CT_INCDIR/sybdb.h; then
     PHP_ADD_LIBRARY(ct,, SYBASE_CT_SHARED_LIBADD)
     SYBASE_CT_LIBS="-L$SYBASE_CT_LIBDIR -lct"
     AC_MSG_RESULT([FreeTDS: $SYBASE_CT_LIBS])
