--- PHP_5_3/ext/imap/php_imap.c	2010/11/23 10:22:34	305685
+++ PHP_5_3/ext/imap/php_imap.c	2010/11/23 10:34:44	305686
@@ -4235,7 +4235,7 @@
 					}
 
 					offset = end_token+2;
-					for (i = 0; (string[offset + i] == ' ') || (string[offset + i] == 0x0a) || (string[offset + i] == 0x0d); i++);
+					for (i = 0; (string[offset + i] == ' ') || (string[offset + i] == 0x0a) || (string[offset + i] == 0x0d) || (string[offset + i] == '\t'); i++);
 					if ((string[offset + i] == '=') && (string[offset + i + 1] == '?') && (offset + i < end)) {
 						offset += i;
 					}
