diff -up php-5.2.17/ext/zip/zip_stream.c.bug-49072 php-5.2.17/ext/zip/zip_stream.c
--- php-5.2.17/ext/zip/zip_stream.c.bug-49072	2011-08-28 14:06:52.000000000 +0700
+++ php-5.2.17/ext/zip/zip_stream.c	2011-08-28 14:09:41.000000000 +0700
@@ -34,7 +34,7 @@ static size_t php_zip_ops_read(php_strea
 	STREAM_DATA_FROM_STREAM();
 
 	if (self->za && self->zf) {
-		n = (size_t)zip_fread(self->zf, buf, (int)count);
+		n = zip_fread(self->zf, buf, count);
 		if (n < 0) {
 			int ze, se;
 			zip_file_error_get(self->zf, &ze, &se);
@@ -42,13 +42,13 @@ static size_t php_zip_ops_read(php_strea
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Zip stream error: %s", zip_file_strerror(self->zf));
 			return 0;
 		}
-		if (n == 0 || n < count) {
+		if (n == 0 || n < (ssize_t)count) {
 			stream->eof = 1;
 		} else {
 			self->cursor += n;
 		}
 	}
-	return n<1 ? 0 : n;
+	return (n < 1 ? 0 : (size_t)n);
 }
 /* }}} */
 
