--- php-5.0.3/TSRM/TSRM.h.orig	2003-12-03 15:26:41.000000000 +0100
+++ php-5.0.3/TSRM/TSRM.h	2005-02-13 20:57:08.000000000 +0100
@@ -151,7 +151,7 @@
 
 #define TSRMLS_FETCH()			void ***tsrm_ls = (void ***) ts_resource_ex(0, NULL)
 #define TSRMLS_FETCH_FROM_CTX(ctx)	void ***tsrm_ls = (void ***) ctx
-#define TSRMLS_SET_CTX(ctx)		(void ***) ctx = tsrm_ls
+#define TSRMLS_SET_CTX(ctx)		ctx = tsrm_ls
 #define TSRMG(id, type, element)	(((type) (*((void ***) tsrm_ls))[TSRM_UNSHUFFLE_RSRC_ID(id)])->element)
 #define TSRMLS_D	void ***tsrm_ls
 #define TSRMLS_DC	, TSRMLS_D
