diff -urN php5.3-200806102030/configure.in mod/configure.in
--- php5.3-200806102030/configure.in    2008-06-02 18:33:55.000000000 +0000
+++ mod/configure.in    2008-06-10 21:04:48.282651967 +0000
@@ -1353,7 +1353,7 @@
 fi;

 all_targets="$lcov_target \$(OVERALL_TARGET) \$(PHP_MODULES) \$(PHP_ZEND_EX) \$(PHP_CLI_TARGET) \$(PHAR_TARGET) $pharcmd"
-install_targets="$install_modules install-build install-headers install-programs $install_pear $pharcmd_install"
+install_targets="$install_modules install-build install-headers install-programs $pharcmd_install"

 case $PHP_SAPI in
   cli)
