--- php-4.4.1/Zend/zend.c	2005-06-09 13:14:25.000000000 +0300
+++ php4-STABLE-200511081543/Zend/zend.c	2005-11-07 16:01:13.000000000 +0200
@@ -17,7 +17,7 @@
    +----------------------------------------------------------------------+
 */
 
-/* $Id$ */
+/* $Id$ */
 
 #include "zend.h"
 #include "zend_extensions.h"
@@ -564,9 +564,7 @@
 #ifdef ZEND_WIN32
 	zend_shutdown_timeout_thread();
 #endif
-#ifndef ZTS
 	zend_destroy_rsrc_list(&EG(persistent_list) TSRMLS_CC);
-#endif	
 	zend_hash_graceful_reverse_destroy(&module_registry);
 
 	zend_shutdown_extensions(TSRMLS_C);
@@ -584,8 +582,6 @@
 	zend_shutdown_constants(TSRMLS_C);
 
 #ifdef ZTS
-	zend_destroy_rsrc_list(&EG(persistent_list) TSRMLS_CC);
-
 	zend_hash_destroy(GLOBAL_CONSTANTS_TABLE);
 	free(GLOBAL_CONSTANTS_TABLE);
 
