--- php-4.3.0/ext/java/config.m4.orig	Sat Jan  4 10:51:49 2003
+++ php-4.3.0/ext/java/config.m4	Sat Jan  4 10:54:48 2003
@@ -58,7 +58,6 @@
 AC_DEFUN(JAVA_CHECK_LIB, [
  AC_MSG_CHECKING([Checking for libjava])
  if test -d $PHP_JAVA/lib/kaffe; then
-   PHP_ADD_LIBPATH($PHP_JAVA/lib)
    JAVA_CFLAGS=-DKAFFE
    JAVA_INCLUDE=-I$PHP_JAVA/include/kaffe
    JAVA_CLASSPATH=$PHP_JAVA/share/kaffe/Klasses.jar
@@ -111,10 +110,6 @@
        JAVA_LIB=java
        JAVA_LIBPATH=$i
 
-       test -d $i/hotspot && PHP_ADD_LIBPATH($i/hotspot)
-       test -d $i/classic && PHP_ADD_LIBPATH($i/classic)
-       test -d $i/server  && PHP_ADD_LIBPATH($i/server)
-       test -d $i/native_threads && PHP_ADD_LIBPATH($i/native_threads)
      fi
    done
 
@@ -160,19 +155,12 @@
       AC_MSG_ERROR([unable to find Java VM libraries in $PHP_JAVA])
     fi
 
-    PHP_ADD_LIBPATH($JAVA_LIBPATH)
     JAVA_CFLAGS="$JAVA_CFLAGS '-DJAVALIB=\"$JAVA_LIBPATH/$java_libext\"'"
   fi
 
   if test "$PHP_SAPI" != "servlet"; then
     PHP_NEW_EXTENSION(java, java.c, shared,, $JAVA_CFLAGS $JAVA_INCLUDE)
 
-    if test "$PHP_SAPI" = "cgi"; then
-      if test "$platform" != "Darwin"; then
-        PHP_ADD_LIBRARY($JAVA_LIB)
-      fi
-    fi
-
     INSTALL_IT="$INSTALL_IT; \$(srcdir)/build/shtool mkdir -p -f -m 0755 \$(INSTALL_ROOT)\$(libdir)"
     INSTALL_IT="$INSTALL_IT; \$(INSTALL) -m 0755 \$(srcdir)/ext/java/php_java.jar \$(INSTALL_ROOT)\$(libdir)"
   fi
