--- php-4.0.6/ext/yp/Makefile.in.wiget	Tue May  2 03:21:26 2000
+++ php-4.0.6/ext/yp/Makefile.in	Wed Jul  4 22:44:51 2001
@@ -2,5 +2,6 @@
 LTLIBRARY_NAME    = libyp.la
 LTLIBRARY_SOURCES = yp.c
 LTLIBRARY_SHARED_NAME = yp.la
+LTLIBRARY_SHARED_LIBADD = $(NSL_SHARED_LIBADD)
 
 include $(top_srcdir)/build/dynlib.mk
--- php-4.1.1/ext/yp/config.m4.wiget	Tue May  2 03:21:26 2000
+++ php-4.1.1/ext/yp/config.m4	Wed Jul  4 22:44:51 2001
@@ -6,8 +6,11 @@
 [  --enable-yp             Include YP support])
 
 if test "$PHP_YP" != "no"; then
-  AC_DEFINE(HAVE_YP,1,[ ])
+  PHP_SUBST(NSL_SHARED_LIBADD)
+  AC_ADD_LIBRARY_WITH_PATH(nsl, /usr/lib, NSL_SHARED_LIBADD)
+  AC_CHECK_LIB(nsl, yp_match, [AC_DEFINE(HAVE_YP,1,[ ])], [AC_MSG_ERROR(YP module requires libnsl)],)
   PHP_EXTENSION(yp, $ext_shared)
+
   case $host_alias in
   *solaris*)
     AC_DEFINE(SOLARIS_YP,1,[ ]) ;;
--- php-4.0.6/configure.in.wiget	Wed Jun 20 22:12:59 2001
+++ php-4.0.6/configure.in	Wed Jul  4 22:49:05 2001
@@ -230,6 +230,7 @@
 
 if test "$php_no_nsl_checks" != "yes"; then
 
+AC_CHECK_FUNC(gethostname,[true], [
 AC_CHECK_LIB(nsl, gethostname, [
  PHP_ADD_LIBRARY(nsl)
  AC_DEFINE(HAVE_LIBNSL,1,[ ]) ],[
@@ -237,6 +238,7 @@
   PHP_ADD_LIBRARY(nsl)
   AC_DEFINE(HAVE_LIBNSL,1,[ ]) ], [])
 ])
+])
 
 fi
 
--- php-4.0.6.orig/ext/gmp/config.m4	Tue Mar 27 22:34:27 2001
+++ php-4.0.6/ext/gmp/config.m4	Fri Aug  3 15:34:22 2001
@@ -23,5 +23,6 @@
 
   PHP_EXTENSION(gmp, $ext_shared)
   AC_DEFINE(HAVE_GMP, 1, [ ])
-  PHP_ADD_LIBRARY_WITH_PATH(gmp, $GMP_DIR/lib)
+  PHP_ADD_LIBRARY_WITH_PATH(gmp, $GMP_DIR/lib, GMP_SHARED_LIBADD)
 fi
+PHP_SUBST(GMP_SHARED_LIBADD)
--- php-4.2.0/ext/xslt/config.m4.orig	Tue Apr 16 12:12:57 2002
+++ php-4.2.0/ext/xslt/config.m4	Wed May  1 16:29:09 2002
@@ -65,8 +65,8 @@
     done
 
     if test "$found_expat" = "no"; then
-      PHP_ADD_LIBRARY(xmlparse)
-      PHP_ADD_LIBRARY(xmltok)
+      PHP_ADD_LIBRARY(xmlparse,, XSLT_SHARED_LIBADD)
+      PHP_ADD_LIBRARY(xmltok,, XSLT_SHARED_LIBADD)
     fi
 
     if test "$PHP_ICONV" = "no"; then
@@ -80,7 +80,7 @@
       found_js=no
       AC_CHECK_LIB(js, JS_GetRuntime, found_js=yes)
       if test "$found_js" = "yes"; then
-        PHP_ADD_LIBRARY(js)
+        PHP_ADD_LIBRARY(js,, XSLT_SHARED_LIBADD)
       fi
     fi
 
