--- php-4.0.4pl1/build/buildcheck.sh~	Sun Aug 20 07:39:37 2000
+++ php-4.0.4pl1/build/buildcheck.sh	Thu May  3 11:37:29 2001
@@ -67,7 +67,7 @@
 fi
 lt_version=`echo $lt_pversion|sed -e 's/\([a-z]*\)$/.\1/'`
 IFS=.; set $lt_version; IFS=' '
-if test "$1" -gt "1" || test "$2" -gt "3" || test "$2" = "3" && (test "$3" = "c" || test "$3" -ge "3")
+if test "$1" -gt "1" || test "$2" -gt "3" || ( test "$2" = "3" && test "$3" -ge "3")
 then
 echo "buildconf: libtool version $lt_pversion (ok)"
 else
