diff -urbB php-5.1.0RC1.org/ext/gd/gd.c php-5.1.0RC1/ext/gd/gd.c
--- php-5.1.0RC1.org/ext/gd/gd.c	2005-08-25 15:26:22.317049000 +0200
+++ php-5.1.0RC1/ext/gd/gd.c	2005-08-25 15:27:37.644598376 +0200
@@ -209,8 +209,8 @@
 	PHP_FE(imagecopyresampled,						NULL)
 #endif
 
-#ifdef HAVE_GD_BUNDLED
 	PHP_FE(imagerotate,     						NULL)
+#ifdef HAVE_GD_BUNDLED
 	PHP_FE(imageantialias,							NULL)
 #endif
 
@@ -429,11 +429,11 @@
 #ifdef GD2_FMT_COMPRESSED
 	REGISTER_LONG_CONSTANT("IMG_GD2_COMPRESSED", GD2_FMT_COMPRESSED, CONST_CS | CONST_PERSISTENT);
 #endif
-#if HAVE_GD_BUNDLED
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_REPLACE", gdEffectReplace, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_ALPHABLEND", gdEffectAlphaBlend, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_NORMAL", gdEffectNormal, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_OVERLAY", gdEffectOverlay, CONST_CS | CONST_PERSISTENT);
+#if HAVE_GD_BUNDLED
 	REGISTER_LONG_CONSTANT("GD_BUNDLED", 1, CONST_CS | CONST_PERSISTENT);
 	/* Section Filters */
 	REGISTER_LONG_CONSTANT("IMG_FILTER_NEGATE", IMAGE_FILTER_NEGATE, CONST_CS | CONST_PERSISTENT);
@@ -1204,7 +1204,6 @@
 /* }}} */
 #endif
 
-#ifdef HAVE_GD_BUNDLED
 /* {{{ proto resource imagerotate(resource src_im, float angle, int bgdcolor)
    Rotate an image using a custom angle */
 PHP_FUNCTION(imagerotate)
@@ -1221,7 +1220,7 @@
 
 	ZEND_FETCH_RESOURCE(im_src, gdImagePtr, &SIM, -1, "Image", le_gd);
 
-	im_dst = gdImageRotate(im_src, degrees, color, ignoretransparent);
+	im_dst = gdImageRotate(im_src, degrees, color);
 
 	if (im_dst != NULL) {
 		ZEND_REGISTER_RESOURCE(return_value, im_dst, le_gd);
@@ -1230,7 +1229,6 @@
 	}
 }
 /* }}} */
-#endif
 
 #if HAVE_GD_IMAGESETTILE
 /* {{{ proto bool imagesettile(resource image, resource tile)
diff -urbB php-5.1.0RC1.org/ext/gd/php_gd.h php-5.1.0RC1/ext/gd/php_gd.h
--- php-5.1.0RC1.org/ext/gd/php_gd.h	2005-08-25 15:26:22.366042000 +0200
+++ php-5.1.0RC1/ext/gd/php_gd.h	2005-08-25 15:27:37.645598224 +0200
@@ -103,8 +103,8 @@
 PHP_FUNCTION(imagecopyresampled);
 #endif
 
-#ifdef HAVE_GD_BUNDLED
 PHP_FUNCTION(imagerotate);
+#ifdef HAVE_GD_BUNDLED
 PHP_FUNCTION(imageantialias);
 #endif
 
