--- php-5.2.4/ext/soap/php_sdl.c~	2007-05-21 16:13:50.000000000 +0300
+++ php-5.2.4/ext/soap/php_sdl.c	2007-10-12 19:53:36.044545571 +0300
@@ -2012,11 +2012,7 @@
 	HashTable tmp_bindings;
 	HashTable tmp_functions;
 
-#ifdef ZEND_WIN32
 	f = open(fn,O_CREAT|O_WRONLY|O_EXCL|O_BINARY,S_IREAD|S_IWRITE);
-#else
-	f = open(fn,O_CREAT|O_WRONLY|O_EXCL|O_BINARY,S_IREAD|S_IWRITE|S_IROTH|S_IWOTH|S_IRGRP|S_IWGRP);
-#endif
 	if (f < 0) {return;}
 
 	zend_hash_init(&tmp_types, 0, NULL, NULL, 0);
