--- php-5.0.3/ext/hwapi/config.m4.orig	2002-09-19 22:14:33.000000000 +0200
+++ php-5.0.3/ext/hwapi/config.m4	2004-12-26 16:34:42.490945536 +0100
@@ -28,8 +28,8 @@
 
   PHP_REQUIRE_CXX
   PHP_NEW_EXTENSION(hwapi, hwapi.cpp, $ext_shared)
-  PHP_ADD_LIBRARY(hwapi)
-  PHP_ADD_LIBRARY(pthread)
-  PHP_ADD_LIBRARY(stdc++)
+  PHP_ADD_LIBRARY(hwapi,,HWAPI_SHARED_LIBADD)
+  PHP_ADD_LIBRARY(pthread,,HWAPI_SHARED_LIBADD)
+  PHP_ADD_LIBRARY(stdc++,,HWAPI_SHARED_LIBADD)
   PHP_ADD_INCLUDE($HWAPI_DIR/include)
 fi
