diff -urNp -x '*.orig' php-5.2.17.org/ext/mysqli/mysqli.c php-5.2.17/ext/mysqli/mysqli.c
--- php-5.2.17.org/ext/mysqli/mysqli.c	2010-08-13 12:43:15.000000000 +0200
+++ php-5.2.17/ext/mysqli/mysqli.c	2021-10-23 19:07:32.689791758 +0200
@@ -451,6 +451,7 @@ PHP_INI_BEGIN()
 	STD_PHP_INI_ENTRY("mysqli.default_pw",				NULL,	PHP_INI_ALL,		OnUpdateString,		default_pw,			zend_mysqli_globals,		mysqli_globals)
 	STD_PHP_INI_ENTRY("mysqli.default_port",			"3306",	PHP_INI_ALL,		OnUpdateLong,		default_port,		zend_mysqli_globals,		mysqli_globals)
 	STD_PHP_INI_ENTRY("mysqli.default_socket",			NULL,	PHP_INI_ALL,		OnUpdateStringUnempty,	default_socket,	zend_mysqli_globals,		mysqli_globals)
+	STD_PHP_INI_ENTRY("mysqli.connect_charset",			NULL,	PHP_INI_ALL,		OnUpdateString,		connect_charset,	zend_mysqli_globals,		mysqli_globals)
 	STD_PHP_INI_BOOLEAN("mysqli.reconnect",				"0",	PHP_INI_SYSTEM,		OnUpdateLong,		reconnect,			zend_mysqli_globals,		mysqli_globals)
 PHP_INI_END()
 
@@ -467,6 +468,7 @@ static PHP_GINIT_FUNCTION(mysqli)
 	mysqli_globals->default_user = NULL;
 	mysqli_globals->default_pw = NULL;
 	mysqli_globals->default_socket = NULL;
+	mysqli_globals->connect_charset = NULL;
 	mysqli_globals->reconnect = 0;
 	mysqli_globals->report_mode = 0;
 	mysqli_globals->report_ht = 0;
diff -urNp -x '*.orig' php-5.2.17.org/ext/mysqli/mysqli_api.c php-5.2.17/ext/mysqli/mysqli_api.c
--- php-5.2.17.org/ext/mysqli/mysqli_api.c	2021-10-23 19:07:30.269791758 +0200
+++ php-5.2.17/ext/mysqli/mysqli_api.c	2021-10-23 19:07:32.689791758 +0200
@@ -1428,7 +1428,7 @@ PHP_FUNCTION(mysqli_prepare)
 PHP_FUNCTION(mysqli_real_connect)
 {
 	MY_MYSQL 		*mysql;
-	char 			*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL;
+	char 			*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL, *connect_charset=NULL;
 	unsigned int 	hostname_len = 0, username_len = 0, passwd_len = 0, dbname_len = 0, socket_len = 0;
 	unsigned long 	port=0, flags=0;
 	zval			*mysql_link;
@@ -1478,6 +1478,12 @@ PHP_FUNCTION(mysqli_real_connect)
 		socket = MyG(default_socket);
 	}
 
+	connect_charset = MyG(connect_charset);
+
+	if (connect_charset != NULL) {
+		mysql_options(mysql->mysql, MYSQL_SET_CHARSET_NAME, connect_charset);
+	}
+
 	if (mysql_real_connect(mysql->mysql,hostname,username,passwd,dbname,port,socket,flags) == NULL) {
 		php_mysqli_set_error(mysql_errno(mysql->mysql), (char *) mysql_error(mysql->mysql) TSRMLS_CC);
 		php_mysqli_throw_sql_exception( mysql->mysql->net.sqlstate, mysql->mysql->net.last_errno TSRMLS_CC,
diff -urNp -x '*.orig' php-5.2.17.org/ext/mysqli/mysqli_nonapi.c php-5.2.17/ext/mysqli/mysqli_nonapi.c
--- php-5.2.17.org/ext/mysqli/mysqli_nonapi.c	2010-01-25 14:23:32.000000000 +0100
+++ php-5.2.17/ext/mysqli/mysqli_nonapi.c	2021-10-23 19:07:32.689791758 +0200
@@ -36,7 +36,7 @@ PHP_FUNCTION(mysqli_connect)
 	MY_MYSQL 			*mysql = NULL;
 	MYSQLI_RESOURCE 	*mysqli_resource = NULL;
 	zval  				*object = getThis();
-	char 				*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL;
+	char 				*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL, *connect_charset=NULL;
 	unsigned int 		hostname_len = 0, username_len = 0, passwd_len = 0, dbname_len = 0, socket_len = 0;
 	long				port=0;
 
@@ -116,6 +116,12 @@ PHP_FUNCTION(mysqli_connect)
 		socket = MyG(default_socket);
 	}
 
+	connect_charset = MyG(connect_charset);
+
+	if (connect_charset != NULL) {
+		mysql_options(mysql->mysql, MYSQL_SET_CHARSET_NAME, connect_charset);
+	}
+
 	if (mysql_real_connect(mysql->mysql,hostname,username,passwd,dbname,port,socket,CLIENT_MULTI_RESULTS) == NULL) {
 		/* Save error messages */
 
diff -urNp -x '*.orig' php-5.2.17.org/ext/mysqli/php_mysqli.h php-5.2.17/ext/mysqli/php_mysqli.h
--- php-5.2.17.org/ext/mysqli/php_mysqli.h	2010-01-25 14:23:32.000000000 +0100
+++ php-5.2.17/ext/mysqli/php_mysqli.h	2021-10-23 19:07:32.689791758 +0200
@@ -442,6 +442,7 @@ ZEND_BEGIN_MODULE_GLOBALS(mysqli)
 	char			*default_user;
 	char			*default_socket;
 	char            *default_pw;
+	char			*connect_charset;
 	int				reconnect;
 	int				strict;
 	long			error_no;
