diff -Nur php-4.2.0/ext/java.orig/Makefile.in php-4.2.0/ext/java/Makefile.in
--- php-4.2.0/ext/java.orig/Makefile.in	Sun Apr 14 02:40:57 2002
+++ php-4.2.0/ext/java/Makefile.in	Thu May  2 00:35:25 2002
@@ -4,7 +4,7 @@
 LTLIBRARY_DEPENDENCIES = php_java.jar
 
 LTLIBRARY_LDFLAGS	= $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS)
-LTLIBRARY_SHARED_LIBADD = $(LTLIBRARY_DEPENDENCIES) $(EXTRA_LIBS)
+LTLIBRARY_SHARED_LIBADD = $(LTLIBRARY_DEPENDENCIES) $(EXTRA_LIBS) $(JAVA_SHARED_LIBADD)
 
 EXTRA_CFLAGS   = $(JAVA_CFLAGS)
 EXTRA_INCLUDES = $(JAVA_INCLUDE)
diff -Nur php-4.2.0/ext/java.orig/config.m4 php-4.2.0/ext/java/config.m4
--- php-4.2.0/ext/java.orig/config.m4	Sun Apr 14 02:40:57 2002
+++ php-4.2.0/ext/java/config.m4	Thu May  2 00:37:36 2002
@@ -8,6 +8,7 @@
                           This extension is always built as shared.],
 [
   if test "$withval" != "no"; then
+    ext_shared=yes
     JAVA_SHARED=libphp_java.la
 
     pltform=`uname -s 2>/dev/null`
@@ -111,7 +111,6 @@
       PHP_EXTENSION(java, shared)
 
-      if test "$PHP_SAPI" = "cgi"; then
-        PHP_ADD_LIBRARY($JAVA_LIB)
+        PHP_SUBST(JAVA_SHARED_LIBADD)
+        PHP_ADD_LIBRARY($JAVA_LIB,, JAVA_SHARED_LIBADD)
-      fi
 
       INSTALL_IT="$INSTALL_IT; \$(srcdir)/build/shtool mkdir -p -f -m 0755 \$(INSTALL_ROOT)\$(libdir)"
