--- PHP_5_3/main/streams/userspace.c	2011/02/01 20:59:25	307933
+++ PHP_5_3/main/streams/userspace.c	2011/02/01 22:55:17	307934
@@ -856,6 +856,7 @@
 
 #define STAT_PROP_ENTRY_EX(name, name2)                        \
 	if (SUCCESS == zend_hash_find(Z_ARRVAL_P(array), #name, sizeof(#name), (void**)&elem)) {     \
+		SEPARATE_ZVAL(elem);																	 \
 		convert_to_long(*elem);                                                                   \
 		ssb->sb.st_##name2 = Z_LVAL_PP(elem);                                                      \
 	}
