--- php-4.3.4RC2/ext/mssql/config.m4.old	2003-10-29 02:26:59.000000000 +0100
+++ php-4.3.4RC2/ext/mssql/config.m4	2003-10-29 02:26:36.000000000 +0100
@@ -32,8 +32,8 @@
     fi
   fi  
 
-  if test ! -r "$FREETDS_INSTALLATION_DIR/lib/libtds.a"; then
-     AC_MSG_ERROR(Could not find $FREETDS_INSTALLATION_DIR/lib/libtds.a)
+  if test ! -r "$FREETDS_INSTALLATION_DIR/lib/libtds.so"; then
+     AC_MSG_ERROR(Could not find $FREETDS_INSTALLATION_DIR/lib/libtds.so)
   fi
 
   MSSQL_INCDIR=$FREETDS_INSTALLATION_DIR/include
