diff -urNp -x '*.orig' php-5.3.29.org/sapi/cgi/cgi_main.c php-5.3.29/sapi/cgi/cgi_main.c
--- php-5.3.29.org/sapi/cgi/cgi_main.c	2021-08-22 23:30:39.448632666 +0200
+++ php-5.3.29/sapi/cgi/cgi_main.c	2021-08-22 23:30:41.575296952 +0200
@@ -698,12 +698,7 @@ static void sapi_cgi_log_message(char *m
 		request = (fcgi_request*) SG(server_context);
 		if (request) {
 			int len = strlen(message);
-			char *buf = malloc(len+2);
-
-			memcpy(buf, message, len);
-			memcpy(buf + len, "\n", sizeof("\n"));
-			fcgi_write(request, FCGI_STDERR, buf, len+1);
-			free(buf);
+			fcgi_write(request, FCGI_STDERR, message, len);
 		} else {
 			fprintf(stderr, "%s\n", message);
 		}
