diff -ur php-4.0.6/ext/oci8/config.m4 php-4.0.6.ora/ext/oci8/config.m4
--- php-4.0.6/ext/oci8/config.m4	Sat May 12 11:29:07 2001
+++ php-4.0.6.ora/ext/oci8/config.m4	Mon Sep 17 16:15:22 2001
@@ -5,6 +5,8 @@
   if test -s "$OCI8_DIR/orainst/unix.rgs"; then
 	OCI8_VERSION=`grep '"ocommon"' $OCI8_DIR/orainst/unix.rgs | sed 's/[ ][ ]*/:/g' | cut -d: -f 6 | cut -c 2-4`
 	test -z "$OCI8_VERSION" && OCI8_VERSION=7.3
+  elif test -f $OCI8_DIR/lib/libclntsh.s?.9.0; then
+	OCI8_VERSION=9.0
   elif test -f $OCI8_DIR/lib/libclntsh.s?.8.0; then
 	OCI8_VERSION=8.1
   elif test -f $OCI8_DIR/lib/libclntsh.s?.1.0; then
@@ -72,6 +74,11 @@
 	  ;;
 
 	8.1)
+  	  PHP_ADD_LIBRARY(clntsh, 1, OCI8_SHARED_LIBADD)
+  	  PHP_ADD_LIBPATH($OCI8_DIR/lib, OCI8_SHARED_LIBADD)
+  	  AC_DEFINE(HAVE_OCI8_TEMP_LOB,1,[ ])
+	  ;;
+	9.0)
   	  PHP_ADD_LIBRARY(clntsh, 1, OCI8_SHARED_LIBADD)
   	  PHP_ADD_LIBPATH($OCI8_DIR/lib, OCI8_SHARED_LIBADD)
   	  AC_DEFINE(HAVE_OCI8_TEMP_LOB,1,[ ])
diff -ur php-4.0.6/ext/oracle/config.m4 php-4.0.6.ora/ext/oracle/config.m4
--- php-4.0.6/ext/oracle/config.m4	Sat May 12 11:29:07 2001
+++ php-4.0.6.ora/ext/oracle/config.m4	Mon Sep 17 16:16:43 2001
@@ -5,6 +5,8 @@
   if test -s "$ORACLE_DIR/orainst/unix.rgs"; then
   	ORACLE_VERSION=`grep '"ocommon"' $ORACLE_DIR/orainst/unix.rgs | sed 's/[ ][ ]*/:/g' | cut -d: -f 6 | cut -c 2-4`
 	test -z "$ORACLE_VERSION" && ORACLE_VERSION=7.3
+  elif test -f $ORACLE_DIR/lib/libclntsh.s?.9.0; then
+    ORACLE_VERSION=9.0
   elif test -f $ORACLE_DIR/lib/libclntsh.s?.8.0; then
     ORACLE_VERSION=8.1
   elif test -f $ORACLE_DIR/lib/libclntsh.s?.1.0; then
@@ -148,6 +150,10 @@
 	  ;;
 
 	8.1)
+  	  PHP_ADD_LIBRARY(clntsh, 1, ORACLE_SHARED_LIBADD)
+  	  PHP_ADD_LIBPATH($ORACLE_DIR/lib, ORACLE_SHARED_LIBADD)
+	  ;;
+	9.0)
   	  PHP_ADD_LIBRARY(clntsh, 1, ORACLE_SHARED_LIBADD)
   	  PHP_ADD_LIBPATH($ORACLE_DIR/lib, ORACLE_SHARED_LIBADD)
 	  ;;
