--- PHP_5_3/ext/filter/filter.c	2010/12/22 16:03:43	306574
+++ PHP_5_3/ext/filter/filter.c	2010/12/22 16:18:59	306575
@@ -559,7 +559,7 @@
 			if (jit_initialization) {
 				zend_is_auto_global("_ENV", sizeof("_ENV")-1 TSRMLS_CC);
 			}
-			array_ptr = IF_G(env_array);
+			array_ptr = IF_G(env_array) ? IF_G(env_array) : PG(http_globals)[TRACK_VARS_ENV];
 			break;
 		case PARSE_SESSION:
 			/* FIXME: Implement session source */
