--- php-4.3.0/ext/hyperwave/config.m4.orig	Thu Jan  2 23:17:53 2003
+++ php-4.3.0/ext/hyperwave/config.m4	Thu Jan  2 23:19:42 2003
@@ -2,19 +2,12 @@
 dnl $Id$
 dnl
 
-AC_MSG_CHECKING(for Hyperwave support)
-AC_ARG_WITH(hyperwave,
-[  --with-hyperwave        Include Hyperwave support],
-[
-  if test "$withval" != "no"; then
+PHP_ARG_WITH(hyperwave, for Hyperwave support,
+[  --with-hyperwave        Include Hyperwave support])
+
+  if test "$PHP_HYPERWAVE" != "no"; then
     AC_DEFINE(HYPERWAVE,1,[ ])
-    AC_MSG_RESULT(yes)
-    PHP_NEW_EXTENSION(hyperwave, hw.c hg_comm.c)
+    PHP_NEW_EXTENSION(hyperwave, hw.c hg_comm.c, $ext_shared)
   else
     AC_DEFINE(HYPERWAVE,0,[ ])
-    AC_MSG_RESULT(no)
   fi
-],[
-  AC_DEFINE(HYPERWAVE,0,[ ])
-  AC_MSG_RESULT(no)
-])
