diff -urNp -x '*.orig' php-5.5.38.org/configure.in php-5.5.38/configure.in
--- php-5.5.38.org/configure.in	2021-08-23 23:02:44.256618284 +0200
+++ php-5.5.38/configure.in	2021-08-23 23:02:45.126617925 +0200
@@ -1075,11 +1075,6 @@ if test -z "$with_pear" && test "$enable
   with_pear=no
 fi
 
-# If CLI is disabled -> disable PEAR
-if test "$PHP_CLI" = "no"; then
-  with_pear=no
-fi
-
 PHP_ARG_WITH(pear, [whether to install PEAR],
 [  --with-pear=DIR         Install PEAR in DIR @<:@PREFIX/lib/php@:>@
   --without-pear          Do not install PEAR], DEFAULT, yes)
@@ -1089,11 +1084,11 @@ if test "$PHP_PEAR" != "no"; then
   dnl
   dnl PEAR dependancies
   dnl
-  if test "$PHP_XML" = "no"; then
-    pear_error_msg="$pear_error_msg 
-                    PEAR requires XML to be enabled.     Add --enable-xml to the configure line. (or --without-pear)"
-  fi
-
+dnl  if test "$PHP_XML" = "no"; then
+dnl    pear_error_msg="$pear_error_msg 
+dnl                    PEAR requires XML to be enabled.     Add --enable-xml to the configure line. (or --without-pear)"
+dnl  fi
+dnl
 dnl
 dnl  if test "$PHP_XMLRPC" = "no"; then
 dnl    pear_error_msg="$pear_error_msg 
diff -urNp -x '*.orig' php-5.5.38.org/sapi/cli/config.m4 php-5.5.38/sapi/cli/config.m4
--- php-5.5.38.org/sapi/cli/config.m4	2021-08-23 23:02:44.256618284 +0200
+++ php-5.5.38/sapi/cli/config.m4	2021-08-23 23:02:45.126617925 +0200
@@ -4,7 +4,7 @@ dnl
 
 PHP_ARG_ENABLE(cli,,
 [  --disable-cli           Disable building CLI version of PHP
-                          (this forces --without-pear)], yes, no)
+                          ], yes, no)
 
 AC_CHECK_FUNCS(setproctitle)
 
