--- acinclude.m4~	2005-03-17 10:10:04.000000000 +0200
+++ acinclude.m4	2005-03-17 09:57:25.000000000 +0200
@@ -1851,8 +1851,14 @@
 dnl This macro is used to get a comparable
 dnl version for apache1/2.
 dnl
+
 AC_DEFUN([PHP_AP_EXTRACT_VERSION],[
+if test -n "$with_apache_version"; then
+  ac_output="foo bar baz $with_apache_version"
+else
   ac_output=`$1 -v 2>&1`
+fi
+
   ac_IFS=$IFS
 IFS="- /.
 "
--- configure.in~	2005-03-16 22:02:44.000000000 +0200
+++ configure.in	2005-03-17 01:03:14.000000000 +0200
@@ -766,6 +766,9 @@
 ])
 AC_MSG_RESULT([$PHP_VERSIONING])
 
+PHP_ARG_WITH(apache-version,[apache version],
+[  --apache-version=MAJOR.MINOR.REVISION      Sets apache version to avoid calling apache/httpd binary], no, no)
+
 divert(5)
 
 dnl ## In diversion 5 we check which extensions should be compiled.
