--- php-4.3.0/ext/odbc/config.m4.orig	Tue Nov 12 18:31:39 2002
+++ php-4.3.0/ext/odbc/config.m4	Thu Jan  2 22:34:43 2003
@@ -539,6 +539,9 @@
     if test "$shared" != "yes"; then
       EXTRA_LDFLAGS="$EXTRA_LDFLAGS $ODBC_LFLAGS"
       EXTRA_LIBS="$EXTRA_LIBS $ODBC_LIBS"
+    else
+      ODBC_SHARED_LIBADD="$ODBC_LDFLAGS $ODBC_LIBS"
+      PHP_SUBST(ODBC_SHARED_LIBADD)
     fi
   fi
   AC_DEFINE(HAVE_UODBC,1,[ ])
