--- php-5.2.0/ext/filter/config.m4~	2006-01-14 17:04:12.000000000 +0200
+++ php-5.2.0/ext/filter/config.m4	2006-11-20 00:06:09.272529853 +0200
@@ -10,10 +10,7 @@
 if test "$PHP_FILTER" != "no"; then
 
   dnl Check if configure is the PHP core configure
-  if test -n "$PHP_VERSION"; then
-    dnl This extension can not be build as shared when in PHP core
-    ext_shared=no
-  else
+  if test -z "$PHP_VERSION"; then
     dnl This is PECL build, check if bundled PCRE library is used
     old_CPPFLAGS=$CPPFLAGS
     CPPFLAGS=$INCLUDES
