--- php-4.0.6/ext/imap/config.m4.imap	Wed Jul 11 09:21:00 2001
+++ php-4.0.6/ext/imap/config.m4	Wed Jul 11 09:28:24 2001
@@ -49,14 +49,11 @@
 
   if test "$PHP_IMAP_SSL" != "no"; then
     PHP_ADD_LIBPATH($PHP_IMAP_SSL/lib, IMAP_SHARED_LIBADD)
-    PHP_ADD_LIBRARY_DEFER(crypto)
-    PHP_ADD_LIBRARY_DEFER(ssl)
+    PHP_ADD_LIBRARY(crypto, 1, IMAP_SHARED_LIBADD)
+    PHP_ADD_LIBRARY(ssl, 1, IMAP_SHARED_LIBADD)
 
     old_LIBS=$LIBS
-    LIBS="$LIBS -lc-client"
-    if test $PHP_KERBEROS != "no"; then
-      LIBS="$LIBS -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err"
-    fi
+    LIBS="$LIBS $IMAP_SHARED_LIBADD"
     AC_TRY_RUN([
       void mm_log(void){}
       void mm_dlog(void){}
@@ -133,8 +130,7 @@
     fi
 
     PHP_ADD_INCLUDE($IMAP_INC_DIR)
-    PHP_ADD_LIBPATH($IMAP_LIBDIR, IMAP_SHARED_LIBADD)
-    PHP_ADD_LIBRARY_DEFER($IMAP_LIB)
+    PHP_ADD_LIBRARY_WITH_PATH($IMAP_LIB, $IMAP_LIBDIR, IMAP_SHARED_LIBADD)
     PHP_IMAP_KRB_CHK
     PHP_IMAP_SSL_CHK
 fi
