diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/breakiterator/breakiterator_class.cpp php-5.6.40/ext/intl/breakiterator/breakiterator_class.cpp
--- php-5.6.40.org/ext/intl/breakiterator/breakiterator_class.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/breakiterator/breakiterator_class.cpp	2021-08-23 23:20:40.309048531 +0200
@@ -38,6 +38,7 @@ extern "C" {
 }
 
 using PHP::CodePointBreakIterator;
+using icu::RuleBasedBreakIterator;
 
 /* {{{ Global variables */
 zend_class_entry *BreakIterator_ce_ptr;
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/breakiterator/breakiterator_class.h php-5.6.40/ext/intl/breakiterator/breakiterator_class.h
--- php-5.6.40.org/ext/intl/breakiterator/breakiterator_class.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/breakiterator/breakiterator_class.h	2021-08-23 23:20:40.309048531 +0200
@@ -26,6 +26,8 @@
 
 #ifndef USE_BREAKITERATOR_POINTER
 typedef void BreakIterator;
+#else
+using icu::BreakIterator;
 #endif
 
 typedef struct {
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/breakiterator/breakiterator_methods.cpp php-5.6.40/ext/intl/breakiterator/breakiterator_methods.cpp
--- php-5.6.40.org/ext/intl/breakiterator/breakiterator_methods.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/breakiterator/breakiterator_methods.cpp	2021-08-23 23:20:40.309048531 +0200
@@ -32,6 +32,8 @@ extern "C" {
 }
 
 using PHP::CodePointBreakIterator;
+using icu::BreakIterator;
+using icu::Locale;
 
 U_CFUNC PHP_METHOD(BreakIterator, __construct)
 {
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/breakiterator/codepointiterator_internal.cpp php-5.6.40/ext/intl/breakiterator/codepointiterator_internal.cpp
--- php-5.6.40.org/ext/intl/breakiterator/codepointiterator_internal.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/breakiterator/codepointiterator_internal.cpp	2021-08-23 23:20:40.309048531 +0200
@@ -30,6 +30,7 @@ typedef union {
 #define U_ALIGNMENT_OFFSET_UP(ptr) (sizeof(UAlignedMemory) - U_ALIGNMENT_OFFSET(ptr))
 
 using namespace PHP;
+using icu::UCharCharacterIterator;
 
 UOBJECT_DEFINE_RTTI_IMPLEMENTATION(CodePointBreakIterator);
 
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/breakiterator/codepointiterator_internal.h php-5.6.40/ext/intl/breakiterator/codepointiterator_internal.h
--- php-5.6.40.org/ext/intl/breakiterator/codepointiterator_internal.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/breakiterator/codepointiterator_internal.h	2021-08-23 23:20:40.312381861 +0200
@@ -18,8 +18,11 @@
 #define CODEPOINTITERATOR_INTERNAL_H
 
 #include <unicode/brkiter.h>
+#include <unicode/unistr.h>
 
-using U_ICU_NAMESPACE::BreakIterator;
+using icu::BreakIterator;
+using icu::CharacterIterator;
+using icu::UnicodeString;
 
 namespace PHP {
 
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/breakiterator/rulebasedbreakiterator_methods.cpp php-5.6.40/ext/intl/breakiterator/rulebasedbreakiterator_methods.cpp
--- php-5.6.40.org/ext/intl/breakiterator/rulebasedbreakiterator_methods.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/breakiterator/rulebasedbreakiterator_methods.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -25,6 +25,9 @@ extern "C" {
 
 #include "../intl_convertcpp.h"
 
+using icu::RuleBasedBreakIterator;
+using icu::Locale;
+
 static inline RuleBasedBreakIterator *fetch_rbbi(BreakIterator_object *bio) {
 	return (RuleBasedBreakIterator*)bio->biter;
 }
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/calendar/calendar_class.cpp php-5.6.40/ext/intl/calendar/calendar_class.cpp
--- php-5.6.40.org/ext/intl/calendar/calendar_class.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/calendar/calendar_class.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -34,6 +34,9 @@ extern "C" {
 #include <assert.h>
 }
 
+using icu::GregorianCalendar;
+using icu::Locale;
+
 /* {{{ Global variables */
 zend_class_entry *Calendar_ce_ptr;
 zend_class_entry *GregorianCalendar_ce_ptr;
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/calendar/calendar_class.h php-5.6.40/ext/intl/calendar/calendar_class.h
--- php-5.6.40.org/ext/intl/calendar/calendar_class.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/calendar/calendar_class.h	2021-08-23 23:20:40.312381861 +0200
@@ -26,6 +26,8 @@
 
 #ifndef USE_CALENDAR_POINTER
 typedef void Calendar;
+#else
+using icu::Calendar;
 #endif
 
 typedef struct {
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/calendar/calendar_methods.cpp php-5.6.40/ext/intl/calendar/calendar_methods.cpp
--- php-5.6.40.org/ext/intl/calendar/calendar_methods.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/calendar/calendar_methods.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -40,6 +40,8 @@ extern "C" {
 }
 #include "../common/common_enum.h"
 
+using icu::Locale;
+
 U_CFUNC PHP_METHOD(IntlCalendar, __construct)
 {
 	zend_throw_exception( NULL,
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/calendar/gregoriancalendar_methods.cpp php-5.6.40/ext/intl/calendar/gregoriancalendar_methods.cpp
--- php-5.6.40.org/ext/intl/calendar/gregoriancalendar_methods.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/calendar/gregoriancalendar_methods.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -32,6 +32,11 @@ extern "C" {
 #include <ext/date/php_date.h>
 }
 
+using icu::GregorianCalendar;
+using icu::Locale;
+using icu::UnicodeString;
+using icu::StringPiece;
+
 static inline GregorianCalendar *fetch_greg(Calendar_object *co) {
 	return (GregorianCalendar*)co->ucal;
 }
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/common/common_date.cpp php-5.6.40/ext/intl/common/common_date.cpp
--- php-5.6.40.org/ext/intl/common/common_date.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/common/common_date.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -17,6 +17,8 @@
 #include "../intl_cppshims.h"
 
 #include <unicode/calendar.h>
+using icu::TimeZone;
+using icu::UnicodeString;
 
 extern "C" {
 #include "../php_intl.h"
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/common/common_date.h php-5.6.40/ext/intl/common/common_date.h
--- php-5.6.40.org/ext/intl/common/common_date.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/common/common_date.h	2021-08-23 23:20:40.312381861 +0200
@@ -27,6 +27,7 @@ U_CDECL_END
 #ifdef __cplusplus
 
 #include <unicode/timezone.h>
+using icu::TimeZone;
 
 U_CFUNC TimeZone *timezone_convert_datetimezone(int type, void *object, int is_datetime, intl_error *outside_error, const char *func TSRMLS_DC);
 U_CFUNC int intl_datetime_decompose(zval *z, double *millis, TimeZone **tz,
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/common/common_enum.h php-5.6.40/ext/intl/common/common_enum.h
--- php-5.6.40.org/ext/intl/common/common_enum.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/common/common_enum.h	2021-08-23 23:20:40.312381861 +0200
@@ -69,6 +69,7 @@ U_CFUNC void zoi_with_current_get_curren
 U_CFUNC void zoi_with_current_invalidate_current(zend_object_iterator *iter TSRMLS_DC);
 
 #ifdef __cplusplus
+using icu::StringEnumeration;
 U_CFUNC void IntlIterator_from_StringEnumeration(StringEnumeration *se, zval *object TSRMLS_DC);
 #endif
 
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/dateformat/dateformat_format_object.cpp php-5.6.40/ext/intl/dateformat/dateformat_format_object.cpp
--- php-5.6.40.org/ext/intl/dateformat/dateformat_format_object.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/dateformat/dateformat_format_object.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -33,6 +33,12 @@ extern "C" {
 #include "../common/common_date.h"
 }
 
+using icu::Locale;
+using icu::DateFormat;
+using icu::GregorianCalendar;
+using icu::StringPiece;
+using icu::SimpleDateFormat;
+
 static const DateFormat::EStyle valid_styles[] = {
 		DateFormat::kNone,
 		DateFormat::kFull,
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/dateformat/dateformat_helpers.cpp php-5.6.40/ext/intl/dateformat/dateformat_helpers.cpp
--- php-5.6.40.org/ext/intl/dateformat/dateformat_helpers.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/dateformat/dateformat_helpers.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -28,6 +28,8 @@ extern "C" {
 #include "../calendar/calendar_class.h"
 }
 
+using icu::GregorianCalendar;
+
 int datefmt_process_calendar_arg(zval* calendar_zv,
 								 Locale const& locale,
 								 const char *func_name,
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/dateformat/dateformat_helpers.h php-5.6.40/ext/intl/dateformat/dateformat_helpers.h
--- php-5.6.40.org/ext/intl/dateformat/dateformat_helpers.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/dateformat/dateformat_helpers.h	2021-08-23 23:20:40.312381861 +0200
@@ -23,10 +23,17 @@
 
 #include <unicode/calendar.h>
 
+#include <unicode/datefmt.h>
+
 extern "C" {
 #include "../php_intl.h"
 }
 
+using icu::Locale;
+using icu::Calendar;
+
+using icu::DateFormat;
+
 int datefmt_process_calendar_arg(zval* calendar_zv,
 								 Locale const& locale,
 								 const char *func_name,
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/intl_convertcpp.h php-5.6.40/ext/intl/intl_convertcpp.h
--- php-5.6.40.org/ext/intl/intl_convertcpp.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/intl_convertcpp.h	2021-08-23 23:20:40.312381861 +0200
@@ -24,6 +24,7 @@
 #endif
 
 #include <unicode/unistr.h>
+using icu::UnicodeString;
 
 int intl_stringFromChar(UnicodeString &ret, char *str, int32_t str_len, UErrorCode *status);
 
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/msgformat/msgformat_helpers.cpp php-5.6.40/ext/intl/msgformat/msgformat_helpers.cpp
--- php-5.6.40.org/ext/intl/msgformat/msgformat_helpers.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/msgformat/msgformat_helpers.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -27,6 +27,7 @@
 #include <unicode/timezone.h>
 #include <unicode/datefmt.h>
 #include <unicode/calendar.h>
+#include <unicode/strenum.h>
 
 #include <vector>
 
@@ -76,6 +77,16 @@ MessageFormatAdapter::getMessagePattern(
 #endif
 U_NAMESPACE_END
 
+using icu::Formattable;
+using icu::Format;
+using icu::DateFormat;
+using icu::MessageFormat;
+#ifdef HAS_MESSAGE_PATTERN
+using icu::MessagePattern;
+#endif
+using icu::MessageFormatAdapter;
+using icu::FieldPosition;
+
 U_CFUNC int32_t umsg_format_arg_count(UMessageFormat *fmt)
 {
 	int32_t fmt_count = 0;
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/timezone/timezone_class.cpp php-5.6.40/ext/intl/timezone/timezone_class.cpp
--- php-5.6.40.org/ext/intl/timezone/timezone_class.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/timezone/timezone_class.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -37,6 +37,8 @@ extern "C" {
 #include <ext/date/php_date.h>
 }
 
+using icu::Calendar;
+
 /* {{{ Global variables */
 U_CDECL_BEGIN
 zend_class_entry *TimeZone_ce_ptr = NULL;
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/timezone/timezone_class.h php-5.6.40/ext/intl/timezone/timezone_class.h
--- php-5.6.40.org/ext/intl/timezone/timezone_class.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/timezone/timezone_class.h	2021-08-23 23:20:40.312381861 +0200
@@ -29,6 +29,8 @@
 
 #ifndef USE_TIMEZONE_POINTER
 typedef void TimeZone;
+#else
+using icu::TimeZone;
 #endif
 
 typedef struct {
diff -urNp -x '*.orig' php-5.6.40.org/ext/intl/timezone/timezone_methods.cpp php-5.6.40/ext/intl/timezone/timezone_methods.cpp
--- php-5.6.40.org/ext/intl/timezone/timezone_methods.cpp	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/ext/intl/timezone/timezone_methods.cpp	2021-08-23 23:20:40.312381861 +0200
@@ -23,6 +23,7 @@
 #include <unicode/locid.h>
 #include <unicode/timezone.h>
 #include <unicode/ustring.h>
+#include <unicode/calendar.h>
 #include "intl_convertcpp.h"
 
 #include "../common/common_date.h"
@@ -37,6 +38,9 @@ extern "C" {
 }
 #include "common/common_enum.h"
 
+using icu::Locale;
+using icu::Calendar;
+
 U_CFUNC PHP_METHOD(IntlTimeZone, __construct)
 {
 	zend_throw_exception( NULL,
