diff -ur php-4.3.7-orig/ext/mnogosearch/config.m4 php-4.3.7-patched/ext/mnogosearch/config.m4
--- php-4.3.7-orig/ext/mnogosearch/config.m4	2003-10-03 05:25:38.000000000 +0000
+++ php-4.3.7-patched/ext/mnogosearch/config.m4	2004-06-09 21:32:04.000000000 +0000
@@ -46,8 +46,10 @@
     PHP_ADD_INCLUDE($MNOGOSEARCH_INCDIR)
     
     AC_DEFINE(HAVE_MNOGOSEARCH,1,[ ])
+    MNOGOSEARCH_CFLAGS="-DNOT_USED_OR_REPLACED"
     
     PHP_SUBST(MNOGOSEARCH_SHARED_LIBADD)
-    PHP_NEW_EXTENSION(mnogosearch, php_mnogo.c, $ext_shared)
+    PHP_SUBST(MNOGOSEARCH_CFLAGS)
+    PHP_NEW_EXTENSION(mnogosearch, php_mnogo.c, $ext_shared,, $MNOGOSEARCH_CFLAGS)
     
   fi
