--- php4-4.0.4final.orig/ext/gd/config.m4
+++ php4-4.0.4final/ext/gd/config.m4
@@ -16,7 +16,6 @@
           else
             AC_ADD_LIBRARY_WITH_PATH(jpeg, $withval/lib)
           fi
-          LIBS="$LIBS -L$withval/lib -ljpeg"
         ],[
           AC_MSG_RESULT(no)
           AC_MSG_WARN(If configure fails try --with-jpeg-dir=<DIR>)
@@ -36,9 +35,13 @@
           LIBS="$LIBS -L$withval/lib -lX11"
           AC_CHECK_LIB(Xpm,XpmFreeXpmImage, [LIBS="$LIBS -L$withval/lib -lXpm"],[AC_MSG_RESULT(no)],)
           LIBS=$old_LIBS
-          AC_ADD_LIBRARY_WITH_PATH(Xpm, $withval/lib)
-          AC_ADD_LIBRARY_WITH_PATH(X11, $withval/lib)
-          LIBS="$LIBS -L$withval/lib -lXpm -L$withval/lib -lX11"
+	  if test "$shared" = "yes"; then
+	    GD_LIBS="$GD_LIBS -lXpm -lX11"
+	    GD_LFLAGS="$GD_LFLAGS -L$withval/lib"
+	  else
+            AC_ADD_LIBRARY_WITH_PATH(Xpm, $withval/lib)
+            AC_ADD_LIBRARY_WITH_PATH(X11, $withval/lib)
+	  fi
         ],[
           AC_MSG_RESULT(no)
           AC_MSG_WARN(If configure fails try --with-xpm-dir=<DIR>)
@@ -112,10 +115,11 @@
       LIBS=$old_LIBS
       LDFLAGS=$old_LDFLAGS
       if test "$ac_cv_lib_gd_gdImageCreateFromPng" = "yes"; then
-        AC_ADD_LIBRARY(png)
-        AC_ADD_LIBRARY(z)
         if test "$shared" = "yes"; then
           GD_LIBS="$GD_LIBS -lpng -lz"
+	else
+          AC_ADD_LIBRARY(png)
+          AC_ADD_LIBRARY(z)
         fi
       fi
 
@@ -165,10 +169,11 @@
         LIBS=$old_LIBS
         LDFLAGS=$old_LDFLAGS
         if test "$ac_cv_lib_gd_gdImageCreateFromPng" = "yes"; then
-          AC_ADD_LIBRARY(z)
-          AC_ADD_LIBRARY(png)
           if test "$shared" = "yes"; then
             GD_LIBS="$GD_LIBS -lpng -lz"
+	  else
+            AC_ADD_LIBRARY(z)
+            AC_ADD_LIBRARY(png)
           fi
         fi
 
@@ -249,7 +254,7 @@
         else
           AC_ADD_LIBRARY_WITH_PATH(t1, "$T1_DIR/lib")
         fi
-        LIBS="$LIBS -L$T1_DIR/lib -lt1"
+#        LIBS="$LIBS -L$T1_DIR/lib -lt1"
       fi 
 
       AC_MSG_RESULT(yes)

