diff -urN php-4.2.1RC2.org/ext/mysql/config.m4 php-4.2.1RC2/ext/mysql/config.m4
--- php-4.2.1RC2.org/ext/mysql/config.m4	Sat May 11 20:49:23 2002
+++ php-4.2.1RC2/ext/mysql/config.m4	Sat May 11 20:55:42 2002
@@ -18,6 +18,14 @@
 ])
 	
 AC_DEFUN(PHP_MYSQL_SOCK,[
+
+  AC_ARG_WITH(mysql_sock,
+  [  --with-mysql-socket=PATH
+  				Specify location of MySQL UNIX socket, 
+				defaults search in know locations],[
+  MYSQL_SOCK=$withval
+  ],)
+  if test "$MYSQL_SOCK" = "" -o "$MYSQL_SOCK" = "no" -o "MYSQL_SOCK" = "yes"; then
   AC_MSG_CHECKING(for MySQL UNIX socket)
   MYSQL_SOCK=/tmp/mysql.sock
   for i in  \
@@ -32,6 +40,7 @@
       MYSQL_SOCK=$i
     fi
   done
+  fi
   AC_DEFINE_UNQUOTED(MYSQL_UNIX_ADDR, "$MYSQL_SOCK", [ ])
   AC_MSG_RESULT($MYSQL_SOCK)
 ])
