--- php-5.2.6/sapi/cgi/fpm/conf/php-fpm.conf.in~	2008-11-16 17:28:21.256858344 +0200
+++ php-5.2.6/sapi/cgi/fpm/conf/php-fpm.conf.in	2008-11-16 17:30:12.321010353 +0200
@@ -38,7 +38,7 @@
 
 			Address to accept fastcgi requests on.
 			Valid syntax is 'ip.ad.re.ss:port' or just 'port' or '/path/to/unix/socket'
-			<value name="listen_address">127.0.0.1:9000</value>
+			<value name="listen_address">/var/run/php/fpm.sock</value>
 
 			<value name="listen_options">
 
@@ -60,10 +60,10 @@
 			</value>
 
 			Unix user of processes
-		<!--	<value name="user">nobody</value>				-->
+			<value name="user">http</value>
 
 			Unix group of processes
-		<!--	<value name="group">@php_fpm_group@</value>		-->
+			<value name="group">http</value>
 
 			Process manager settings
 			<value name="pm">
