--- php-4.3.2/ext/dba/config.m4.orig	Wed May 21 02:35:29 2003
+++ php-4.3.2/ext/dba/config.m4	Mon Jun 23 11:53:32 2003
@@ -176,13 +176,13 @@
     AC_MSG_CHECKING(if db can be used as shared extension)
     AC_EGREP_CPP(yes,[
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR > 3 || (DB_VERSION_MAJOR == 3 && DB_VERSION_MINOR > 2)
+#if DB_VERSION_MAJOR > 3 || (DB_VERSION_MAJOR == 3 && DB_VERSION_MINOR > 0)
       yes
 #endif
     ],[
       AC_MSG_RESULT(yes)
     ],[
-      AC_MSG_ERROR(At least version 3.3 is required)
+      AC_MSG_ERROR(At least version 3.1 is required)
     ])
   fi
   if test -n "$THIS_LIBS"; then
