--- PHP_5_3/main/php_open_temporary_file.c	2011/03/28 16:34:07	309791
+++ PHP_5_3/main/php_open_temporary_file.c	2011/03/28 16:43:49	309792
@@ -204,9 +204,13 @@
 	 */
 	{
 		char sTemp[MAX_PATH];
-		DWORD n = GetTempPath(sizeof(sTemp),sTemp);
-		assert(0 < n);  /* should *never* fail! */
-		temporary_directory = strdup(sTemp);
+		DWORD len = GetTempPath(sizeof(sTemp),sTemp);
+		assert(0 < len);  /* should *never* fail! */
+		if (sTemp[len - 1] == DEFAULT_SLASH) {
+			temporary_directory = zend_strndup(sTemp, len - 1);
+		} else {
+			temporary_directory = zend_strndup(sTemp, len);
+		}
 		return temporary_directory;
 	}
 #else
