--- php-4.0RC2/Zend/zend_operators.c.wiget	Thu May 11 09:56:16 2000
+++ php-4.0RC2/Zend/zend_operators.c	Thu May 11 09:58:43 2000
@@ -1341,7 +1341,7 @@
 	
 	if ((ret1=is_numeric_string(s1->value.str.val, s1->value.str.len, &lval1, &dval1)) &&
 		(ret2=is_numeric_string(s2->value.str.val, s2->value.str.len, &lval2, &dval2))) {
-#if WITH_BCMATH
+#if 0
 		if ((ret1==FLAG_IS_BC) || (ret2==FLAG_IS_BC)) {
 			bc_num first, second;
 			
