--- php-5.2.0/ext/gd/config.m4.orig	2005-11-29 19:25:59.000000000 +0100
+++ php-5.2.0/ext/gd/config.m4	2006-12-22 23:41:12.363074440 +0100
@@ -261,6 +261,7 @@
   PHP_CHECK_LIBRARY(gd, gdFontCacheShutdown,    [AC_DEFINE(HAVE_GD_FONTCACHESHUTDOWN,1, [ ])], [], [ -L$GD_LIB $GD_SHARED_LIBADD ])
   PHP_CHECK_LIBRARY(gd, gdFreeFontCache,        [AC_DEFINE(HAVE_GD_FREEFONTCACHE,    1, [ ])], [], [ -L$GD_LIB $GD_SHARED_LIBADD ])
   PHP_CHECK_LIBRARY(gd, gdNewDynamicCtxEx,      [AC_DEFINE(HAVE_GD_DYNAMIC_CTX_EX,   1, [ ])], [], [ -L$GD_LIB $GD_SHARED_LIBADD ])
+  PHP_CHECK_LIBRARY(gd, gdImageRotate,          [AC_DEFINE(HAVE_GD_ROTATE,           1, [ ])], [], [ -L$GD_LIB $GD_SHARED_LIBADD ])
 ])
 
 dnl
@@ -313,6 +314,7 @@
   AC_DEFINE(HAVE_GD_FONTCACHESHUTDOWN,1, [ ])
   AC_DEFINE(HAVE_GD_DYNAMIC_CTX_EX,   1, [ ])
   AC_DEFINE(HAVE_GD_GIF_CTX,          1, [ ])
+  AC_DEFINE(HAVE_GD_ROTATE,           1, [ ])
 
 dnl Make sure the libgd/ is first in the include path
   GDLIB_CFLAGS="-DHAVE_LIBPNG"
--- php-5.2.0/ext/gd/gd.c.orig	2006-10-18 18:04:37.000000000 +0200
+++ php-5.2.0/ext/gd/gd.c	2006-12-22 23:45:04.600308887 +0100
@@ -314,7 +314,7 @@
 ZEND_END_ARG_INFO()
 #endif
 
-#ifdef HAVE_GD_BUNDLED
+#ifdef HAVE_GD_ROTATE
 static
 ZEND_BEGIN_ARG_INFO_EX(arginfo_imagerotate, 0, 0, 3)
 	ZEND_ARG_INFO(0, im)
@@ -1020,8 +1020,10 @@
 	PHP_FE(imagecopyresampled,						arginfo_imagecopyresampled)
 #endif
 
-#ifdef HAVE_GD_BUNDLED
+#ifdef HAVE_GD_ROTATE
 	PHP_FE(imagerotate,     						arginfo_imagerotate)
+#endif
+#ifdef HAVE_GD_BUNDLED
 	PHP_FE(imageantialias,							arginfo_imageantialias)
 #endif
 
@@ -1248,11 +1250,13 @@
 #ifdef GD2_FMT_COMPRESSED
 	REGISTER_LONG_CONSTANT("IMG_GD2_COMPRESSED", GD2_FMT_COMPRESSED, CONST_CS | CONST_PERSISTENT);
 #endif
-#if HAVE_GD_BUNDLED
+#ifdef HAVE_GD_ROTATE
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_REPLACE", gdEffectReplace, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_ALPHABLEND", gdEffectAlphaBlend, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_NORMAL", gdEffectNormal, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMG_EFFECT_OVERLAY", gdEffectOverlay, CONST_CS | CONST_PERSISTENT);
+#endif
+#if HAVE_GD_BUNDLED
 	REGISTER_LONG_CONSTANT("GD_BUNDLED", 1, CONST_CS | CONST_PERSISTENT);
 	/* Section Filters */
 	REGISTER_LONG_CONSTANT("IMG_FILTER_NEGATE", IMAGE_FILTER_NEGATE, CONST_CS | CONST_PERSISTENT);
@@ -2055,7 +2059,7 @@
 /* }}} */
 #endif
 
-#ifdef HAVE_GD_BUNDLED
+#ifdef HAVE_GD_ROTATE
 /* {{{ proto resource imagerotate(resource src_im, float angle, int bgdcolor [, int ignoretransparent])
    Rotate an image using a custom angle */
 PHP_FUNCTION(imagerotate)
--- php-5.2.0/ext/gd/php_gd.h.orig	2006-01-01 13:50:06.000000000 +0100
+++ php-5.2.0/ext/gd/php_gd.h	2006-12-22 23:42:04.138024927 +0100
@@ -112,8 +112,10 @@
 PHP_FUNCTION(imagecopyresampled);
 #endif
 
-#ifdef HAVE_GD_BUNDLED
+#ifdef HAVE_GD_ROTATE
 PHP_FUNCTION(imagerotate);
+#endif
+#ifdef HAVE_GD_BUNDLED
 PHP_FUNCTION(imageantialias);
 #endif
 
