diff -burN php-4.3.10.orig/Zend/zend_strtod.c php-4.3.10/Zend/zend_strtod.c
--- php-4.3.10.orig/Zend/zend_strtod.c	2004-12-28 20:08:49.616697784 +0100
+++ php-4.3.10/Zend/zend_strtod.c	2004-12-28 20:09:17.572447864 +0100
@@ -90,6 +90,7 @@
  */
 
 #include <zend_strtod.h>
+#include <stdint.h>
 
 #if defined(LIBC_SCCS) && !defined(lint)
 static char *rcsid = "$OpenBSD: strtod.c,v 1.19 2004/02/03 16:52:11 drahn Exp $";
