--- php-5.0.4/ext/cpdf/config.m4.orig	2004-12-30 08:04:10.000000000 +0100
+++ php-5.0.4/ext/cpdf/config.m4	2005-04-04 22:44:23.169848344 +0200
@@ -49,6 +49,7 @@
 
 if test "$PHP_CPDFLIB" != "no"; then
   PHP_NEW_EXTENSION(cpdf, cpdf.c, $ext_shared,, \\$(GDLIB_CFLAGS))
+  PHP_ADD_LIBRARY_WITH_PATH(gd, /usr/lib, CPDF_SHARED_LIBADD)
   PHP_SUBST(CPDF_SHARED_LIBADD)
   CPDF_JPEG_TEST
   CPDF_TIFF_TEST
