--- php-4.0.6/sapi/fastcgi/config.m4.wiget	Tue Mar 27 22:34:45 2001
+++ php-4.0.6/sapi/fastcgi/config.m4	Wed Jul  4 23:14:00 2001
@@ -7,10 +7,15 @@
         FASTCGIPATH=$withval
   fi
   test -f "$FASTCGIPATH/lib/libfcgi.a" || AC_MSG_ERROR(Unable to find libfcgi.a in $FASTCGIPATH/lib)
-  test -f "$FASTCGIPATH/include/fastcgi.h" || AC_MSG_ERROR(Unable to find fastcgi.h in $FASTCGIPATH/include)
+  if test -f "$FASTCGIPATH/include/fastcgi.h" ; then 
+  	PHP_ADD_INCLUDE($FASTCGIPATH/include)
+  elif test -f "$FASTCGIPATH/include/fastcgi/fastcgi.h" ; then 
+  	PHP_ADD_INCLUDE($FASTCGIPATH/fastcgi/include)
+  else
+	AC_MSG_ERROR(Unable to find fastcgi.h in $FASTCGIPATH/include)
+  fi
   PHP_SAPI=fastcgi
   PHP_LIBS=$FASTCGIPATH/lib/libfcgi.a
-  PHP_ADD_INCLUDE($FASTCGIPATH/include)
   EXT_PROGRAM_LDADD="$EXT_PROGRAM_LDADD $FASTCGIPATH/lib/libfcgi.a"
   INSTALL_IT="\$(INSTALL) -m 0755 $SAPI_PROGRAM \$(bindir)/$SAPI_FASTCGI"
   RESULT="yes"
--- php-4.0.6/sapi/fastcgi/fastcgi.c.wiget	Fri May  4 11:49:14 2001
+++ php-4.0.6/sapi/fastcgi/fastcgi.c	Wed Jul  4 23:09:36 2001
@@ -200,6 +200,11 @@
 	struct stat st;
 	char *pi = getenv( "PATH_INFO" );
 	char *pt = getenv( "PATH_TRANSLATED" );
+
+	if (!pi)
+		pi = getenv( "SCRIPT_NAME" );
+	if (!pt)
+		pt = getenv( "SCRIPT_FILENAME" );
 	path_info = strdup( pi );
 
 	SG(request_info).request_method = getenv("REQUEST_METHOD");
