diff -burN php-4.3.2/ext/java/config.m4 php-4.3.2-java-norpath/ext/java/config.m4
--- php-4.3.2/ext/java/config.m4	Fri Apr  4 19:43:13 2003
+++ php-4.3.2-java-norpath/ext/java/config.m4	Wed Jun 11 09:38:23 2003
@@ -58,7 +58,6 @@
 AC_DEFUN(JAVA_CHECK_LIB, [
  AC_MSG_CHECKING([Checking for libjava])
  if test -d $PHP_JAVA/lib/kaffe; then
-   PHP_ADD_LIBPATH($PHP_JAVA/lib)
    JAVA_CFLAGS=-DKAFFE
    JAVA_INCLUDE=-I$PHP_JAVA/include/kaffe
    JAVA_CLASSPATH=$PHP_JAVA/share/kaffe/Klasses.jar
@@ -111,10 +110,6 @@
        JAVA_LIB=java
        JAVA_LIBPATH=$i
 
-       test -d $i/hotspot && PHP_ADD_LIBPATH($i/hotspot)
-       test -d $i/classic && PHP_ADD_LIBPATH($i/classic)
-       test -d $i/server  && PHP_ADD_LIBPATH($i/server)
-       test -d $i/native_threads && PHP_ADD_LIBPATH($i/native_threads)
      fi
    done
 
@@ -160,19 +155,12 @@
       AC_MSG_ERROR([unable to find Java VM libraries in $PHP_JAVA])
     fi
 
-    PHP_ADD_LIBPATH($JAVA_LIBPATH)
     JAVA_CFLAGS="$JAVA_CFLAGS '-DJAVALIB=\"$JAVA_LIBPATH/$java_libext\"'"
   fi
 
   if test "$PHP_SAPI" != "servlet"; then
     PHP_NEW_EXTENSION(java, java.c, shared,, $JAVA_CFLAGS $JAVA_INCLUDE)
 
-    if test "$PHP_SAPI" = "cgi"; then
-      if test "$platform" != "Darwin"; then
-        PHP_ADD_LIBRARY($JAVA_LIB)
-      fi
-    fi
-
     if test -n "$INSTALL_IT"; then
       INSTALL_IT="$INSTALL_IT ;"
     fi
