--- ./ext/gmp/config.m4.org	Mon Jun 24 09:55:32 2002
+++ ./ext/gmp/config.m4	Sat Sep  7 11:02:51 2002
@@ -23,3 +23,4 @@
   PHP_SUBST(GMP_SHARED_LIBADD)
   PHP_ADD_LIBRARY_WITH_PATH(gmp, $GMP_DIR/lib, GMP_SHARED_LIBADD)
 fi
+PHP_SUBST(GMP_SHARED_LIBADD)
--- ./ext/xslt/config.m4.org	Thu May  2 12:24:03 2002
+++ ./ext/xslt/config.m4	Sat Sep  7 11:03:45 2002
@@ -81,7 +81,7 @@
       found_js=no
       AC_CHECK_LIB(js, JS_GetRuntime, found_js=yes)
       if test "$found_js" = "yes"; then
-        PHP_ADD_LIBRARY(js)
+        PHP_ADD_LIBRARY(js,, XSLT_SHARED_LIBADD)
       fi
     fi
 
--- ./ext/yp/Makefile.in.org	Tue May  2 03:21:26 2000
+++ ./ext/yp/Makefile.in	Sat Sep  7 11:05:11 2002
@@ -2,5 +2,6 @@
 LTLIBRARY_NAME    = libyp.la
 LTLIBRARY_SOURCES = yp.c
 LTLIBRARY_SHARED_NAME = yp.la
+LTLIBRARY_SHARED_LIBADD = $(NSL_SHARED_LIBADD)
 
 include $(top_srcdir)/build/dynlib.mk
--- ./ext/yp/config.m4.org	Fri Nov 30 20:00:10 2001
+++ ./ext/yp/config.m4	Sat Sep  7 11:06:31 2002
@@ -6,8 +6,11 @@
 [  --enable-yp             Include YP support.])
 
 if test "$PHP_YP" != "no"; then
-  AC_DEFINE(HAVE_YP,1,[ ])
+  PHP_SUBST(NSL_SHARED_LIBADD)
+  AC_ADD_LIBRARY_WITH_PATH(nsl, /usr/lib, NSL_SHARED_LIBADD)
+  AC_CHECK_LIB(nsl, yp_match, [AC_DEFINE(HAVE_YP,1,[ ])], [AC_MSG_ERROR(YP module requires libnsl)],)
   PHP_EXTENSION(yp, $ext_shared)
+
   case $host_alias in
   *solaris*)
     AC_DEFINE(SOLARIS_YP,1,[ ]) ;;
--- ./configure.in.org	Fri Sep  6 01:20:49 2002
+++ ./configure.in	Sat Sep  7 11:01:54 2002
@@ -292,6 +292,7 @@
 
 if test "$php_no_nsl_checks" != "yes"; then
 
+AC_CHECK_FUNC(gethostname,[true], [
 AC_CHECK_LIB(nsl, gethostname, [
  PHP_ADD_LIBRARY(nsl)
  AC_DEFINE(HAVE_LIBNSL,1,[ ]) ],[
@@ -299,6 +300,7 @@
   PHP_ADD_LIBRARY(nsl)
   AC_DEFINE(HAVE_LIBNSL,1,[ ]) ], [])
 ])
+])
 
 fi
 
