--- php-4.0.6/ext/domxml/config.m4~	Thu Nov 15 00:32:45 2001
+++ php-4.0.6/ext/domxml/config.m4	Thu Nov 15 00:32:45 2001
@@ -3,7 +3,7 @@
 
 AC_DEFUN(PHP_DOM_CHECK_VERSION,[
   old_CPPFLAGS=$CPPFLAGS
-  CPPFLAGS=-I$DOMXML_DIR/include
+  CPPFLAGS=-I$DOMXML_DIR/include/libxml2
   AC_MSG_CHECKING(for libxml version)
   AC_EGREP_CPP(yes,[
   #include <libxml/xmlversion.h>
@@ -25,12 +25,12 @@
 
 if test "$PHP_DOM" != "no"; then
 
-  if test -r $PHP_DOM/include/libxml/tree.h; then
+  if test -r $PHP_DOM/include/libxml2/libxml/tree.h; then
     DOMXML_DIR=$PHP_DOM
   else
     AC_MSG_CHECKING(for DOM in default path)
     for i in /usr/local /usr; do
-      if test -r $i/include/libxml/tree.h; then
+      if test -r $i/include/libxml2/libxml/tree.h; then
         DOMXML_DIR=$i
         AC_MSG_RESULT(found in $i)
       fi
--- php-4.0.6/ext/domxml/Makefile.in~	Thu Nov 15 01:20:18 2001
+++ php-4.0.6/ext/domxml/Makefile.in	Thu Nov 15 01:20:18 2001
@@ -5,4 +5,6 @@
 LTLIBRARY_SHARED_NAME = domxml.la
 LTLIBRARY_SHARED_LIBADD  = $(DOMXML_SHARED_LIBADD)
 
+EXTRA_CFLAGS = `pkg-config libxml-2.0 --cflags`
+
 include $(top_srcdir)/build/dynlib.mk
