--- php-5.4.7/sapi/fpm/php-fpm.conf.in	2012-10-05 22:16:44.895998479 +0300
+++ php-5.3.17/sapi/fpm/php-fpm.conf.in	2012-10-12 12:35:47.601602945 +0300
@@ -22,14 +22,14 @@
 ; Pid file
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: none
-;pid = run/php-fpm.pid
+pid = /var/run/@processname@.pid
 
 ; Error log file
 ; If it's set to "syslog", log is sent to syslogd instead of being written
 ; in a local file.
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: log/php-fpm.log
-;error_log = log/php-fpm.log
+error_log = /var/log/php/@processname@.log
 
 ; syslog_facility is used to specify what type of program is logging the
 ; message. This lets syslogd specify that messages from different facilities
--- php-5.5.25/sapi/fpm/php-fpm.conf-d.in~	2015-05-22 07:21:41.000000000 +0300
+++ php-5.5.25/sapi/fpm/php-fpm.conf-d.in	2015-05-22 07:24:22.913173238 +0300
@@ -35,7 +35,8 @@
 ;                            (IPv6 and IPv4-mapped) on a specific port;
 ;   '/path/to/unix/socket' - to listen on a unix socket.
 ; Note: This value is mandatory.
-listen = 127.0.0.1:9000
+;listen = 127.0.0.1:9000
+listen = /var/run/php/@processname@.sock
 
 ; Set listen(2) backlog.
 ; Default Value: 65535 (-1 on FreeBSD and OpenBSD)
@@ -56,7 +57,7 @@
 ; must be separated by a comma. If this value is left blank, connections will be
 ; accepted from any ip address.
 ; Default Value: any
-;listen.allowed_clients = 127.0.0.1
+listen.allowed_clients = 127.0.0.1
 
 ; Specify the nice(2) priority to apply to the pool processes (only if set)
 ; The value can vary from -19 (highest priority) to 20 (lower priority)
@@ -375,7 +376,7 @@
 ; the current environment.
 ; Default Value: clean env
 ;env[HOSTNAME] = $HOSTNAME
-;env[PATH] = /usr/local/bin:/usr/bin:/bin
+env[PATH] = $PATH
 ;env[TMP] = /tmp
 ;env[TMPDIR] = /tmp
 ;env[TEMP] = /tmp
--- php-5.5.25/sapi/fpm/Makefile.frag~	2015-05-22 07:35:13.000000000 +0300
+++ php-5.5.25/sapi/fpm/Makefile.frag	2015-05-22 07:35:35.001377912 +0300
@@ -12,8 +12,8 @@
 
 	@echo "Installing PHP FPM config:        $(INSTALL_ROOT)$(sysconfdir)/" && \
 	$(mkinstalldirs) $(INSTALL_ROOT)$(sysconfdir)/fpm.d || :
-	@$(INSTALL_DATA) sapi/fpm/php-fpm.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.conf.default || :
-	@$(INSTALL_DATA) sapi/fpm/php-fpm.conf-d $(INSTALL_ROOT)$(sysconfdir)/fpm.d/www.conf.default || :
+	@$(INSTALL_DATA) sapi/fpm/php-fpm.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.conf || :
+	@$(INSTALL_DATA) sapi/fpm/php-fpm.conf-d $(INSTALL_ROOT)$(sysconfdir)/fpm.d/www.conf || :
 
 	@echo "Installing PHP FPM man page:      $(INSTALL_ROOT)$(mandir)/man8/"
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(mandir)/man8
--- php-5.3.9/sapi/fpm/status.html.in~	2011-10-08 23:43:12.000000000 +0300
+++ php-5.3.9/sapi/fpm/status.html.in	2012-01-12 02:19:09.573191879 +0200
@@ -96,7 +96,7 @@
 			var sort_index;
 			var sort_order;
 
-			doc_url.value = location.protocol + '//' + location.host + "/status?json&full";
+			doc_url.value = location.protocol + '//' + location.host + "/fpm-status?json&full";
 
 			ths = document.getElementsByTagName("th");
 			for (var i=0; i<ths.length; i++) {
