--- PHP_5_3/ext/readline/readline.c	2011/01/10 17:34:26	307342
+++ PHP_5_3/ext/readline/readline.c	2011/01/10 18:19:02	307343
@@ -196,7 +196,7 @@
 	int prompt_len;
 	char *result;
 
-	if (FAILURE == zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s!", &prompt, &prompt_len)) {
+	if (FAILURE == zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "|s!", &prompt, &prompt_len)) {
 		RETURN_FALSE;
 	}
 
