diff -ur php-4.3.4/sapi/apache/mod_php4.c php-4.3.5RC2/sapi/apache/mod_php4.c
--- php-4.3.4/sapi/apache/mod_php4.c	Tue Jun  3 07:41:49 2003
+++ php-4.3.5RC2/sapi/apache/mod_php4.c	Wed Jan 28 18:42:20 2004
@@ -830,6 +830,9 @@
 	}
 	if(!AP(xbithack)) {
 		r->allowed |= (1 << METHODS) - 1;
+		zend_try {
+			zend_ini_deactivate(TSRMLS_C);
+		} zend_end_try();
 		return DECLINED;
 	}
 	return send_parsed_php(r);
