diff -burN php-4.3.2/main/SAPI.c php-4.3.2-dl_zlib/main/SAPI.c
--- php-4.3.2/main/SAPI.c	Wed Feb 12 00:30:13 2003
+++ php-4.3.2-dl_zlib/main/SAPI.c	Wed Jun 11 09:41:47 2003
@@ -556,7 +556,7 @@
 				while (*ptr == ' ' && *ptr != '\0') {
 					ptr++;
 				}
-#if HAVE_ZLIB
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 				if(!strncmp(ptr, "image/", sizeof("image/")-1)) {
 					zend_alter_ini_entry("zlib.output_compression", sizeof("zlib.output_compression"), "0", sizeof("0") - 1, PHP_INI_USER, PHP_INI_STAGE_RUNTIME);
 				}
@@ -704,7 +704,7 @@
 		return SUCCESS;
 	}
 
-#if HAVE_ZLIB
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 	/* Add output compression headers at this late stage in order to make
 	   it possible to switch it off inside the script. */
 
