diff -urN php-4.3.4RC2.orig/ext/ncurses/config.m4 php-4.3.4RC2/ext/ncurses/config.m4
--- php-4.3.4RC2.orig/ext/ncurses/config.m4	2003-10-22 11:15:33.000000000 +0200
+++ php-4.3.4RC2/ext/ncurses/config.m4	2003-10-22 13:11:00.000000000 +0200
@@ -11,24 +11,29 @@
 
    for i in $SEARCH_PATH ; do
      if test -d $i/include; then
+       NCURSES_DIR=$i
        if test -r $i/include/ncurses.h; then
-         NCURSES_DIR=$i
+	 NCURSES_INC=$i/include
+         AC_DEFINE(HAVE_NCURSES_H,1,[ ])
+         break
+       elif test -r $i/include/ncurses/ncurses.h; then
+	 NCURSES_INC=$i/include/ncurses;
          AC_DEFINE(HAVE_NCURSES_H,1,[ ])
          break
        elif test -r $i/include/curses.h; then
-         NCURSES_DIR=$i
+         NCURSES_INC=$i/include
          break
        fi
      fi
    done
   
-   if test -z "$NCURSES_DIR"; then
+   if test -z "$NCURSES_INC"; then
      AC_MSG_RESULT(not found)
      AC_MSG_ERROR(Please reinstall the ncurses distribution)
    fi
 
    # --with-ncurses -> add include path
-   PHP_ADD_INCLUDE($NCURSES_DIR/include)
+   PHP_ADD_INCLUDE($NCURSES_INC)
 
    # --with-ncurses -> chech for lib and symbol presence
    LIBNAME=ncurses 
diff -urN php-4.3.4RC2.orig/ext/readline/config.m4 php-4.3.4RC2/ext/readline/config.m4
--- php-4.3.4RC2.orig/ext/readline/config.m4	2003-10-22 11:15:34.000000000 +0200
+++ php-4.3.4RC2/ext/readline/config.m4	2003-10-22 13:12:04.000000000 +0200
@@ -19,6 +19,10 @@
 
   PHP_ADD_INCLUDE($READLINE_DIR/include)
 
+  AC_CHECK_LIB(tinfo, tgetent,
+  [
+    PHP_ADD_LIBRARY(tinfo,,READLINE_SHARED_LIBADD)
+  ],[
   AC_CHECK_LIB(ncurses, tgetent,
   [
     PHP_ADD_LIBRARY(ncurses,,READLINE_SHARED_LIBADD)
@@ -28,6 +32,7 @@
       PHP_ADD_LIBRARY(termcap,,READLINE_SHARED_LIBADD)
     ])
   ])
+  ])
 
   PHP_CHECK_LIBRARY(readline, readline,
   [
@@ -63,6 +68,10 @@
 
   PHP_ADD_INCLUDE($LIBEDIT_DIR/include)
 
+  AC_CHECK_LIB(tinfo, tgetent,
+  [
+    PHP_ADD_LIBRARY(tinfo,,READLINE_SHARED_LIBADD)
+  ],[
   AC_CHECK_LIB(ncurses, tgetent,
   [
     PHP_ADD_LIBRARY(ncurses,,READLINE_SHARED_LIBADD)
@@ -72,6 +81,7 @@
       PHP_ADD_LIBRARY(termcap,,READLINE_SHARED_LIBADD)
     ])
   ])
+  ])
 
   PHP_CHECK_LIBRARY(edit, readline,
   [
