--- PHP_5_3/ext/soap/soap.c	2011/03/19 17:14:28	309432
+++ PHP_5_3/ext/soap/soap.c	2011/03/19 17:36:01	309433
@@ -1213,9 +1213,11 @@
 		zval **tmp;
 
 		if (zend_hash_find(ht, "soap_version", sizeof("soap_version"), (void**)&tmp) == SUCCESS) {
-			if (Z_TYPE_PP(tmp) == IS_LONG ||
-			    (Z_LVAL_PP(tmp) == SOAP_1_1 && Z_LVAL_PP(tmp) == SOAP_1_2)) {
+			if (Z_TYPE_PP(tmp) == IS_LONG &&
+			    (Z_LVAL_PP(tmp) == SOAP_1_1 || Z_LVAL_PP(tmp) == SOAP_1_2)) {
 				version = Z_LVAL_PP(tmp);
+			} else {
+				php_error_docref(NULL TSRMLS_CC, E_ERROR, "'soap_version' option must be SOAP_1_1 or SOAP_1_2");
 			}
 		}
 
