--- php-5.2.0/scripts/php-config.in	2006-12-14 01:11:33.348180526 +0200
+++ php-5.2.4/scripts/php-config.in	2007-08-31 19:43:04.061307670 +0300
@@ -2,6 +2,7 @@
 SED="@SED@"
 prefix="@prefix@"
 datarootdir="@datarootdir@"
+sysconfdir='@sysconfdir@'
 exec_prefix="@exec_prefix@"
 version="@PHP_VERSION@"
 vernum="@PHP_VERSION_ID@"
@@ -67,6 +67,8 @@ case "$1" in
   echo $extension_dir;;
 --include-dir)
   echo $include_dir;;
+--sysconfdir)
+  echo $sysconfdir;;
 --lib-dir)
   echo $lib_dir;;
 --lib-embed)
@@ -97,6 +99,7 @@ Options:
   --libs              [$libs]
   --extension-dir     [$extension_dir]
   --include-dir       [$include_dir]
+  --sysconfdir        [$sysconfdir]
   --lib-dir           [$lib_dir]
   --lib-embed         [$php_embed_lib]
   --man-dir           [$man_dir]

