diff -urN php-5.0.0.org/ext/cpdf/config.m4 php-5.0.0/ext/cpdf/config.m4
--- php-5.0.0.org/ext/cpdf/config.m4	2004-08-04 14:27:03.216825280 +0200
+++ php-5.0.0/ext/cpdf/config.m4	2004-08-04 14:39:37.100217584 +0200
@@ -3,8 +3,8 @@
 AC_DEFUN(CPDF_JPEG_TEST,[
   AC_ARG_WITH(jpeg-dir,
   [  --with-jpeg-dir[=DIR]     CPDF: Set the path to libjpeg install prefix.],[
-    for i in $withval /usr/local /usr; do
-      if test -f "$i/lib/libjpeg.$SHLIB_SUFFIX_NAME" -o -f "$i/lib/libjpeg.a"; then
+    for i in $withval /usr/local/lib /usr/lib /usr/local/lib64 /usr/lib64; do
+      if test -f "$i/libjpeg.$SHLIB_SUFFIX_NAME" -o -f "$i/libjpeg.a"; then
         CPDF_JPEG_DIR=$i
         break;
       fi
@@ -13,11 +13,11 @@
       AC_MSG_ERROR([libjpeg.(a|so) not found.])
     fi
     PHP_CHECK_LIBRARY(jpeg, jpeg_read_header, [
-      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $CPDF_JPEG_DIR/lib, CPDF_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(jpeg, $CPDF_JPEG_DIR, CPDF_SHARED_LIBADD)
     ] ,[
       AC_MSG_ERROR([CPDF: Problem with libjpeg.(a|so). Please check config.log for more information.])
     ], [
-      -L$CPDF_JPEG_DIR/lib
+      -L$CPDF_JPEG_DIR
     ])
   ],)
 ])
@@ -25,8 +25,8 @@
 AC_DEFUN(CPDF_TIFF_TEST,[
   AC_ARG_WITH(tiff-dir,
   [  --with-tiff-dir[=DIR]     CPDF: Set the path to libtiff install prefix.],[
-    for i in $withval /usr/local /usr; do
-      if test -f "$i/lib/libtiff.$SHLIB_SUFFIX_NAME" -o -f "$i/lib/libtiff.a"; then
+    for i in $withval /usr/local/lib /usr/lib /usr/local/lib64 /usr/lib64; do
+      if test -f "$i/libtiff.$SHLIB_SUFFIX_NAME" -o -f "$i/libtiff.a"; then
         CPDF_TIFF_DIR=$i
         break;
       fi
@@ -35,11 +35,11 @@
       AC_MSG_ERROR([libtiff.(a|so) not found.])
     fi
     PHP_CHECK_LIBRARY(tiff, TIFFOpen, [
-      PHP_ADD_LIBRARY_WITH_PATH(tiff, $CPDF_TIFF_DIR/lib, CPDF_SHARED_LIBADD)
+      PHP_ADD_LIBRARY_WITH_PATH(tiff, $CPDF_TIFF_DIR, CPDF_SHARED_LIBADD)
     ] ,[
       AC_MSG_ERROR([CPDF: Problem with libtiff.(a|so). Please check config.log for more information.])
     ], [
-      -L$CPDF_TIFF_DIR/lib
+      -L$CPDF_TIFF_DIR
     ])
   ],)
 ])
@@ -56,15 +56,20 @@
   for i in $PHP_CPDFLIB /usr/local /usr; do
     if test -f "$i/include/cpdflib.h"; then
       CPDFLIB_INCLUDE=$i/include
+      if test -d "$i/lib64"; then
+	j="$i/lib64"
+      else
+        j="$i/lib"
+      fi
 
       PHP_CHECK_LIBRARY(cpdf, cpdf_open, [
         PHP_ADD_INCLUDE($CPDFLIB_INCLUDE)
-        PHP_ADD_LIBRARY_WITH_PATH(cpdf, $i/lib, CPDF_SHARED_LIBADD)
+        PHP_ADD_LIBRARY_WITH_PATH(cpdf, $j, CPDF_SHARED_LIBADD)
         AC_DEFINE(HAVE_CPDFLIB,1,[Whether you have cpdflib])
       ], [
         AC_MSG_ERROR([Cpdflib module requires cpdflib >= 2.])
       ], [
-        -L$i/lib $CPDF_SHARED_LIBADD
+        -L$j $CPDF_SHARED_LIBADD
       ])
       break
     fi
