diff -burN php-5.0.3RC2.orig/ext/sybase/config.m4 php-5.0.3RC2/ext/sybase/config.m4
--- php-5.0.3RC2.orig/ext/sybase/config.m4	2004-12-13 11:20:44.245174736 +0100
+++ php-5.0.3RC2/ext/sybase/config.m4	2004-12-13 11:21:02.818351184 +0100
@@ -18,6 +18,7 @@
   PHP_ADD_INCLUDE($SYBASE_INCDIR)
   PHP_ADD_LIBRARY_WITH_PATH(sybdb, $SYBASE_LIBDIR, SYBASE_SHARED_LIBADD)
   PHP_NEW_EXTENSION(sybase, php_sybase_db.c, $ext_shared)
+  PHP_SUBST(SYBASE_SHARED_LIBADD)
   AC_CHECK_LIB(dnet_stub, dnet_addr,
      [ PHP_ADD_LIBRARY_WITH_PATH(dnet_stub,,SYBASE_SHARED_LIBADD)
         AC_DEFINE(HAVE_LIBDNET_STUB,1,[ ])
diff -burN php-5.0.3RC2.orig/ext/sybase/php_sybase_db.c php-5.0.3RC2/ext/sybase/php_sybase_db.c
--- php-5.0.3RC2.orig/ext/sybase/php_sybase_db.c	2004-12-13 11:20:44.262172152 +0100
+++ php-5.0.3RC2/ext/sybase/php_sybase_db.c	2004-12-13 11:21:33.205731600 +0100
@@ -1108,9 +1108,11 @@
 {
 	php_sybase_fetch_hash(INTERNAL_FUNCTION_PARAM_PASSTHRU);
 	if (Z_TYPE_P(return_value)==IS_ARRAY) {
-		Z_TYPE_P(return_value)=IS_OBJECT;
-		Z_OBJPROP_P(return_value) = Z_ARRVAL_P(return_value);
-		Z_OBJCE_P(return_value) = ZEND_STANDARD_CLASS_DEF_PTR;
+		object_and_properties_init(
+				return_value, 
+				ZEND_STANDARD_CLASS_DEF_PTR, 
+				Z_ARRVAL_P(return_value)
+		);
 	}
 }
 /* }}} */
