--- php-4.3.0/ext/xmlrpc/config.m4.orig	Wed Sep  4 20:47:25 2002
+++ php-4.3.0/ext/xmlrpc/config.m4	Thu Jan  2 21:29:53 2003
@@ -85,6 +85,8 @@
     AC_MSG_ERROR(Please reinstall the XMLRPC-EPI distribution)
   fi
 
+  PHP_NEW_EXTENSION(xmlrpc,xmlrpc-epi-php.c,$ext_shared)
+  XMLRPC_MODULE_TYPE=external
   PHP_ADD_INCLUDE($XMLRPC_DIR)
   PHP_ADD_LIBRARY_WITH_PATH(xmlrpc, $XMLRPC_DIR/lib, XMLRPC_SHARED_LIBADD)
   
