diff -durN php-4.2.2.orig/php.ini php-4.2.2/php.ini
--- php-4.2.2.orig/php.ini	Wed Aug  7 18:24:45 2002
+++ php-4.2.2/php.ini	Wed Aug  7 18:30:27 2002
@@ -74,7 +74,7 @@
 asp_tags = Off
 
 ; The number of significant digits displayed in floating point numbers.
-precision    =  12
+precision    =  14
 
 ; Enforce year 2000 compliance (will cause problems with non-compliant browsers)
 y2k_compliance = Off
@@ -371,7 +371,7 @@
 user_dir =
 
 ; Directory in which the loadable extensions (modules) reside.
-extension_dir = ./
+extension_dir = /usr/lib/php
 
 ; Whether or not to enable the dl() function.  The dl() function does NOT work
 ; properly in multithreaded servers, such as IIS or Zeus, and is automatically
@@ -692,7 +692,7 @@
 ; Argument passed to save_handler.  In the case of files, this is the path
 ; where data files are stored. Note: Windows users have to change this 
 ; variable in order to use PHP's session functions.
-session.save_path = /tmp
+session.save_path = /var/run/php
 
 ; Whether to use cookies.
 session.use_cookies = 1
