diff -urbB php-5.0.5.org/scripts/Makefile.frag php-5.0.5/scripts/Makefile.frag
--- php-5.0.5.org/scripts/Makefile.frag	2005-09-10 21:29:04.884122000 +0200
+++ php-5.0.5/scripts/Makefile.frag	2005-09-10 21:30:09.333324848 +0200
@@ -4,7 +4,7 @@
 #
 
 phpincludedir = $(includedir)/php
-phpbuilddir = $(libdir)/build
+phpbuilddir = $(libdir)/php/build
 
 BUILD_FILES = \
 	scripts/phpize.m4 \
diff -urbB php-5.0.5.org/scripts/phpize.in php-5.0.5/scripts/phpize.in
--- php-5.0.5.org/scripts/phpize.in	2005-06-15 19:19:41.000000000 +0200
+++ php-5.0.5/scripts/phpize.in	2005-09-10 21:51:45.523274024 +0200
@@ -3,7 +3,7 @@
 # Variable declaration
 prefix='@prefix@'
 exec_prefix="`eval echo @exec_prefix@`"
-phpdir="`eval echo @libdir@`/build"
+phpdir="`eval echo @libdir@`/php/build"
 includedir="`eval echo @includedir@`/php"
 builddir="`pwd`"
 
