--- php-4.2.2/ext/mysql/config.m4.orig	Fri Aug  9 23:49:00 2002
+++ php-4.2.2/ext/mysql/config.m4	Sat Aug 10 00:05:40 2002
@@ -49,6 +49,7 @@
 [  --with-mysql[=DIR]      Include MySQL support. DIR is the MySQL base directory.
                           If unspecified, the bundled MySQL library will be used.], yes)
 
+PHP_REQUIRE_CXX
 if test "$PHP_MYSQL" != "no"; then
   AC_DEFINE(HAVE_MYSQL, 1, [Whether you have MySQL])
   if test "$PHP_SAFE_MODE" = "yes"; then
