--- php-5.0.0/scripts/phpize.m4~	2003-09-03 08:08:12.000000000 +0000
+++ php-5.0.0/scripts/phpize.m4	2004-08-07 19:08:40.369661968 +0000
@@ -29,10 +29,10 @@
   AC_MSG_RESULT($EXTENSION_DIR)
 ])
 dnl
-AC_DEFUN(PHP_EXT_BUILDDIR,[.])dnl
-AC_DEFUN(PHP_EXT_DIR,[""])dnl
-AC_DEFUN(PHP_EXT_SRCDIR,[$abs_srcdir])dnl
-AC_DEFUN(PHP_ALWAYS_SHARED,[
+AC_DEFUN([PHP_EXT_BUILDDIR],[.])dnl
+AC_DEFUN([PHP_EXT_DIR],[""])dnl
+AC_DEFUN([PHP_EXT_SRCDIR],[$abs_srcdir])dnl
+AC_DEFUN([PHP_ALWAYS_SHARED],[
   ext_output="yes, shared"
   ext_shared=yes
   test "[$]$1" = "no" && $1=yes
