diff -urN php-5.1.0RC1.org/ext/ming/config.m4 php-5.1.0RC1/ext/ming/config.m4
--- php-5.1.0RC1.org/ext/ming/config.m4	2005-07-18 01:58:39.000000000 +0200
+++ php-5.1.0RC1/ext/ming/config.m4	2005-08-25 19:23:28.356268128 +0200
@@ -9,7 +9,7 @@
   AC_CHECK_LIB(m, sin)
 
   for i in $PHP_MING /usr/local /usr; do
-    if test -f $i/lib/libming.$SHLIB_SUFFIX_NAME -o -f $i/lib/libming.a; then
+    if test -f $i/$PHP_LIBDIR/libming.$SHLIB_SUFFIX_NAME -o -f $i/$PHP_LIBDIR/libming.a; then
       MING_DIR=$i
       break
     fi
