--- php-4.0.2/ext/ldap/config.m4.wiget	Wed Aug 30 12:30:42 2000
+++ php-4.0.2/ext/ldap/config.m4	Wed Aug 30 15:10:17 2000
@@ -40,7 +40,7 @@
 
 	PHP_SUBST(LDAP_SHARED_LIBADD)
 
-	if test -f $LDAP_LIBDIR/liblber.a; then
+	if test -f $LDAP_LIBDIR/liblber.so; then
 		AC_ADD_LIBRARY_WITH_PATH(lber, $LDAP_LIBDIR, LDAP_SHARED_LIBADD)
 		AC_ADD_LIBRARY_WITH_PATH(ldap, $LDAP_LIBDIR, LDAP_SHARED_LIBADD)
 	elif test -f $LDAP_LIBDIR/libldapssl30.so; then
