--- php-4.3.0/configure.in.orig	Fri Dec 27 05:07:18 2002
+++ php-4.3.0/configure.in	Thu Jan  2 20:34:43 2003
@@ -288,7 +288,6 @@
 PHP_CHECK_FUNC(htonl, socket)
 PHP_CHECK_FUNC(gethostname, nsl)
 PHP_CHECK_FUNC(gethostbyaddr, nsl)
-PHP_CHECK_FUNC(yp_get_default_domain, nsl)
 
 PHP_CHECK_FUNC(dlopen, dl)
 if test "$ac_cv_func_dlopen" = "yes"; then
--- php-5.2.0/acinclude.m4.orig	2006-12-22 21:52:13.766461055 +0100
+++ php-5.2.0/acinclude.m4	2006-12-22 21:52:37.159794165 +0100
@@ -1740,18 +1740,18 @@
 dnl Wrapper for AC_CHECK_LIB
 dnl
 AC_DEFUN([PHP_CHECK_LIBRARY], [
-  save_old_LDFLAGS=$LDFLAGS
+  save_old_LIBS=$LIBS
   ac_stuff="$5"
 
   save_ext_shared=$ext_shared
   ext_shared=yes
-  PHP_EVAL_LIBLINE([$]ac_stuff, LDFLAGS)
+  PHP_EVAL_LIBLINE([$]ac_stuff, LIBS)
   AC_CHECK_LIB([$1],[$2],[
-    LDFLAGS=$save_old_LDFLAGS
+    LIBS=$save_old_LIBS
     ext_shared=$save_ext_shared
     $3
   ],[
-    LDFLAGS=$save_old_LDFLAGS
+    LIBS=$save_old_LIBS
     ext_shared=$save_ext_shared
     unset ac_cv_lib_$1[]_$2
     $4
