see also http://bugs.php.net/bug.php?id=34793

--- php-5.1.2/main/php_ini.c	2006-01-17 17:01:16.000000000 +0200
+++ php-5.1.4/main/php_ini.c	2006-06-12 11:16:06.594944020 +0300
@@ -291,10 +291,10 @@
 	} else if (!sapi_module.php_ini_ignore) {
 		char *default_location;
 		char *env_location;
-		char *binary_location;
 		static const char paths_separator[] = { ZEND_PATHS_SEPARATOR, 0 };
 #ifdef PHP_WIN32
 		char *reg_location;
+		char *binary_location;
 #endif
 
 		env_location = getenv("PHPRC");
@@ -330,6 +330,7 @@
 			strcat(php_ini_search_path, env_location);
 		}
 
+#ifdef PHP_WIN32
 		/* Add cwd (only with CLI) */
 		if (strcmp(sapi_module.name, "cli") == 0) {
 			if (*php_ini_search_path) {
@@ -339,23 +340,12 @@
 		}
 
 		/* Add binary directory */
-#ifdef PHP_WIN32
 		binary_location = (char *) emalloc(MAXPATHLEN);
 		if (GetModuleFileName(0, binary_location, MAXPATHLEN) == 0) {
 			efree(binary_location);
 			binary_location = NULL;
 		}
-#else
-		if (sapi_module.executable_location) {
-			binary_location = (char *)emalloc(PATH_MAX);
-			if (!realpath(sapi_module.executable_location, binary_location)) {
-				efree(binary_location);
-				binary_location = NULL;			 
-			}
-		} else {
-			binary_location = NULL;
-		}
-#endif
+
 		if (binary_location) {
 			char *separator_location = strrchr(binary_location, DEFAULT_SLASH);
 			
@@ -370,7 +360,6 @@
 		}
 
 		/* Add default location */
-#ifdef PHP_WIN32
 		default_location = (char *) emalloc(MAXPATHLEN + 1);
 	
 		if (0 < GetWindowsDirectory(default_location, MAXPATHLEN)) {
