--- php-4.2.0/configure.in.orig	Wed May  1 22:39:35 2002
+++ php-4.2.0/configure.in	Wed May  1 22:41:49 2002
@@ -659,16 +659,6 @@
    	AC_MSG_RESULT([/usr/local/php/bin])
 ])
 
-PHP_ARG_WITH(openssl,for OpenSSL support,
-[  --with-openssl[=DIR]    Include OpenSSL support (requires OpenSSL >= 0.9.5) ])
-if test "$PHP_OPENSSL" = "yes"; then
-  PHP_OPENSSL="/usr/local/ssl /usr/local /usr /usr/local/openssl"
-fi
-
-if test "$PHP_OPENSSL" != "no"; then
-  PHP_SETUP_OPENSSL
-fi
-
 PHP_ARG_ENABLE(sigchild,whether to enable PHP's own SIGCHLD handler,
 [  --enable-sigchild       Enable PHP's own SIGCHLD handler.],no)
 
diff -Nur php-4.2.0/ext/openssl.orig/Makefile.in php-4.2.0/ext/openssl/Makefile.in
--- php-4.2.0/ext/openssl.orig/Makefile.in	Mon Nov 13 20:47:19 2000
+++ php-4.2.0/ext/openssl/Makefile.in	Wed May  1 23:12:07 2002
@@ -3,5 +3,6 @@
 LTLIBRARY_NAME        = libopenssl.la
 LTLIBRARY_SOURCES     = openssl.c
 LTLIBRARY_SHARED_NAME = openssl.la
+LTLIBRARY_SHARED_LIBADD = $(OPENSSL_SHARED_LIBADD)
 
 include $(top_srcdir)/build/dynlib.mk
diff -Nur php-4.2.0/ext/openssl.orig/config.m4 php-4.2.0/ext/openssl/config.m4
--- php-4.2.0/ext/openssl.orig/config.m4	Fri Nov 30 19:59:48 2001
+++ php-4.2.0/ext/openssl/config.m4	Wed May  1 23:16:39 2002
@@ -2,6 +2,16 @@
 dnl $Id$
 dnl
 
+PHP_ARG_WITH(openssl,for OpenSSL support,
+[  --with-openssl[=DIR]    Include OpenSSL support (requires OpenSSL >= 0.9.5) ])
+if test "$PHP_OPENSSL" = "yes"; then
+  PHP_OPENSSL="/usr/local/ssl /usr/local /usr /usr/local/openssl"
+fi
+
+if test "$PHP_OPENSSL" != "no"; then
+  PHP_SETUP_OPENSSL
+fi
+
 if test "$OPENSSL_DIR"; then
   PHP_EXTENSION(openssl, $ext_shared)
   AC_DEFINE(HAVE_OPENSSL_EXT,1,[ ])
--- php-4.2.0/acinclude.m4.orig	Wed May  1 23:16:39 2002
+++ php-4.2.0/acinclude.m4	Wed May  1 23:18:31 2002
@@ -85,17 +85,18 @@
   PHP_ADD_LIBPATH($OPENSSL_DIR/lib)
 
   AC_CHECK_LIB(crypto, CRYPTO_free, [
-    PHP_ADD_LIBRARY(crypto)
+    PHP_ADD_LIBRARY(crypto,, OPENSSL_SHARED_LIBADD)
   ],[
     AC_MSG_ERROR([libcrypto not found!])
   ])
 
   AC_CHECK_LIB(ssl, SSL_CTX_set_ssl_version, [
-    PHP_ADD_LIBRARY(ssl)
+    PHP_ADD_LIBRARY(ssl,, OPENSSL_SHARED_LIBADD)
   ],[
     AC_MSG_ERROR([libssl not found!])
   ])
   PHP_ADD_INCLUDE($OPENSSL_INC)
+  PHP_SUBST(OPENSSL_SHARED_LIBADD)
 ])
 
 dnl PHP_EVAL_LIBLINE(LINE, SHARED-LIBADD)
