--- php-8.4.1/build/php.m4~	2024-11-22 15:00:06.000000000 +0100
+++ php-8.4.1/build/php.m4	2024-11-22 15:03:55.783321362 +0100
@@ -2036,6 +2036,18 @@ dnl This macro is used to get a comparab
 dnl
 AC_DEFUN([PHP_AP_EXTRACT_VERSION], [m4_warn([obsolete],
   [The macro 'PHP_AP_EXTRACT_VERSION' is obsolete. Use 'apxs -q HTTPD_VERSION'])
+  PHP_ARG_WITH(apache-version, Set Apache version,
+  [  --with-apache-version=VERSION      Set Apache version instead of detecting from httpd binary])
+  if test "$with_apache_version" != "no"; then
+    ac_output="foo bar baz $with_apache_version"
+    ac_IFS=$IFS
+IFS="- /.
+"
+    set $ac_output
+    IFS=$ac_IFS
+
+    APACHE_VERSION=$(expr [$]4 \* 1000000 + [$]5 \* 1000 + [$]6)
+  else
 AS_IF([test -x "$1"], [
   ac_output=$($1 -v 2>&1 | grep version | $SED -e 's/Oracle-HTTP-//')
   ac_IFS=$IFS
@@ -2046,6 +2058,7 @@ IFS="- /.
 
   APACHE_VERSION=$(expr [$]4 \* 1000000 + [$]5 \* 1000 + [$]6)
 ])
+  fi
 ])
 
 dnl
