--- php-4.3.0/main/SAPI.c.orig	Thu Dec  5 23:15:00 2002
+++ php-4.3.0/main/SAPI.c	Fri Jan  3 10:28:01 2003
@@ -521,7 +521,7 @@
 				while (*ptr == ' ' && *ptr != '\0') {
 					ptr++;
 				}
-#if HAVE_ZLIB
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 				if(!strncmp(ptr, "image/", sizeof("image/")-1)) {
 					ZLIBG(output_compression) = 0;
 				}
@@ -669,7 +669,7 @@
 		return SUCCESS;
 	}
 
-#if HAVE_ZLIB
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 	/* Add output compression headers at this late stage in order to make
 	   it possible to switch it off inside the script. */
 	if (ZLIBG(output_compression)) {
--- php-4.3.0/ext/standard/image.c.orig	Thu Dec  5 16:52:35 2002
+++ php-4.3.0/ext/standard/image.c	Fri Jan  3 10:29:20 2003
@@ -34,7 +34,7 @@
 #endif
 #include "php_image.h"
 
-#if HAVE_ZLIB
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 #include "zlib.h"
 #endif
 
@@ -80,7 +80,7 @@
 	REGISTER_LONG_CONSTANT("IMAGETYPE_JP2",     IMAGE_FILETYPE_JP2,     CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMAGETYPE_JPX",     IMAGE_FILETYPE_JPX,     CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("IMAGETYPE_JB2",     IMAGE_FILETYPE_JB2,     CONST_CS | CONST_PERSISTENT);
-#if HAVE_ZLIB	
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 	REGISTER_LONG_CONSTANT("IMAGETYPE_SWC",     IMAGE_FILETYPE_SWC,     CONST_CS | CONST_PERSISTENT);
 #endif	
 	REGISTER_LONG_CONSTANT("IMAGETYPE_IFF",     IMAGE_FILETYPE_IFF,     CONST_CS | CONST_PERSISTENT);
@@ -170,7 +170,7 @@
 }
 /* }}} */
 
-#if HAVE_ZLIB
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 /* {{{ php_handle_swc
  */
 static struct gfxinfo *php_handle_swc(php_stream * stream TSRMLS_DC)
@@ -944,7 +944,7 @@
 		case IMAGE_FILETYPE_SWF:
 			result = php_handle_swf(stream TSRMLS_CC);
 			break;
-#if HAVE_ZLIB
+#if HAVE_ZLIB && !defined(COMPILE_DL_ZLIB)
 		case IMAGE_FILETYPE_SWC:
 			result = php_handle_swc(stream TSRMLS_CC);
 			break;
