--- php-5.3.17/sapi/litespeed/lsapi_main.c~	2010-11-20 00:24:12.000000000 +0100
+++ php-5.3.17/sapi/litespeed/lsapi_main.c	2012-09-15 20:58:09.482395102 +0200
@@ -635,7 +635,7 @@
         "  args...    Arguments passed to script.\n";
     php_output_startup();
     php_output_activate(TSRMLS_C);
-    php_printf( usage );
+    php_printf( "%s", usage );
 #ifdef PHP_OUTPUT_NEWAPI
     php_output_end_all(TSRMLS_C);
 #else
