diff -urNp -x '*.orig' php-5.2.17.org/scripts/Makefile.frag php-5.2.17/scripts/Makefile.frag
--- php-5.2.17.org/scripts/Makefile.frag	2010-07-13 19:24:44.000000000 +0200
+++ php-5.2.17/scripts/Makefile.frag	2021-10-23 19:01:40.383125127 +0200
@@ -4,7 +4,7 @@
 #
 
 phpincludedir = $(includedir)/php
-phpbuilddir = $(libdir)/build
+phpbuilddir = $(libdir)/php/build
 
 BUILD_FILES = \
 	scripts/phpize.m4 \
diff -urNp -x '*.orig' php-5.2.17.org/scripts/phpize.in php-5.2.17/scripts/phpize.in
--- php-5.2.17.org/scripts/phpize.in	2009-06-24 09:43:18.000000000 +0200
+++ php-5.2.17/scripts/phpize.in	2021-10-23 19:01:40.383125127 +0200
@@ -3,7 +3,7 @@
 # Variable declaration
 prefix='@prefix@'
 exec_prefix="`eval echo @exec_prefix@`"
-phpdir="`eval echo @libdir@`/build"
+phpdir="`eval echo @libdir@`/php/build"
 includedir="`eval echo @includedir@`/php"
 builddir="`pwd`"
 SED="@SED@"
