--- php.ini-dist	Mon Apr  8 03:15:16 2002
+++ php.ini	Sat Jul 27 18:41:04 2002
@@ -74,7 +74,7 @@
 asp_tags = Off
 
 ; The number of significant digits displayed in floating point numbers.
-precision    =  12
+precision    =  14
 
 ; Enforce year 2000 compliance (will cause problems with non-compliant browsers)
 y2k_compliance = Off
@@ -692,7 +692,7 @@
 ; Argument passed to save_handler.  In the case of files, this is the path
 ; where data files are stored. Note: Windows users have to change this 
 ; variable in order to use PHP's session functions.
-session.save_path = /tmp
+session.save_path = /var/run/php
 
 ; Whether to use cookies.
 session.use_cookies = 1
