diff -urNp -x '*.orig' php-5.2.17.org/ext/mysql/config.m4 php-5.2.17/ext/mysql/config.m4
--- php-5.2.17.org/ext/mysql/config.m4	2006-06-01 21:14:48.000000000 +0200
+++ php-5.2.17/ext/mysql/config.m4	2021-10-23 18:59:19.959791807 +0200
@@ -52,6 +52,7 @@ if test -z "$PHP_ZLIB_DIR"; then
 fi
 
 
+PHP_REQUIRE_CXX
 if test "$PHP_MYSQL" != "no"; then
   AC_DEFINE(HAVE_MYSQL, 1, [Whether you have MySQL])
 
