--- php-4.0.5/ext/imap/php_imap.c.wiget	Tue Apr 24 16:17:41 2001
+++ php-4.0.5/ext/imap/php_imap.c	Fri May  4 13:07:19 2001
@@ -428,28 +428,28 @@
 
 
 #if 1
-#ifndef PHP_WIN32
+#if 0
 	mail_link(&unixdriver);   /* link in the unix driver */
 #endif
 	mail_link(&imapdriver);      /* link in the imap driver */
 	mail_link(&nntpdriver);      /* link in the nntp driver */
 	mail_link(&pop3driver);      /* link in the pop3 driver */
-#ifndef PHP_WIN32
+#if 0
 	mail_link(&mhdriver);        /* link in the mh driver */
 	mail_link(&mxdriver);        /* link in the mx driver */
-#endif
 	mail_link(&mbxdriver);       /* link in the mbx driver */
 	mail_link(&tenexdriver);     /* link in the tenex driver */
 	mail_link(&mtxdriver);       /* link in the mtx driver */
+#endif
-#ifndef PHP_WIN32
+#if 0
 	mail_link(&mmdfdriver);      /* link in the mmdf driver */
 	mail_link(&newsdriver);      /* link in the news driver */
 	mail_link(&philedriver);     /* link in the phile driver */
+#endif
 	auth_link(&auth_log);        /* link in the log authenticator */
 #ifdef  HAVE_IMAP_SSL
 	ssl_onceonlyinit ();
 	auth_link (&auth_ssl);       /* link in the ssl authenticator */
-#endif
 #endif
 	mail_link(&dummydriver);     /* link in the dummy driver */
 #else
