--- php-5.3.3/acinclude.m4.orig	2010-10-19 21:38:42.000000000 +0200
+++ php-5.3.3/acinclude.m4	2010-10-19 22:53:11.792893220 +0200
@@ -2252,7 +2252,7 @@
     KERBEROS_LIBS=`$KRB5_CONFIG --libs gssapi`
     KERBEROS_CFLAGS=`$KRB5_CONFIG --cflags gssapi`
 
-    if test -n "$KERBEROS_LIBS" && test -n "$KERBEROS_CFLAGS"; then
+    if test -n "$KERBEROS_LIBS" || test -n "$KERBEROS_CFLAGS"; then
       found_kerberos=yes
       PHP_EVAL_LIBLINE($KERBEROS_LIBS, $1)
       PHP_EVAL_INCLINE($KERBEROS_CFLAGS)
