--- php-5.6.6/sapi/cgi/cgi_main.c~	2015-02-27 18:09:34.000000000 +0200
+++ php-5.6.6/sapi/cgi/cgi_main.c	2015-02-27 18:11:26.296171361 +0200
@@ -737,12 +737,8 @@
 		request = (fcgi_request*) SG(server_context);
 		if (request) {
 			int ret, len = strlen(message);
-			char *buf = malloc(len+2);
 
-			memcpy(buf, message, len);
-			memcpy(buf + len, "\n", sizeof("\n"));
-			ret = fcgi_write(request, FCGI_STDERR, buf, len + 1);
-			free(buf);
+			ret = fcgi_write(request, FCGI_STDERR, message, len);
 			if (ret < 0) {
 				php_handle_aborted_connection();
 			}
