--- php-4.3.0/ext/xml/config.m4.orig	Thu Nov  7 20:31:08 2002
+++ php-4.3.0/ext/xml/config.m4	Thu Jan  2 19:54:39 2003
@@ -27,7 +27,7 @@
   else
     PHP_NEW_EXTENSION(xml, xml.c, $ext_shared)
 
-    for i in $PHP_XML $PHP_EXPAT_DIR; do
+    for i in /usr /usr/local $PHP_XML `echo $PHP_EXPAT_DIR | sed -e s/shared,//` ; do
       if test -f $i/lib/libexpat.a -o -f $i/lib/libexpat.$SHLIB_SUFFIX_NAME ; then
         EXPAT_DIR=$i
       fi
