--- php-7.0.0beta1/main/php_ini.c~	2015-07-13 16:04:48.000000000 +0300
+++ php-7.0.0beta1/main/php_ini.c	2015-07-13 16:10:07.496198519 +0300
@@ -379,12 +379,19 @@
 }
 /* }}} */
 
+static int php_csort(const struct dirent **a, const struct dirent **b)
+{
+	return strcmp((*a)->d_name,(*b)->d_name);
+}
+
 /* {{{ php_init_config
  */
 int php_init_config(void)
 {
 	char *php_ini_file_name = NULL;
 	char *php_ini_search_path = NULL;
+	// value for php_ini_loaded_file() to be stored into php_ini_opened_path
+	char *php_ini_loaded_file = NULL;
 	int php_ini_scanned_path_len;
 	char *open_basedir;
 	int free_ini_search_path = 0;
@@ -561,8 +568,29 @@
 				}
 			}
 		}
+		/* Search (global) php.ini file in search path */
+		if (!fh.handle.fp) {
+			fh.handle.fp = php_fopen_with_path("php.ini", "r", php_ini_search_path, &php_ini_opened_path TSRMLS_CC);
+			if (fh.handle.fp) {
+				fh.filename = php_ini_opened_path;
+
+				// parse the ini file now
+				PG(open_basedir) = open_basedir;
+				fh.type = ZEND_HANDLE_FP;
+				zend_parse_ini_file(&fh, 1, ZEND_INI_SCANNER_NORMAL, (zend_ini_parser_cb_t) php_ini_parser_cb, &configuration_hash TSRMLS_CC);
+				open_basedir = PG(open_basedir);
+				PG(open_basedir) = NULL;
+				fh.handle.fp = NULL;
 
-		/* Otherwise search for php-%sapi-module-name%.ini file in search path */
+				// save this for later
+				php_ini_loaded_file = estrdup(php_ini_opened_path);
+
+				efree(php_ini_opened_path);
+				fh.filename = php_ini_opened_path = NULL;
+			}
+		}
+
+		/* Also search for php-%sapi-module-name%.ini file in search path */
 		if (!fh.handle.fp) {
 			const char *fmt = "php-%s.ini";
 			char *ini_fname;
@@ -597,15 +597,12 @@
 			if (fh.handle.fp) {
 				fh.filename = ZSTR_VAL(opened_path);
 			}
-		}
 
-		/* If still no ini file found, search for php.ini file in search path */
-		if (!fh.handle.fp) {
-			fh.handle.fp = php_fopen_with_path("php.ini", "r", php_ini_search_path, &opened_path);
-			if (fh.handle.fp) {
-				fh.filename = ZSTR_VAL(opened_path);
-			}
-		}
+			// restore previous value, so php_ini_loaded_file() works as expected
+			if (php_ini_loaded_file) {
+				php_ini_opened_path = php_ini_loaded_file;
+ 			}
+ 		}
 	}
 
 	if (free_ini_search_path) {
@@ -597,6 +622,10 @@
 
 		{
 			zval tmp;
+			// use main php.ini if loaded
+			if (php_ini_loaded_file) {
+				fh.filename = php_ini_loaded_file;
+			}
 
 			Z_STRLEN(tmp) = strlen(fh.filename);
 			Z_STRVAL(tmp) = zend_strndup(fh.filename, Z_STRLEN(tmp));
@@ -630,13 +659,15 @@
 		zend_llist scanned_ini_list;
 		zend_llist_element *element;
 		int l, total_l = 0;
+		const char *fmt = "%s:" PHP_CONFIG_FILE_PATH "/%s.d";
 		char *bufpath, *debpath, *endpath;
 		int lenpath;
 
 		zend_llist_init(&scanned_ini_list, sizeof(char *), (llist_dtor_func_t) free_estring, 1);
 		memset(&fh2, 0, sizeof(fh2));
 
-		bufpath = estrdup(php_ini_scanned_path);
+		bufpath = emalloc(strlen(php_ini_scanned_path) + strlen(fmt) + strlen(sapi_module.name));
+		sprintf(bufpath, fmt, php_ini_scanned_path, sapi_module.name);
 		for (debpath = bufpath ; debpath ; debpath=endpath) {
 			endpath = strchr(debpath, DEFAULT_DIR_SEPARATOR);
 			if (endpath) {
@@ -649,7 +680,7 @@
 			}
 			lenpath = strlen(debpath);
 
-			if (lenpath > 0 && (ndir = php_scandir(debpath, &namelist, 0, php_alphasort)) > 0) {
+			if (lenpath > 0 && (ndir = php_scandir(debpath, &namelist, 0, php_csort)) > 0) {
 
 				for (i = 0; i < ndir; i++) {
 
