diff -urBb php-5.1.0RC1.org/sapi/apache2handler/config.m4 php-5.1.0RC1/sapi/apache2handler/config.m4
--- php-5.1.0RC1.org/sapi/apache2handler/config.m4	2005-08-05 10:14:54.000000000 +0200
+++ php-5.1.0RC1/sapi/apache2handler/config.m4	2005-08-25 15:09:38.486655320 +0200
@@ -58,9 +58,9 @@
 
   # Test that we're trying to configure with apache 2.x
   PHP_AP_EXTRACT_VERSION($APXS_HTTPD)
-  if test "$APACHE_VERSION" -le 2000000; then
+  if test "$APACHE_VERSION" -gt 0 -a "$APACHE_VERSION" -le 2000000; then
     AC_MSG_ERROR([You have enabled Apache 2 support while your server is Apache 1.3.  Please use the appropiate switch --with-apxs (without the 2)])
-  elif test "$APACHE_VERSION" -lt 2000044; then
+  elif test "$APACHE_VERSION" -gt 0 -a "$APACHE_VERSION" -lt 2000044; then
     AC_MSG_ERROR([Please note that Apache version >= 2.0.44 is required])
   fi
 
