--- php4/ext/standard/mail.c	2 Apr 2001 16:37:50 -0000	1.41
+++ php4/ext/standard/mail.c	25 Sep 2001 22:48:43 -0000	1.44
@@ -12,7 +12,7 @@
    +----------------------------------------------------------------------+
  */
 
-/* $Id$ */
+/* $Id$ */
 
 #include <stdlib.h>
 #include <ctype.h>
@@ -54,8 +56,8 @@
 	}
 
 	convert_to_string_ex(pstr);
-	if ((*pstr)->value.str.val) {
-		str = (*pstr)->value.str.val;
+	if (Z_STRVAL_PP(pstr)) {
+		str = Z_STRVAL_PP(pstr);
 	} else {
 		php_error(E_WARNING, "Must give string parameter to ezmlm_hash()");
 		RETURN_FALSE;
@@ -85,8 +88,8 @@
 	}
 	/* To: */
 	convert_to_string_ex(argv[0]);
-	if ((*argv[0])->value.str.val) {
-		to = (*argv[0])->value.str.val;
+	if (Z_STRVAL_PP(argv[0])) {
+		to = Z_STRVAL_PP(argv[0]);
 	} else {
 		php_error(E_WARNING, "No to field in mail command");
 		RETURN_FALSE;
@@ -94,8 +97,8 @@
 
 	/* Subject: */
 	convert_to_string_ex(argv[1]);
-	if ((*argv[1])->value.str.val) {
+	if (Z_STRVAL_PP(argv[1])) {
 		subject = Z_STRVAL_PP(argv[1]);
 	} else {
 		php_error(E_WARNING, "No subject field in mail command");
 		RETURN_FALSE;
@@ -103,8 +106,8 @@
 
 	/* message body */
 	convert_to_string_ex(argv[2]);
-	if ((*argv[2])->value.str.val) {
+	if (Z_STRVAL_PP(argv[2])) {
 		message = Z_STRVAL_PP(argv[2]);
 	} else {
 		/* this is not really an error, so it is allowed. */
 		php_error(E_WARNING, "No message string in mail command");
