--- php/pear/Makefile.frag.orig	Wed Jun 18 14:15:02 2003
+++ php/pear/Makefile.frag	Wed Jun 18 14:15:07 2003
@@ -3,7 +3,7 @@
 peardir=$(PEAR_INSTALLDIR)
 
 # Skip all php.ini files altogether
-PEAR_INSTALL_FLAGS = -n -dshort_open_tag=0 -dsafe_mode=0
+PEAR_INSTALL_FLAGS = -c $(top_builddir)/pear -dshort_open_tag=0 -dsafe_mode=0
 
 install-pear-installer: $(top_builddir)/sapi/cli/php
 	@$(top_builddir)/sapi/cli/php $(PEAR_INSTALL_FLAGS) $(srcdir)/install-pear.php $(srcdir)/package-*.xml
