diff -urNp -x '*.orig' php-5.6.40.org/scripts/Makefile.frag php-5.6.40/scripts/Makefile.frag
--- php-5.6.40.org/scripts/Makefile.frag	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/scripts/Makefile.frag	2021-08-23 23:19:13.635805958 +0200
@@ -4,7 +4,7 @@
 #
 
 phpincludedir = $(includedir)/php
-phpbuilddir = $(libdir)/build
+phpbuilddir = $(libdir)/php/build
 
 BUILD_FILES = \
 	scripts/phpize.m4 \
diff -urNp -x '*.orig' php-5.6.40.org/scripts/phpize.in php-5.6.40/scripts/phpize.in
--- php-5.6.40.org/scripts/phpize.in	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40/scripts/phpize.in	2021-08-23 23:19:13.635805958 +0200
@@ -4,7 +4,7 @@
 prefix='@prefix@'
 datarootdir='@datarootdir@'
 exec_prefix="`eval echo @exec_prefix@`"
-phpdir="`eval echo @libdir@`/build"
+phpdir="`eval echo @libdir@`/php/build"
 includedir="`eval echo @includedir@`/php"
 builddir="`pwd`"
 SED="@SED@"
