--- php-7.0/sapi/cgi/cgi_main.c~	2015-05-21 23:12:23.000000000 +0300
+++ php-7.0/sapi/cgi/cgi_main.c	2015-05-21 23:13:24.947976867 +0300
@@ -701,12 +701,8 @@
 		request = (fcgi_request*) SG(server_context);
 		if (request) {
 			int ret, len = (int)strlen(message);
-			char *buf = malloc(len+2);
 
-			memcpy(buf, message, len);
-			memcpy(buf + len, "\n", sizeof("\n"));
-			ret = fcgi_write(request, FCGI_STDERR, buf, (int)(len + 1));
-			free(buf);
+			ret = fcgi_write(request, FCGI_STDERR, message, len);
 			if (ret < 0) {
 				php_handle_aborted_connection();
 			}
