--- php-4.4.2/Makefile.global~	2006-04-04 00:35:27.582384109 +0300
+++ php-4.4.2/Makefile.global	2006-04-04 00:40:19.318848123 +0300
@@ -1,6 +1,7 @@
 mkinstalldirs = $(top_srcdir)/build/shtool mkdir -p
 INSTALL = $(top_srcdir)/build/shtool install -c
 INSTALL_DATA = $(INSTALL) -m 644
+LD_VERSION_SCRIPT = -Wl,--version-script=$(top_srcdir)/php.ver
 
 DEFS = -DPHP_ATOM_INC -I$(top_builddir)/include -I$(top_builddir)/main -I$(top_srcdir)
 COMMON_FLAGS = $(DEFS) $(INCLUDES) $(EXTRA_INCLUDES) $(CPPFLAGS) $(PHP_FRAMEWORKPATH)
--- /dev/null	2006-03-28 14:00:37.387408000 +0300
+++ php-4.4.2/php.ver	2006-04-04 00:40:55.609652026 +0300
@@ -0,0 +1 @@
+php5 { global: *; };
--- php-4.4.2/acinclude.m4~	2006-04-03 21:43:42.852081966 +0300
+++ php-4.4.2/acinclude.m4	2006-04-04 00:47:58.259016152 +0300
@@ -1205,7 +1205,7 @@
       ;;
     *[)]
       suffix=la
-      link_cmd='$(LIBTOOL) --mode=link ifelse($4,,[$(CC)],[$(CXX)]) $(LDFLAGS) -o [$]@ -export-dynamic -avoid-version -prefer-pic -module -rpath $(phplibdir) $(EXTRA_LDFLAGS) $($2) $(translit($1,a-z_-,A-Z__)_SHARED_LIBADD)'
+      link_cmd='$(LIBTOOL) --mode=link ifelse($4,,[$(CC)],[$(CXX)]) $(LDFLAGS) -o [$]@ -export-dynamic -avoid-version -prefer-pic -module -rpath $(phplibdir) libphp_common.la $(EXTRA_LDFLAGS) $(LD_VERSION_SCRIPT) $($2) $(translit($1,a-z_-,A-Z__)_SHARED_LIBADD)'
       ;;
   esac
 
--- php-5.1.2/Makefile.global~	2006-04-04 01:20:14.511991339 +0300
+++ php-5.1.2/Makefile.global	2006-04-04 01:20:51.782819766 +0300
@@ -15,15 +15,15 @@
 build-modules: $(PHP_MODULES)
 
 libphp_common.la: $(PHP_GLOBAL_OBJS)
-	$(LIBTOOL) --mode=link $(CC) -rpath $(phptempdir) -avoid-version $(LDFLAGS) $(PHP_RPATHS) $(PHP_GLOBAL_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@ -release $(PHP_VERSION)
+	$(LIBTOOL) --mode=link $(CC) -rpath $(phptempdir) -avoid-version $(LDFLAGS) $(LD_VERSION_SCRIPT) $(PHP_RPATHS) $(PHP_GLOBAL_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@ -release $(PHP_VERSION)
 	-@$(LIBTOOL) --silent --mode=install cp libphp_common.la $(phptempdir)/libphp_common.la >/dev/null 2>&1
 
 libphp5.la: libphp_common.la $(PHP_SAPI_OBJS)
-	$(LIBTOOL) --mode=link $(CC) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS) $(phptempdir)/libphp_common.la $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
+	$(LIBTOOL) --mode=link $(CC) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(LD_VERSION_SCRIPT) $(PHP_RPATHS) $(phptempdir)/libphp_common.la $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
 	-@$(LIBTOOL) --silent --mode=install cp libphp5.la $(phptempdir)/libphp5.la >/dev/null 2>&1
 
 $(LIBTOOL_SAPI): libphp_common.la $(PHP_SAPI_OBJS)
-	$(LIBTOOL) --mode=link $(CC) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS) $(phptempdir)/libphp_common.la $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
+	$(LIBTOOL) --mode=link $(CC) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(LD_VERSION_SCRIPT) $(PHP_RPATHS) $(phptempdir)/libphp_common.la $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
 	-@$(LIBTOOL) --silent --mode=install cp libphp5.la $(phptempdir)/libphp5.la >/dev/null 2>&1
 
 libtool-sapi: $(LIBTOOL_SAPI)
--- php-5.1.2/sapi/apache/mod_php5.c~	2006-01-06 20:06:38.000000000 +0200
+++ php-5.1.2/sapi/apache/mod_php5.c	2006-04-04 02:12:59.462271322 +0300
@@ -930,7 +930,9 @@
 handler_rec php_handlers[] =
 {
 	{"application/x-httpd-php", send_parsed_php},
+	{"application/x-httpd-php5", send_parsed_php},
 	{"application/x-httpd-php-source", send_parsed_php_source},
+	{"application/x-httpd-php5-source", send_parsed_php_source},
 	{"text/html", php_xbithack_handler},
 	{NULL}
 };
