--- php-4.3.2/ext/xslt/config.m4.orig	Wed Apr 23 08:55:00 2003
+++ php-4.3.2/ext/xslt/config.m4	Fri Jun 13 13:08:17 2003
@@ -66,7 +66,7 @@
        SABINF_CFLAGS=`$XSLT_DIR/bin/sablot-config --cflags`
        SABINF_LIBS=`$XSLT_DIR/bin/sablot-config --libs`
        SABINF_PREFIX=`$XSLT_DIR/bin/sablot-config --prefix`
-       SABINF_ALL="\"Cflags: $SABINF_CFLAGS\nLibs: $SABINF_LIBS\nPrefix: $SABINF_PREFIX\""
+       SABINF_ALL="\"Cflags: $SABINF_CFLAGS\\\\nLibs: $SABINF_LIBS\\\\nPrefix: $SABINF_PREFIX\""
        PHP_DEFINE(SAB_INFO, "$SABINF_ALL")
     else
        AC_MSG_RESULT(not found)
