--- php-7.4.0alpha3/sapi/apache2handler/config.m4~	2019-07-09 10:39:56.000000000 +0300
+++ php-7.4.0alpha3/sapi/apache2handler/config.m4	2019-07-14 23:54:57.845267916 +0300
@@ -59,7 +59,7 @@
 
   dnl Test that we're trying to configure with apache 2.x
   PHP_AP_EXTRACT_VERSION($APXS_HTTPD)
-  if test "$APACHE_VERSION" -lt 2000044; then
+  if test "$APACHE_VERSION" -gt 0 -a "$APACHE_VERSION" -lt 2000044; then
     AC_MSG_ERROR([Please note that Apache version >= 2.0.44 is required])
   fi
 
