diff -urNp -x '*.orig' php-7.0.33.org/Makefile.global php-7.0.33/Makefile.global
--- php-7.0.33.org/Makefile.global	2018-12-04 18:22:44.000000000 +0100
+++ php-7.0.33/Makefile.global	2021-08-23 23:35:29.816640040 +0200
@@ -15,8 +15,12 @@ build-modules: $(PHP_MODULES) $(PHP_ZEND
 
 build-binaries: $(PHP_BINARIES)
 
-libphp$(PHP_MAJOR_VERSION).la: $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS)
-	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(EXTRA_CFLAGS) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS) $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
+libphp_common.la: $(PHP_GLOBAL_OBJS)
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(EXTRA_CFLAGS) -rpath $(phptempdir) -avoid-version $(LDFLAGS) $(PHP_RPATHS) $(PHP_GLOBAL_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@ -release $(PHP_VERSION)
+	-@$(LIBTOOL) --silent --mode=install cp $@ $(phptempdir)/$@ >/dev/null 2>&1
+
+libphp$(PHP_MAJOR_VERSION).la: libphp_common.la $(PHP_SAPI_OBJS)
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(EXTRA_CFLAGS) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS) $(phptempdir)/libphp_common.la $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
 	-@$(LIBTOOL) --silent --mode=install cp $@ $(phptempdir)/$@ >/dev/null 2>&1
 
 libs/libphp$(PHP_MAJOR_VERSION).bundle: $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS)
@@ -27,10 +31,10 @@ install: $(all_targets) $(install_target
 install-sapi: $(OVERALL_TARGET)
 	@echo "Installing PHP SAPI module:       $(PHP_SAPI)"
 	-@$(mkinstalldirs) $(INSTALL_ROOT)$(bindir)
-	-@if test ! -r $(phptempdir)/libphp$(PHP_MAJOR_VERSION).$(SHLIB_DL_SUFFIX_NAME); then \
+	-@if test ! -r $(phptempdir)/libphp_common.$(SHLIB_DL_SUFFIX_NAME); then \
 		for i in 0.0.0 0.0 0; do \
-			if test -r $(phptempdir)/libphp$(PHP_MAJOR_VERSION).$(SHLIB_DL_SUFFIX_NAME).$$i; then \
-				$(LN_S) $(phptempdir)/libphp$(PHP_MAJOR_VERSION).$(SHLIB_DL_SUFFIX_NAME).$$i $(phptempdir)/libphp$(PHP_MAJOR_VERSION).$(SHLIB_DL_SUFFIX_NAME); \
+			if test -r $(phptempdir)/libphp_common.$(SHLIB_DL_SUFFIX_NAME).$$i; then \
+				$(LN_S) $(phptempdir)/libphp_common.$(SHLIB_DL_SUFFIX_NAME).$$i $(phptempdir)/libphp_common.$(SHLIB_DL_SUFFIX_NAME); \
 				break; \
 			fi; \
 		done; \
diff -urNp -x '*.orig' php-7.0.33.org/configure.in php-7.0.33/configure.in
--- php-7.0.33.org/configure.in	2018-12-04 18:22:44.000000000 +0100
+++ php-7.0.33/configure.in	2021-08-23 23:35:29.816640040 +0200
@@ -224,6 +224,9 @@ PHP_ARG_ENABLE(rpath, whether to enable
 dnl check for -R, etc. switch
 PHP_RUNPATH_SWITCH
 
+dnl Always shared
+PHP_BUILD_SHARED
+
 dnl Checks for some support/generator progs
 PHP_PROG_AWK
 PHP_PROG_BISON
diff -urNp -x '*.orig' php-7.0.33.org/sapi/cgi/Makefile.frag php-7.0.33/sapi/cgi/Makefile.frag
--- php-7.0.33.org/sapi/cgi/Makefile.frag	2018-12-04 18:23:01.000000000 +0100
+++ php-7.0.33/sapi/cgi/Makefile.frag	2021-08-23 23:35:29.816640040 +0200
@@ -1,6 +1,6 @@
 cgi: $(SAPI_CGI_PATH)
 
-$(SAPI_CGI_PATH): $(PHP_GLOBAL_OBJS) $(PHP_BINARY_OBJS) $(PHP_FASTCGI_OBJS) $(PHP_CGI_OBJS)
+$(SAPI_CGI_PATH): libphp_common.la $(PHP_BINARY_OBJS) $(PHP_FASTCGI_OBJS) $(PHP_CGI_OBJS)
 	$(BUILD_CGI)
 
 install-cgi: $(SAPI_CGI_PATH)
diff -urNp -x '*.orig' php-7.0.33.org/sapi/cgi/config9.m4 php-7.0.33/sapi/cgi/config9.m4
--- php-7.0.33.org/sapi/cgi/config9.m4	2018-12-04 18:23:01.000000000 +0100
+++ php-7.0.33/sapi/cgi/config9.m4	2021-08-23 23:35:29.816640040 +0200
@@ -64,7 +64,7 @@ if test "$PHP_CGI" != "no"; then
         BUILD_CGI="\$(CC) \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(NATIVE_RPATHS) \$(PHP_GLOBAL_OBJS:.lo=.o) \$(PHP_BINARY_OBJS:.lo=.o) \$(PHP_FASTCGI_OBJS:.lo=.o) \$(PHP_CGI_OBJS:.lo=.o) \$(PHP_FRAMEWORKS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CGI_PATH)"
       ;;
       *)
-        BUILD_CGI="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \$(PHP_GLOBAL_OBJS) \$(PHP_BINARY_OBJS) \$(PHP_FASTCGI_OBJS) \$(PHP_CGI_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CGI_PATH)"
+        BUILD_CGI="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) libphp_common.la \$(PHP_BINARY_OBJS) \$(PHP_FASTCGI_OBJS) \$(PHP_CGI_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CGI_PATH)"
       ;;
     esac
 
diff -urNp -x '*.orig' php-7.0.33.org/sapi/cli/Makefile.frag php-7.0.33/sapi/cli/Makefile.frag
--- php-7.0.33.org/sapi/cli/Makefile.frag	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/sapi/cli/Makefile.frag	2021-08-23 23:35:29.816640040 +0200
@@ -1,6 +1,6 @@
 cli: $(SAPI_CLI_PATH)
 
-$(SAPI_CLI_PATH): $(PHP_GLOBAL_OBJS) $(PHP_BINARY_OBJS) $(PHP_CLI_OBJS)
+$(SAPI_CLI_PATH): libphp_common.la $(PHP_BINARY_OBJS) $(PHP_CLI_OBJS)
 	$(BUILD_CLI)
 
 install-cli: $(SAPI_CLI_PATH)
diff -urNp -x '*.orig' php-7.0.33.org/sapi/cli/config.m4 php-7.0.33/sapi/cli/config.m4
--- php-7.0.33.org/sapi/cli/config.m4	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/sapi/cli/config.m4	2021-08-23 23:35:29.816640040 +0200
@@ -48,7 +48,7 @@ if test "$PHP_CLI" != "no"; then
     BUILD_CLI="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \$(PHP_BINARY_OBJS) \$(PHP_CLI_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -Lnetware -lphp7lib -o \$(SAPI_CLI_PATH)"
     ;;
   *)
-    BUILD_CLI="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \$(PHP_GLOBAL_OBJS) \$(PHP_BINARY_OBJS) \$(PHP_CLI_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CLI_PATH)"
+    BUILD_CLI="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) libphp_common.la \$(PHP_BINARY_OBJS) \$(PHP_CLI_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CLI_PATH)"
     ;;
   esac
 
diff -urNp -x '*.orig' php-7.0.33.org/sapi/litespeed/Makefile.frag php-7.0.33/sapi/litespeed/Makefile.frag
--- php-7.0.33.org/sapi/litespeed/Makefile.frag	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/sapi/litespeed/Makefile.frag	2021-08-23 23:35:29.816640040 +0200
@@ -1,6 +1,6 @@
 litespeed: $(SAPI_LITESPEED_PATH)
 
-$(SAPI_LITESPEED_PATH): $(PHP_GLOBAL_OBJS) $(PHP_BINARY_OBJS) $(PHP_LITESPEED_OBJS)
+$(SAPI_LITESPEED_PATH): libphp_common.la $(PHP_BINARY_OBJS) $(PHP_LITESPEED_OBJS)
 	$(BUILD_LITESPEED)
 
 install-litespeed: $(SAPI_LITESPEED_PATH)
diff -urNp -x '*.orig' php-7.0.33.org/sapi/litespeed/config.m4 php-7.0.33/sapi/litespeed/config.m4
--- php-7.0.33.org/sapi/litespeed/config.m4	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/sapi/litespeed/config.m4	2021-08-23 23:35:29.816640040 +0200
@@ -17,10 +17,10 @@ if test "$PHP_LITESPEED" != "no"; then
     ;;
   *cygwin*)
     SAPI_LITESPEED_PATH=sapi/litespeed/php.exe
-    BUILD_LITESPEED="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \$(PHP_GLOBAL_OBJS) \$(PHP_BINARY_OBJS) \$(PHP_LITESPEED_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_LITESPEED_PATH)"
+    BUILD_LITESPEED="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) libphp_common.la \$(PHP_BINARY_OBJS) \$(PHP_LITESPEED_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_LITESPEED_PATH)"
     ;;
   *)
-    BUILD_LITESPEED="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \$(PHP_GLOBAL_OBJS) \$(PHP_BINARY_OBJS) \$(PHP_LITESPEED_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_LITESPEED_PATH)"
+    BUILD_LITESPEED="\$(LIBTOOL) --mode=link \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) libphp_common.la \$(PHP_BINARY_OBJS) \$(PHP_LITESPEED_OBJS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_LITESPEED_PATH)"
     ;;
   esac
 
diff -urNp -x '*.orig' php-7.0.33.org/sapi/phpdbg/Makefile.frag php-7.0.33/sapi/phpdbg/Makefile.frag
--- php-7.0.33.org/sapi/phpdbg/Makefile.frag	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/sapi/phpdbg/Makefile.frag	2021-08-23 23:35:29.816640040 +0200
@@ -2,10 +2,10 @@ phpdbg: $(BUILD_BINARY)
 
 phpdbg-shared: $(BUILD_SHARED)
 
-$(BUILD_SHARED): $(PHP_GLOBAL_OBJS) $(PHP_BINARY_OBJS) $(PHP_PHPDBG_OBJS)
+$(BUILD_SHARED): libphp_common.la $(PHP_BINARY_OBJS) $(PHP_PHPDBG_OBJS)
 	$(BUILD_PHPDBG_SHARED)
 
-$(BUILD_BINARY): $(PHP_GLOBAL_OBJS) $(PHP_BINARY_OBJS) $(PHP_PHPDBG_OBJS)
+$(BUILD_BINARY): libphp_common.la $(PHP_BINARY_OBJS) $(PHP_PHPDBG_OBJS)
 	$(BUILD_PHPDBG)
 
 %.c: %.y
diff -urNp -x '*.orig' php-7.0.33.org/sapi/phpdbg/config.m4 php-7.0.33/sapi/phpdbg/config.m4
--- php-7.0.33.org/sapi/phpdbg/config.m4	2018-12-04 18:23:02.000000000 +0100
+++ php-7.0.33/sapi/phpdbg/config.m4	2021-08-23 23:35:29.816640040 +0200
@@ -40,7 +40,7 @@ if test "$BUILD_PHPDBG" = "" && test "$P
   
   BUILD_PHPDBG="\$(LIBTOOL) --mode=link \
         \$(CC) -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \
-                \$(PHP_GLOBAL_OBJS) \
+                libphp_common.la \
                 \$(PHP_BINARY_OBJS) \
                 \$(PHP_PHPDBG_OBJS) \
                 \$(EXTRA_LIBS) \
@@ -50,7 +50,7 @@ if test "$BUILD_PHPDBG" = "" && test "$P
 
   BUILD_PHPDBG_SHARED="\$(LIBTOOL) --mode=link \
         \$(CC) -shared -Wl,-soname,libphpdbg.so -export-dynamic \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS_PROGRAM) \$(LDFLAGS) \$(PHP_RPATHS) \
-                \$(PHP_GLOBAL_OBJS) \
+                libphp_common.la \
                 \$(PHP_BINARY_OBJS) \
                 \$(PHP_PHPDBG_OBJS) \
                 \$(EXTRA_LIBS) \
