Index: php4/main/rfc1867.c
===================================================================
RCS file: /repository/php4/main/rfc1867.c,v
retrieving revision 1.55
retrieving revision 1.56
diff -u -r1.55 -r1.56
--- php4/main/rfc1867.c	2000/12/08 14:28:14	1.55
+++ php4/main/rfc1867.c	2001/01/03 10:52:26	1.56
@@ -167,11 +167,20 @@
 					SAFE_RETURN;
 				}
 				loc = memchr(ptr, '\n', rem);
-				name = strstr(ptr, " name=\"");
+				name = strstr(ptr, " name=");
 				if (name && name < loc) {
-					name += 7;
+					name += 6;
 					s = memchr(name, '\"', loc - name);
-					if (!s) {
+					if ( name == s ) { 
+						name++;
+						s = memchr(name, '\"', loc - name);
+						if(!s) {
+							php_error(E_WARNING, "File Upload Mime headers garbled name: [%c%c%c%c%c]", *name, *(name + 1), *(name + 2), *(name + 3), *(name + 4));
+							SAFE_RETURN;
+						}
+					} else if(!s) {
+						s = loc;
+					} else {
 						php_error(E_WARNING, "File Upload Mime headers garbled name: [%c%c%c%c%c]", *name, *(name + 1), *(name + 2), *(name + 3), *(name + 4));
 						SAFE_RETURN;
 					}
