--- php-src-PHP-7.4/ext/spl/config.m4~	2019-05-04 15:24:15.000000000 +0300
+++ php-src-PHP-7.4/ext/spl/config.m4	2019-05-05 18:45:49.109324631 +0300
@@ -1,5 +1,5 @@
 dnl config.m4 for extension SPL
 
-  PHP_NEW_EXTENSION(spl, php_spl.c spl_functions.c spl_engine.c spl_iterators.c spl_array.c spl_directory.c spl_exceptions.c spl_observer.c spl_dllist.c spl_heap.c spl_fixedarray.c, no,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+  PHP_NEW_EXTENSION(spl, php_spl.c spl_functions.c spl_engine.c spl_iterators.c spl_array.c spl_directory.c spl_exceptions.c spl_observer.c spl_dllist.c spl_heap.c spl_fixedarray.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
   PHP_INSTALL_HEADERS([ext/spl], [php_spl.h spl_array.h spl_directory.h spl_engine.h spl_exceptions.h spl_functions.h spl_iterators.h spl_observer.h spl_dllist.h spl_heap.h spl_fixedarray.h])
   PHP_ADD_EXTENSION_DEP(spl, pcre, true)
--- php-7.1.0alpha1/ext/pdo/pdo.c~	2016-06-07 23:56:47.000000000 +0300
+++ php-7.1.0alpha1/ext/pdo/pdo.c	2016-06-13 23:24:51.238259131 +0300
@@ -117,7 +117,9 @@
 
 /* {{{ pdo_functions[] */
 static const zend_module_dep pdo_deps[] = {
+#ifdef HAVE_SPL && !COMPILE_DL_SPL
 	ZEND_MOD_REQUIRED("spl")
+#endif
 	ZEND_MOD_END
 };
 /* }}} */
--- php-5.3.1/ext/phar/Makefile.frag~	2009-07-23 15:48:04.000000000 +0000
+++ php-5.3.1/ext/phar/Makefile.frag	2009-11-30 16:10:29.687175948 +0000
@@ -10,7 +10,7 @@
 		$(top_srcdir)/build/shtool echo -n -- "$(top_builddir)/$(SAPI_CLI_PATH) -n"; \
 		if test "x$(PHP_MODULES)" != "x"; then \
 		$(top_srcdir)/build/shtool echo -n -- " -d extension_dir=$(top_builddir)/modules"; \
-		for i in bz2 zlib phar; do \
+		for i in spl bz2 zlib phar; do \
 			if test -f "$(top_builddir)/modules/$$i.la"; then \
 				. $(top_builddir)/modules/$$i.la; $(top_srcdir)/build/shtool echo -n -- " -d extension=$$dlname"; \
 			fi; \
